// @porschehn/navigation/dist/collection/components/menu-item/menu-item.js
import { h, Host } from "@stencil/core";
import { ClickElementTypes, EventActions } from "../../utility/constants";
import { isString } from "../../utility/helper";
import MenuItemText from "./menu-item-text";
import Icon from "./menu-item-icon";
import { sendDataLayerEvent } from "../../services/analytics/data-layer-event-sender";
export class MenuItem {
  constructor() {
    this.text = '';
    this.analyticsId = undefined;
    this.subtext = '';
    this.fontWeight = 'regular';
    this.elementId = '';
    this.size = 'inherit';
    this.iconLeft = '';
    this.iconRight = '';
    this.link = '';
    this.isLoading = false;
    this.datalayerEventAction = EventActions.NAVIGATION_CLICK;
    this.customDatalayerProperties = {};
    this.target = '_self';
    this.variant = '';
    this.isModels = false;
  }
  sendDataLayerEvent(clickElementType, elementId) {
    sendDataLayerEvent({
      analyticsId: this.analyticsId,
      text: this.text,
      elementId,
      clickElementType,
      link: this.link,
      customDatalayerProperties: this.customDatalayerProperties,
      datalayerEventAction: this.datalayerEventAction
    });
  }
  existsSlot(slotName) {
    return this.el.querySelector(`[slot="${slotName}"]`) !== null;
  }
  render() {
    const body = (h("div", { class: "phn-flex-row phn-align-center" }, this.existsSlot('media') ? (h("div", { class: "media-container skeleton-item" }, h("slot", { name: "media" }))) : null, h(Icon, { isLoading: this.isLoading, icon: this.iconLeft }), !this.isModels && isString(this.text) ? (h(MenuItemText, { size: this.size, text: this.text, subtext: this.subtext, fontWeight: this.fontWeight })) : undefined, this.existsSlot('suffix') ? (h("div", null, h("slot", { name: "suffix" }))) : null));
    if (isString(this.link)) {
      return (h(Host, { class: this.variant }, h("phn-p-link-pure", { class: `${isString(this.iconLeft) ? 'has-icon' : ''}`, icon: isString(this.iconRight) ? this.iconRight : 'none', "align-label": "left", stretch: isString(this.iconRight) ? 'true' : 'false', "data-id": this.elementId, target: this.target, onClick: () => {
          this.sendDataLayerEvent(ClickElementTypes.Link, this.elementId);
        } }, h("a", { href: this.link }, body))));
    }
    return (h(Host, { class: this.variant }, h("phn-p-button-pure", { class: `${isString(this.iconLeft) ? 'has-icon' : ''}`, size: "inherit", icon: isString(this.iconRight) ? this.iconRight : 'none', "align-label": "left", stretch: isString(this.iconRight) ? 'true' : 'false', "data-id": this.elementId, onClick: () => {
        this.sendDataLayerEvent(ClickElementTypes.Button, this.elementId);
      } }, body)));
  }
  static get is() { return "phn-menu-item"; }
  static get encapsulation() { return "scoped"; }
  static get originalStyleUrls() {
    return {
      "$": ["menu-item.scss"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["menu-item.css"]
    };
  }
  static get properties() {
    return {
      "text": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "text",
        "reflect": false,
        "defaultValue": "''"
      },
      "analyticsId": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": true,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "analytics-id",
        "reflect": false
      },
      "subtext": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "subtext",
        "reflect": false,
        "defaultValue": "''"
      },
      "fontWeight": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "font-weight",
        "reflect": false,
        "defaultValue": "'regular'"
      },
      "elementId": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "element-id",
        "reflect": false,
        "defaultValue": "''"
      },
      "size": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "size",
        "reflect": false,
        "defaultValue": "'inherit'"
      },
      "iconLeft": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "icon-left",
        "reflect": false,
        "defaultValue": "''"
      },
      "iconRight": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "icon-right",
        "reflect": false,
        "defaultValue": "''"
      },
      "link": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "link",
        "reflect": false,
        "defaultValue": "''"
      },
      "isLoading": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "is-loading",
        "reflect": false,
        "defaultValue": "false"
      },
      "datalayerEventAction": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "EventAction",
          "resolved": "\"PAGHomeNav_General_Load\" | \"PAGHomeNav_CountryRecommender_Load\" | \"PAGHomeNav_CountryRecommenderChange_Click\" | \"PAGHomeNav_CountryRecommenderClose_Click\" | \"PAGHomeNav_Crest_Click\" | \"PAGHomeNav_Navigation_Load\" | \"PAGHomeNav_ModelRangeOverlay_Load\" | \"PAGHomeNav_MenuBack_Click\" | \"PAGHomeNav_MenuClose_Click\" | \"PAGHomeNav_MetaFunction_Click\" | \"PAGHomeNav_Model_Click\" | \"PAGHomeNav_ModelMetaFunction_Click\" | \"PAGHomeNav_ModelCTA_Click\" | \"PAGHomeNav_ModelRange_Click\" | \"PAGHomeNav_More_Click\" | \"PAGHomeNav_Navigation_Click\" | \"PAGHomeNav_NavigationLink_Click\" | \"PAGHomeNav_FindADealerSearchScreen_Load\" | \"PAGHomeNav_FindADealerResultsScreen_Load\" | \"PAGHomeNav_FindADealerSearch_Click\" | \"PAGHomeNav_FindADealerLocateMe_Click\" | \"PAGHomeNav_FindADealerDealer_Click\" | \"PAGHomeNav_FindADealerDealerScreen_Load\" | \"PAGHomeNav_FindADealerDealerContact_Click\" | \"PAGHomeNav_FindADealerMap_Click\" | \"PAGHomeNav_IconContextLink_Click\" | \"PAGHomeNav_Drawer_Load\" | \"PAGHomeNav_IconGlobalLink_Click\"",
          "references": {
            "EventAction": {
              "location": "import",
              "path": "../../utility/constants",
              "id": "src/utility/constants/index.ts::EventAction"
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "datalayer-event-action",
        "reflect": false,
        "defaultValue": "EventActions.NAVIGATION_CLICK"
      },
      "customDatalayerProperties": {
        "type": "unknown",
        "mutable": false,
        "complexType": {
          "original": "TDataLayer",
          "resolved": "{ basicProduct?: TBasicProduct | undefined; car?: TCar | undefined; componentAudio?: TComponentAudio | undefined; componentClick?: TComponentClick | undefined; componentDisplay?: TComponentDisplay | undefined; componentFile?: TComponentFile | undefined; componentFilter?: TComponentFilter | undefined; componentForm?: TComponentForm | undefined; componentNavigation?: TComponentNavigation | undefined; componentSearch?: TComponentSearch | undefined; componentSort?: TComponentSort | undefined; componentVideo?: TComponentVideo | undefined; consents?: TConsents | undefined; context?: TContext | undefined; financialProduct?: TFinancialProduct | undefined; ids?: TIds | undefined; pageExperience?: TPageExperience | undefined; partner?: TPartner | undefined; personalization?: TPersonalization | undefined; preferences?: TPreferences | undefined; process?: TProcess | undefined; serviceProduct?: TServiceProduct | undefined; socialShareConsents?: TSocialShareConsents | undefined; visitor?: TVisitor | undefined; }",
          "references": {
            "TDataLayer": {
              "location": "import",
              "path": "@porsche/globaldatalayer-events/projects/pcom/all/types",
              "id": ""
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "defaultValue": "{}"
      },
      "target": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "target",
        "reflect": false,
        "defaultValue": "'_self'"
      },
      "variant": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "variant",
        "reflect": false,
        "defaultValue": "''"
      },
      "isModels": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "is-models",
        "reflect": false,
        "defaultValue": "false"
      }
    };
  }
  static get elementRef() { return "el"; }
}
