// @porsche-design-system/components-js/styles/esm/hover/getHoverStyle.mjs
import { themeLightStateHover } from '../theme/themeLightStateHover.mjs';
import { borderRadiusSmall } from '../border/borderRadiusSmall.mjs';
import { borderRadiusMedium } from '../border/borderRadiusMedium.mjs';

const offsetHorizontal = '2px';
const getHoverStyle = (opts) => {
    const borderRadius = opts?.borderRadius === 'small'
        ? borderRadiusSmall
        : opts?.borderRadius === 'medium'
            ? borderRadiusMedium
            : opts?.borderRadius || borderRadiusSmall;
    return {
        borderRadius,
        marginLeft: `-${offsetHorizontal}`,
        marginRight: `-${offsetHorizontal}`,
        paddingLeft: offsetHorizontal,
        paddingRight: offsetHorizontal,
        // TODO: how can we test this later in vrt?
        '@media(hover:hover)': {
            '&': {
                transition: 'background var(--p-transition-duration, .24s) ease',
            },
            '&:hover': {
                // ...frostedGlassStyle,
                background: themeLightStateHover, // hover color is equal for light and dark theme
            },
        },
    };
};

export { getHoverStyle };
