// @porschehn/navigation/dist/collection/test/experiments.testdata.js
/* istanbul ignore file */
const experimentsDefinitions = {
  experiment2: {
    name: 'experiment2',
    includedTargets: [{ attribute: 'url', type: 'starts_with', value: 'http://www.example.org', inverted: '0' }],
    excludedTargets: [],
    variations: [
      {
        name: 'original',
        activate: function () {
          // usually nothing needs to be done here
        }
      },
      {
        name: 'blue',
        activate: function () {
          // eg $('#btn').attr('style', 'color: ' + this.name + ';');
        }
      }
    ]
  }
};
export const contentDefinitions = [
  {
    name: 'experiment1',
    percentage: 100,
    includedTargets: [{ attribute: 'url', inverted: '0', type: 'any', value: '' }],
    excludedTargets: [],
    variations: [
      {
        name: 'variation1',
        percentage: 100,
        children: [
          {
            id: '1',
            text: 'var1 text',
            link: 'var1 link',
            children: []
          }
        ]
      }
    ]
  }
];
export const testMenu = {
  id: '1',
  text: 'test text',
  link: 'test link',
  children: [
    {
      id: '2',
      text: 'var text',
      link: 'var link',
      children: []
    }
  ],
  experienceName: 'experiment1'
};
export default experimentsDefinitions;
