// @porschehn/navigation/dist/collection/test/test-data.js
/* istanbul ignore file */
import { RoutingKeys } from "../entities/routing-key";
import { EventActions } from "../utility/constants";
export const firstAccessToken = 'first-access-token';
export const secondAccessToken = 'second-access-token';
export const expiredAccessToken = 'expired-access-token';
export const validLocale = 'de-CH';
export const notFoundLocale = 'xx-YY';
export const sixDigitLocale = 'xx-YY-ZZ';
export const firstUserProfileData = {
  ciamId: 'f0hasos1or8p7nbi',
  porscheId: 'm.doe@porsche.ag',
  dealers: [],
  firstName: 'Mirinda',
  lastName: 'Doe',
  language: 'en-gb',
  localizedFirstName: 'Mirinda',
  localizedLastName: 'Doe',
  status: 'CONFIRMED',
  preferredCountry: 'GB'
};
export const secondUserProfileData = {
  firstName: 'Second',
  lastName: 'User'
};
export const mockCoordinates = { latitude: 47.9999615, longitude: 11.3390089 };
export const mockInvalidCoordinates = { latitude: 999, longitude: 999 };
export const mockFailFetchCoordinates = { latitude: 1000, longitude: 1000 };
export const mockInvalidResponseCoordinates = { latitude: 1001, longitude: 1001 };
export const mockCoordinatesWithInvalidJson = { latitude: 1002, longitude: 1002 };
export const mockInvalidCountryCode = 'XX';
export const mockPPNDealerSearchResultsWithLocalizedAddress = [
  {
    ppnDealer: {
      id: '6a6d8e8c-5b04-4a4b-a119-66021b528788',
      porschePartnerNo: '4500248',
      ppnOrgId: '12716',
      name: 'Porsche Zentrum Stuttgart',
      nameLocalized: 'localized Porsche Zentrum Stuttgart',
      address: {
        street: 'Porscheplatz 9',
        postalCode: '70435',
        city: 'Stuttgart',
        state: 'BW',
        countryCode: 'DE'
      },
      addressLocalized: {
        street: 'localized Porscheplatz 9',
        city: 'localized Stuttgart',
        state: 'localized BW'
      },
      contactDetails: {
        phoneNumber: '+49 711 91126220',
        emailAddress: 'info@porsche-stuttgart.de',
        homepage: 'https://www.iamalsoahome.page',
        contactOpeningHours: [
          {
            day: 'MONDAY',
            open: '09:00',
            close: '19:00'
          }
        ]
      },
      coordinates: mockCoordinates
    },
    distance: { value: 3, unit: 'KM' }
  },
  {
    ppnDealer: {
      id: '14f3b8a2-b78d-4f96-b7b5-123124',
      porschePartnerNo: '4500248',
      ppnOrgId: '17038',
      name: 'Porsche Zentrum Stuttgart-Flughafen',
      nameLocalized: 'localized Porsche Zentrum Stuttgart-Flughafen',
      address: {
        street: 'Karl-Benz-Straße 15',
        postalCode: '70794',
        city: 'Filderstadt-Bernhausen',
        state: '',
        countryCode: 'DE'
      },
      addressLocalized: {
        street: 'localized Porscheplatz 9',
        city: 'localized Stuttgart',
        state: 'localized BW'
      },
      contactDetails: {
        phoneNumber: '+49-711-707008-0',
        emailAddress: 'info@porsche-stuttgart.com',
        homepage: 'http://www.porsche-stuttgart-flughafen.de/',
        contactOpeningHours: [
          {
            day: 'THURSDAY',
            open: '09:00',
            close: '19:00'
          }
        ]
      },
      coordinates: mockCoordinates
    },
    distance: { value: 2, unit: 'KM' }
  },
  {
    ppnDealer: {
      id: '14f3b8a2-b78d-4f96-b7b5-7b1b4723323542',
      porschePartnerNo: '4500248',
      ppnOrgId: '17037',
      name: 'Porsche Zentrum Böblingen',
      nameLocalized: 'localized Porsche Zentrum Böblingen',
      address: {
        street: 'Konrad-Zuse-Str. 2',
        postalCode: '71034',
        city: 'Böblingen',
        state: '',
        countryCode: 'DE'
      },
      addressLocalized: {
        street: 'localized Porscheplatz 9',
        city: 'localized Stuttgart',
        state: 'localized BW'
      },
      contactDetails: {
        phoneNumber: '+49-7031-45011-0',
        emailAddress: 'info@porsche-boeblingen.com',
        homepage: 'http://www.porsche-boeblingen.de/',
        contactOpeningHours: [
          {
            day: 'FRIDAY',
            open: '09:00',
            close: '19:00'
          }
        ]
      },
      coordinates: mockCoordinates
    },
    distance: { value: 4, unit: 'KM' }
  },
  {
    ppnDealer: {
      id: '14f3b8a2-b78d-4f96-b7b5-7b1b453b23542',
      porschePartnerNo: '4500248',
      ppnOrgId: '13249',
      name: 'Porsche Zentrum Reutlingen',
      nameLocalized: 'localized Porsche Zentrum Reutlingen',
      address: {
        street: 'Ernst-Abbe Str. 29',
        postalCode: '72770',
        city: 'Reutlingen',
        state: '',
        countryCode: 'DE'
      },
      addressLocalized: {
        street: 'localized Porscheplatz 9',
        city: 'localized Stuttgart',
        state: 'localized BW'
      },
      contactDetails: {
        phoneNumber: '+49-7121-3015-0',
        emailAddress: 'info@porsche-reutlingen.com',
        homepage: 'http://www.porsche-reutlingen.de/',
        contactOpeningHours: [
          {
            day: 'WEDNESDAY',
            open: '09:00',
            close: '14:00'
          }
        ]
      },
      coordinates: mockCoordinates
    },
    distance: { value: 2, unit: 'KM' }
  },
  {
    ppnDealer: {
      id: '14f3b8a2-b78d-4f96-b7b5-7b11b23542',
      porschePartnerNo: '4500248',
      ppnOrgId: '13249',
      name: 'Porsche Zentrum Göppingen',
      nameLocalized: 'localized Porsche Zentrum Göppingen',
      address: {
        street: 'Holzheimer STr. 1',
        postalCode: '73037',
        city: 'Göppingen',
        state: '',
        countryCode: 'DE'
      },
      addressLocalized: {
        street: 'localized Porscheplatz 9',
        city: 'localized Stuttgart',
        state: 'localized BW'
      },
      contactDetails: {
        phoneNumber: '+49-7161-97828-90',
        emailAddress: 'info@porsche-goeppingen.com',
        homepage: 'http://www.porsche-goeppingen.de/',
        contactOpeningHours: [
          {
            day: 'MONDAY',
            open: '08:00',
            close: '18:00'
          }
        ]
      },
      coordinates: mockCoordinates
    },
    distance: { value: 1, unit: 'KM' }
  },
  {
    ppnDealer: {
      id: '36e8fba0-e931-11dc-ada3-c822588bbdee',
      name: 'Porsche Zentrum Stuttgart',
      nameLocalized: 'localized Porsche Zentrum Stuttgart',
      porschePartnerNo: '1150190',
      ppnOrgId: '11556',
      address: {
        street: 'Porscheplatz 9',
        postalCode: '70435',
        city: 'Stuttgart',
        state: 'BW',
        countryCode: 'DE'
      },
      addressLocalized: {
        street: 'localized Porscheplatz 9',
        city: 'localized Stuttgart',
        state: 'localized BW'
      },
      contactDetails: {
        phoneNumber: '+49 711 91126220',
        emailAddress: 'info@porsche-stuttgart.de',
        homepage: 'https://www.porsche-stuttgart.de',
        contactOpeningHours: [
          {
            day: 'TUESDAY',
            open: '09:00',
            close: '19:00'
          }
        ]
      },
      coordinates: {
        latitude: 48.775833,
        longitude: 9.182778
      }
    },
    distance: { value: 0.0115, unit: 'KM' }
  }
];
export const mockPPNDealerOneSearchResults = [
  {
    ppnDealer: {
      id: '6a6d8e8c-5b04-4a4b-a119-66021b528788',
      porschePartnerNo: '4500248',
      ppnOrgId: '12716',
      name: 'Porsche Zentrum Stuttgart',
      nameLocalized: 'localized Porsche Zentrum Stuttgart',
      address: {
        street: 'Porscheplatz 9',
        postalCode: '70435',
        city: 'Stuttgart',
        state: 'BW',
        countryCode: 'DE'
      },
      addressLocalized: {
        street: 'localized Porscheplatz 9',
        city: 'localized Stuttgart',
        state: 'localized BW'
      },
      contactDetails: {
        phoneNumber: '+49 711 91126220',
        emailAddress: 'info@porsche-stuttgart.de',
        homepage: 'https://www.iamalsoahome.page',
        contactOpeningHours: [
          {
            day: 'MONDAY',
            open: '07:00',
            close: '15:00'
          }
        ]
      },
      coordinates: mockCoordinates
    },
    distance: { value: 3, unit: 'KM' }
  }
];
export const mockPPNDealer = {
  id: 'abcde',
  porschePartnerNo: '1234567',
  ppnOrgId: '13249',
  name: 'Porsche Zentrum Berlin',
  nameLocalized: 'localized Porsche Zentrum Berlin',
  address: {
    street: 'Franklinstr.23',
    postalCode: '10587',
    city: 'Berlin',
    state: '',
    countryCode: 'DE'
  },
  addressLocalized: {
    street: '',
    city: '',
    state: ''
  },
  contactDetails: {
    phoneNumber: '+49-711-911-12345',
    emailAddress: 'none@none.com',
    homepage: 'https://www.iamstillahome.page',
    contactOpeningHours: [
      {
        day: 'TUESDAY',
        open: '09:00',
        close: '19:00'
      }
    ]
  },
  coordinates: mockCoordinates
};
export const mockPPNDistance = {
  value: 14,
  unit: 'km'
};
export const mockPredictions = [
  {
    text: 'Stuttgart, Germany',
    id: 'alkdfalsdkjfaevh'
  },
  {
    text: 'Stuart, FL, USA',
    id: 'rbvlajdvkljabfgkajsdbav'
  },
  {
    text: 'StuyTown, 1st Avenue, New York, NY, USA',
    id: 'alksdhfasldkjhfkjhnnl'
  }
];
export const mockPlaceFromCoordinates = '12345 Stuttgart, Germany';
export const mockSeriesContent = {
  id: 'mainmenu.718',
  name: '718',
  all: {
    id: 'mainmenu.718.modeloverview',
    text: '718 Models',
    description: '',
    link: 'https://www.porsche.com/uk/models/718',
    icon: ''
  },
  image: '718',
  links: [
    {
      id: 'mainmenu.718.configure',
      text: 'Configure',
      description: '',
      link: 'https://www.porsche.com/uk/modelstart/all/?modelrange=718',
      icon: 'configurate'
    },
    {
      id: 'mainmenu.718.compare',
      text: 'Compare',
      description: '',
      link: 'https://www.porsche.com/uk/models/718/?compare',
      icon: 'compare'
    },
    {
      id: 'mainmenu.718.findvehicles',
      text: 'Find New and Pre-owned Vehicles',
      description: '',
      link: 'https://finder.porsche.com/gb/en_GB/search/list?modelSeriesKeys=718',
      icon: 'search'
    },
    {
      id: 'mainmenu.718.booktestdrive',
      text: 'Book Test Drive',
      description: '',
      link: 'https://www.porsche.com/uk/testdrive/?modelsddPInt=id_ao1_primint%7Cid_ao2_primint',
      icon: 'steering-wheel'
    }
  ],
  tags: [
    {
      text: 'Gasoline',
      icon: 'fuel-station'
    }
  ],
  bodyTypes: [
    {
      id: 'mainmenu.718.cayman.models',
      description: '718 description',
      name: '718 Cayman Models',
      image: '718-cayman',
      link: 'https://www.porsche.com/uk/models/718/718-models/718-cayman/'
    },
    {
      id: 'mainmenu.718.boxster.models',
      description: '',
      name: '718 Boxster Models',
      image: '718-boxster',
      link: 'https://www.porsche.com/uk/models/718/718-models/718-boxster/'
    },
    {
      id: 'mainmenu.718.caymangt4.models',
      description: '',
      name: '718 Cayman GT4',
      image: '718-cayman-gt4',
      link: 'https://www.porsche.com/uk/models/718/718-cayman-gt4/'
    },
    {
      id: 'mainmenu.718.caymangt4rs.models',
      description: '',
      name: 'Cayman GT4 RS',
      image: '718-cayman-gt4-rs',
      link: 'https://www.porsche.com/uk/models/718/718-cayman-gt4-rs'
    },
    {
      id: 'mainmenu.718.spyder.models',
      description: '',
      name: '718 Spyder',
      image: '718-spyder',
      link: 'https://www.porsche.com/uk/models/718/718-spyder/'
    }
  ]
};
export const mockModelsContent = {
  id: 'mainmenu.models',
  text: 'Models',
  all: {
    id: 'mainmenu.allmodelsoverview',
    text: 'All Series & Models',
    description: '',
    link: 'https://www.porsche.com/uk/models/',
    icon: ''
  },
  series: [
    mockSeriesContent,
    {
      id: 'mainmenu.911',
      name: '911',
      all: {
        id: 'mainmenu.911.modeloverview',
        text: '911 Models',
        description: '',
        link: 'https://www.porsche.com/uk/models/911',
        icon: ''
      },
      image: '911',
      links: [
        {
          id: 'mainmenu.911.configure',
          text: 'Configure',
          description: '',
          link: 'https://www.porsche.com/uk/modelstart/all/?modelrange=911',
          icon: 'configurate'
        },
        {
          id: 'mainmenu.911.compare',
          text: 'Compare',
          description: '',
          link: 'https://www.porsche.com/uk/models/911/?compare',
          icon: 'compare'
        },
        {
          id: 'mainmenu.911.findvehicles',
          text: 'Find New and Pre-owned Vehicles',
          description: '',
          link: 'https://finder.porsche.com/gb/en_GB/search/list?modelSeriesKeys=911',
          icon: 'search'
        },
        {
          id: 'mainmenu.911.booktestdrive',
          text: 'Book Test Drive',
          description: '',
          link: 'https://www.porsche.com/uk/testdrive/?modelsddPInt=id_ao3_primint',
          icon: 'steering-wheel'
        }
      ],
      tags: [
        {
          text: 'Gasoline',
          icon: 'fuel-station'
        }
      ],
      bodyTypes: [
        {
          id: 'mainmenu.911.carrera1.models',
          description: '',
          name: 'Carrera Models',
          image: '911-carrera',
          link: 'https://www.porsche.com/uk/models/911/911-models/carrera/'
        },
        {
          id: 'mainmenu.911.carreracabrio.models',
          description: '',
          name: 'Carrera Cabriolet Models',
          image: '911-carrera-cabriolet',
          link: 'https://www.porsche.com/uk/models/911/911-models/carrera-cabriolet/'
        },
        {
          id: 'mainmenu.911.targa.models',
          description: '',
          name: 'Targa Models',
          image: '911-targa',
          link: 'https://www.porsche.com/uk/models/911/911-models/targa-4/'
        },
        {
          id: 'mainmenu.911.turbo.models',
          description: '',
          name: '911 Turbo Models',
          image: '911-turbo',
          link: 'https://www.porsche.com/uk/models/911/911-turbo-models/911-turbo/'
        },
        {
          id: 'mainmenu.911.turbocabrio.models',
          description: '',
          name: 'Turbo Cabriolet Models',
          image: '911-turbo-cabriolet',
          link: 'https://www.porsche.com/uk/models/911/911-turbo-models/911-turbo-cabriolet/'
        },
        {
          id: 'mainmenu.911.gt3.models',
          description: '',
          name: '911 GT3 Models',
          image: '911-gt3',
          link: 'https://www.porsche.com/uk/models/911/911-gt3-models/911-gt3/'
        },
        {
          id: 'mainmenu.911.gt3rs.models',
          description: '',
          name: '911 GT3 RS',
          image: '911-gt3-rs',
          link: 'https://www.porsche.com/uk/models/911/911-gt3-models/911-gt3-touring/'
        },
        {
          id: 'mainmenu.911.sportclassic.models',
          description: '',
          name: '911 Sport Classic',
          image: '911-sportclassic',
          link: 'https://www.porsche.com/uk/models/911/911-sport-classic/'
        }
      ]
    },
    {
      id: 'mainmenu.taycan',
      name: 'Taycan',
      all: {
        id: 'mainmenu.taycan.modeloverview',
        text: 'Taycan Models',
        description: '',
        link: 'https://www.porsche.com/uk/models/taycan',
        icon: ''
      },
      image: 'taycan',
      links: [
        {
          id: 'mainmenu.taycan.configure',
          text: 'Configure',
          description: '',
          link: 'https://www.porsche.com/uk/modelstart/all/?modelrange=taycan',
          icon: 'configurate'
        },
        {
          id: 'mainmenu.taycan.compare',
          text: 'Compare',
          description: '',
          link: 'https://www.porsche.com/uk/models/taycan/?compare',
          icon: 'compare'
        },
        {
          id: 'mainmenu.taycan.findvehicles',
          text: 'Find New and Pre-owned Vehicles',
          description: '',
          link: 'https://finder.porsche.com/gb/en_GB/search/list?modelSeriesKeys=taycan',
          icon: 'search'
        },
        {
          id: 'mainmenu.taycan.booktestdrive',
          text: 'Book Test Drive',
          description: '',
          link: 'https://www.porsche.com/uk/testdrive/?modelsddPInt=id_ao3_primint',
          icon: 'steering-wheel'
        }
      ],
      tags: [
        {
          text: 'Electric',
          icon: 'fuel-station'
        }
      ],
      bodyTypes: []
    },
    {
      id: 'mainmenu.panamera',
      name: 'Panamera',
      all: {
        id: 'mainmenu.panamera.modeloverview',
        text: 'Panamera Models',
        description: '',
        link: 'https://www.porsche.com/uk/models/panamera',
        icon: ''
      },
      image: 'panamera',
      links: [
        {
          id: 'mainmenu.panamera.configure',
          text: 'Configure',
          description: '',
          link: 'https://www.porsche.com/uk/modelstart/all/?modelrange=panamera',
          icon: 'configurate'
        },
        {
          id: 'mainmenu.panamera.compare',
          text: 'Compare',
          description: '',
          link: 'https://www.porsche.com/uk/models/panamera/?compare',
          icon: 'compare'
        },
        {
          id: 'mainmenu.panamera.findvehicles',
          text: 'Find New and Pre-owned Vehicles',
          description: '',
          link: 'https://finder.porsche.com/gb/en_GB/search/list?modelSeriesKeys=panamera',
          icon: 'search'
        },
        {
          id: 'mainmenu.panamera.booktestdrive',
          text: 'Book Test Drive',
          description: '',
          link: 'https://www.porsche.com/uk/testdrive/?modelsddPInt=id_ao3_primint',
          icon: 'steering-wheel'
        }
      ],
      tags: [
        {
          text: 'Gasoline',
          icon: 'fuel-station'
        },
        {
          text: 'Hybrid',
          icon: 'fuel-station'
        }
      ],
      bodyTypes: []
    },
    {
      id: 'mainmenu.macan',
      name: 'Macan',
      all: {
        id: 'mainmenu.macan.modeloverview',
        text: 'Macan Models',
        description: '',
        link: 'https://www.porsche.com/uk/models/macan',
        icon: ''
      },
      image: 'macan',
      links: [
        {
          id: 'mainmenu.macan.configure',
          text: 'Configure',
          description: '',
          link: 'https://www.porsche.com/uk/modelstart/all/?modelrange=macan',
          icon: 'configurate'
        },
        {
          id: 'mainmenu.macan.compare',
          text: 'Compare',
          description: '',
          link: 'https://www.porsche.com/uk/models/panamera/?compare',
          icon: 'compare'
        },
        {
          id: 'mainmenu.macan.findvehicles',
          text: 'Find New and Pre-owned Vehicles',
          description: '',
          link: 'https://finder.porsche.com/gb/en_GB/search/list?modelSeriesKeys=macan',
          icon: 'search'
        },
        {
          id: 'mainmenu.macan.booktestdrive',
          text: 'Book Test Drive',
          description: '',
          link: 'https://www.porsche.com/uk/testdrive/?modelsddPInt=id_ao3_primint',
          icon: 'steering-wheel'
        }
      ],
      tags: [
        {
          text: 'Gasoline',
          icon: 'fuel-station'
        }
      ],
      bodyTypes: []
    },
    {
      id: 'mainmenu.cayenne',
      name: 'Cayenne',
      all: {
        id: 'mainmenu.macan.modeloverview',
        text: 'Cayenne Models',
        description: '',
        link: 'https://www.porsche.com/uk/models/cayenne',
        icon: ''
      },
      image: 'cayenne',
      links: [
        {
          id: 'mainmenu.cayenne.configure',
          text: 'Configure',
          description: '',
          link: 'https://www.porsche.com/uk/modelstart/all/?modelrange=cayenne',
          icon: 'configurate'
        },
        {
          id: 'mainmenu.cayenne.compare',
          text: 'Compare',
          description: '',
          link: 'https://www.porsche.com/uk/models/panamera/?compare',
          icon: 'compare'
        },
        {
          id: 'mainmenu.macan.findvehicles',
          text: 'Find New and Pre-owned Vehicles',
          description: '',
          link: 'https://finder.porsche.com/gb/en_GB/search/list?modelSeriesKeys=cayenne',
          icon: 'search'
        },
        {
          id: 'mainmenu.macan.booktestdrive',
          text: 'Book Test Drive',
          description: '',
          link: 'https://www.porsche.com/uk/testdrive/?modelsddPInt=id_ao3_primint',
          icon: 'steering-wheel'
        }
      ],
      tags: [
        {
          text: 'Gasoline',
          icon: 'fuel-station'
        },
        {
          text: 'Hybrid',
          icon: 'fuel-station'
        }
      ],
      bodyTypes: []
    }
  ]
};
export const mockCountryRecommenderContent = {
  statement: 'You are currently seeing contents for:',
  region: 'Great Britain',
  language: 'English',
  change: {
    text: 'Change',
    description: '',
    id: 'mainmenu.banner.change'
  }
};
export const mockMyPorscheContent = {
  id: 'mainmenu.myporsche',
  text: 'My Porsche',
  link: 'https://my.porsche.com/core/gb/en_GB/home',
  loggedInFallback: 'Logged in',
  login: {
    id: 'mainmenu.myporsche.login',
    text: 'Log in',
    description: '',
    link: 'https://my.porsche.com/core/gb/en_GB/home',
    icon: ''
  },
  loginText: 'Login / Register',
  logout: {
    id: 'mainmenu.myporsche.logout',
    text: 'Logout',
    description: '',
    link: 'https://login.porsche.com/auth/api/v1/gb/en_GB/public/logout',
    icon: ''
  },
  messages: {
    id: 'mainmenu.myporsche.messages',
    text: 'Messages',
    description: '',
    link: 'https://inbox.porsche.com/gb/en_GB/messages',
    icon: ''
  },
  more: {
    id: 'mainmenu.myporsche.more',
    text: 'Discover more',
    description: '',
    link: 'https://www.porsche.com/uk/accessoriesandservice/myporsche/',
    icon: ''
  },
  signup: {
    id: 'mainmenu.myporsche.signup',
    text: 'Register now',
    description: '',
    link: 'https://signup.porsche.com/gb/en_GB/register',
    icon: ''
  },
  signupText: 'to discover the digital world of Porsche.',
  links: [
    {
      id: 'mainmenu.myporsche.overview',
      text: 'Overview',
      description: '',
      link: 'https://my.porsche.com/core/gb/en_GB/home',
      icon: ''
    },
    {
      id: 'mainmenu.myporsche.accountadministration',
      text: 'Profile settings',
      description: '',
      link: 'https://profile.porsche.com/myprofile/gb/en_GB/personal-data',
      icon: ''
    },
    {
      id: 'mainmenu.myporsche.contact',
      text: 'Contact & Support',
      description: '',
      link: 'https://my.porsche.com/core/gb/en_GB/contact',
      icon: ''
    }
  ],
  savedSearches: {
    id: 'mainmenu.myporsche.savedsearches',
    text: 'Saved Searches',
    description: '',
    link: 'https://finder.porsche.com/gb/en_GB/saved-searches',
    icon: ''
  },
  savedVehicles: {
    id: 'mainmenu.myporsche.savedvehicles',
    text: 'Saved Vehicles',
    description: '',
    link: 'https://finder.porsche.com/gb/en_GB/favorites',
    icon: ''
  }
};
export const mockDealerContent = {
  id: 'mainmenu.finddealer',
  text: 'Find a Dealer',
  numberOfDealers: 42,
  search: {
    id: 'mainmenu.finddealer.headline',
    text: 'Find a Dealer',
    form: {
      inputPlaceholder: 'Location',
      inputDescription: 'Input description'
    },
    clear: {
      text: 'Clear',
      description: 'Clear description',
      id: 'mainmenu.finddealer.clear'
    },
    or: 'Or',
    geolocation: {
      text: 'Current location',
      description: 'Current location description',
      id: 'mainmenu.finddealer.currentlocation'
    },
    dropdown: {
      province: 'Province',
      city: 'City',
      selectAProvince: 'Select a Province',
      selectACity: 'Select a City'
    },
    load: 'Load',
    map: {
      id: 'mainmenu.finddealer.map',
      text: 'All Dealers Map',
      description: '',
      link: 'http://www.porsche.com/all-dealers-map',
      icon: ''
    },
    outboundLink: {
      id: 'mainmenu.finddealer',
      text: 'Porsche Singapore',
      description: '',
      link: 'http://www.porsche.com/outbound-link',
      icon: ''
    }
  },
  details: {
    id: 'mainmenu.finddealer.details',
    backToSearch: 'Back to search',
    bookService: {
      id: 'mainmenu.finddealer.service',
      text: 'Service',
      description: '',
      link: 'http://www.porsche.com/gb/en_GB/test-service',
      icon: ''
    },
    testDrive: {
      id: 'mainmenu.finddealer.testdrive',
      text: 'Testdrive',
      description: '',
      link: 'http://contact.porsche.com/china/en/test-drive/',
      icon: ''
    },
    finder: {
      id: 'mainmenu.finddealer.findercars',
      text: 'Finder cars',
      description: '',
      link: 'http://www.porsche.com/gb/en_GB/test-finder',
      icon: ''
    }
  },
  openingHours: {
    monday: 'Mon',
    tuesday: 'Tue',
    wednesday: 'Wed',
    thursday: 'Thur',
    friday: 'Fri',
    saturday: 'Sat',
    sunday: 'Sun',
    daily_lunch: 'Daily Lunch'
  }
};
export const mockDealerContentForMarketWithOneDealer = {
  id: 'mainmenu.finddealer',
  text: 'Find a Dealer',
  numberOfDealers: 1,
  search: {
    id: 'mainmenu.finddealer.headline',
    text: 'Find a Dealer',
    form: {
      inputPlaceholder: 'Location',
      inputDescription: 'Input description'
    },
    clear: {
      text: 'Clear',
      description: 'Clear description',
      id: 'mainmenu.finddealer.clear'
    },
    or: 'Or',
    geolocation: {
      text: 'Current location',
      description: 'Current location description',
      id: 'mainmenu.finddealer.currentlocation'
    },
    dropdown: {
      province: 'Province',
      city: 'City',
      selectAProvince: 'Select a Province',
      selectACity: 'Select a City'
    },
    load: 'Load',
    map: {
      id: 'mainmenu.finddealer.map',
      text: 'All Dealers Map',
      description: '',
      link: 'http://www.porsche.com/all-dealers-map',
      icon: ''
    },
    outboundLink: {
      id: 'mainmenu.finddealer',
      text: 'Porsche Singapore',
      description: '',
      link: 'http://www.porsche.com/outbound-link',
      icon: ''
    }
  },
  details: {
    id: 'mainmenu.finddealer.details',
    backToSearch: 'Back to search',
    bookService: {
      id: 'mainmenu.finddealer.service',
      text: 'Service',
      description: '',
      link: 'http://www.porsche.com/gb/en_GB/test-service',
      icon: ''
    },
    testDrive: {
      id: 'mainmenu.finddealer.testdrive',
      text: 'Testdrive',
      description: '',
      link: 'http://contact.porsche.com/china/en/test-drive/',
      icon: ''
    },
    finder: {
      id: 'mainmenu.finddealer.findercars',
      text: 'Finder cars',
      description: '',
      link: 'http://www.porsche.com/gb/en_GB/test-finder',
      icon: ''
    }
  },
  openingHours: {
    monday: 'Mon',
    tuesday: 'Tue',
    wednesday: 'Wed',
    thursday: 'Thur',
    friday: 'Fri',
    saturday: 'Sat',
    sunday: 'Sun',
    daily_lunch: 'Daily Lunch'
  }
};
export const mockErrorContent = {
  wrong: 'Wrong',
  retry: 'Retry',
  noResults: 'No results',
  noInput: 'No input',
  invalidInput: 'Invalid input',
  geolocationDenied: 'Geolocation denied'
};
export const mockShopLinkContent = {
  id: 'mainmenu.shops',
  text: 'Shop',
  link: 'https://shop.porsche.com/de/de-DE',
  children: [],
  additionalContent: [],
  availableLocales: [],
  availableLocalesOnlyForShop: [],
  logoLinksToShopLocales: []
};
export const mockShopContent = {
  id: 'mainmenu.shops',
  text: 'Shop',
  link: 'https://shop.porsche.com/gb/en-GB/',
  cart: 'https://shop.porsche.com/gb/en-GB/cart',
  wishlist: {
    id: 'mainmenu.shops.wishlist',
    text: 'Wishlist',
    description: '',
    link: 'https://shop.porsche.com/gb/en-GB/wishlist',
    icon: ''
  },
  children: [
    {
      id: 'mainmenu.services.accessories',
      text: 'Accessories, Services & Fashion',
      link: '',
      children: [
        {
          id: 'mainmenu.service.tequipment',
          text: 'Tequipment – Genuine Accessories',
          link: 'https://shop.porsche.com/gb/en/teq_brand',
          children: []
        },
        {
          id: 'mainmenu.services.lifestyle',
          text: 'Lifestyle – Fashion & Accessories',
          link: 'https://shop.porsche.com/gb/en/',
          children: []
        }
      ]
    },
    {
      id: 'mainmenu.shops.whatever',
      text: 'More Shop stuff',
      link: 'more-shop-stuff-link',
      children: []
    }
  ],
  additionalContent: [
    {
      id: 'wishlist',
      text: 'Wishlist ({wishlistItemsNumber})',
      link: '',
      iconUrl: 'https://wishlisticon.icon'
    },
    {
      id: 'login',
      text: 'Login / Register',
      link: 'https://login.com',
      icon: 'user',
      authStatus: 'loggedOut'
    },
    {
      id: 'username',
      text: 'Welcome, {username}',
      link: 'https://login.com',
      icon: 'user',
      authStatus: 'loggedIn'
    },
    {
      id: 'My Porsche',
      text: 'My Porsche Id',
      link: 'https://login.com',
      authStatus: 'loggedIn'
    },
    {
      id: 'login_security',
      text: 'Login e Security',
      link: 'https://login.com',
      authStatus: 'loggedIn'
    },
    {
      id: 'personal_details',
      text: 'Personal Details',
      link: 'https://personal_details.com',
      authStatus: 'loggedIn'
    },
    {
      id: 'payment',
      text: 'Payment & Orders',
      link: 'https://payment.com',
      authStatus: 'loggedIn'
    },
    {
      id: 'my_orders',
      text: 'My Orders',
      link: 'https://my_orders.com',
      authStatus: 'loggedIn'
    },
    {
      id: 'help',
      text: 'Help',
      link: 'https://login.com'
    },
    {
      id: 'Country Selector',
      text: 'North America',
      link: 'https://login.com'
    }
  ],
  availableLocales: ['en-GB'],
  availableLocalesOnlyForShop: [],
  logoLinksToShopLocales: []
};
export const mockShopContentResponse = {
  menuItems: mockShopContent.children,
  additionalContent: mockShopContent.additionalContent
};
export const mockMoreContent = {
  id: 'mainmenu.more',
  text: 'More',
  link: '',
  children: [
    {
      id: 'mainmenu.services',
      text: 'Services',
      link: '',
      children: [
        {
          id: 'mainmenu.vehiclepurchase.eperformance',
          text: 'E-Mobility & E-Performance',
          link: 'https://www.porsche.com/uk/aboutporsche/e-performance/',
          children: []
        },
        {
          id: 'mainmenu.services.tequipment',
          text: 'Tequipment – Genuine Accessories',
          link: 'https://www.porsche.com/uk/accessoriesandservice/tequipment/',
          children: []
        },
        {
          id: 'mainmenu.services.exclusivemanufaktur',
          text: 'Exclusive Manufaktur – Individual Vehicles',
          link: 'https://www.porsche.com/uk/accessoriesandservice/exclusive-manufaktur/',
          children: []
        },
        {
          id: 'mainmenu.services.financialservices',
          text: 'Financial Services – Leasing & Financing',
          link: 'https://www.porsche.com/uk/accessoriesandservice/financialservices/',
          children: []
        },
        {
          id: 'mainmenu.services.approved',
          text: 'Approved – Pre-owned Cars & Warranty',
          link: 'https://www.porsche.com/uk/approvedused/porscheapproved/',
          children: []
        },
        {
          id: 'mainmenu.services.garageservice',
          text: 'Service & Maintenance',
          link: 'https://www.porsche.com/uk/accessoriesandservice/porscheservice/',
          children: []
        },
        {
          id: 'mainmenu.services.classic',
          text: 'Classic – Service & Parts',
          link: 'https://www.porsche.com/uk/accessoriesandservice/classic/',
          children: []
        },
        {
          id: 'mainmenu.services.buyingowning',
          text: 'Buying & Owning',
          link: 'https://www.porsche.com/uk/accessoriesandservice/owningyourporsche/',
          children: []
        },
        {
          id: 'mainmenu.services.autoinsurance',
          text: 'Vehicle Insurance',
          link: 'https://www.insurewithporsche.co.uk',
          children: []
        },
        {
          id: 'mainmenu.services.factorycollection',
          text: 'Factory Delivery',
          link: 'https://www.porsche.com/uk/accessoriesandservice/factorycollection/',
          children: []
        },
        {
          id: 'mainmenu.services.serviceforbusinesses',
          text: 'Contract hire for business',
          link: 'https://www.porsche.com/uk/accessoriesandservice/financialservices-contacthire/',
          children: []
        },
        {
          id: 'mainmenu.services.sellyourporsche',
          text: 'Sell your Porsche',
          link: 'https://www.porsche.com/uk/approvedused/sellyourporsche/',
          children: []
        },
        {
          id: 'mainmenu.services.recallsearch',
          text: 'Recall Search',
          link: 'https://recall.porsche.com/prod/pag/vinrecalllookup.nsf/VIN?ReadForm&c=gb ',
          children: []
        }
      ]
    },
    {
      id: 'mainmenu.experience',
      text: 'Experience',
      link: '',
      children: [
        {
          id: 'mainmenu.experience.silverstone',
          text: 'Experience Centre Silverstone',
          link: 'https://www.porsche.com/silverstone/en/',
          children: []
        },
        {
          id: 'mainmenu.experience.events',
          text: 'Events & Partnerships',
          link: 'https://www.porsche.com/uk/motorsportandevents/events-and-partnerships/',
          children: []
        },
        {
          id: 'mainmenu.experience.experience',
          text: 'Porsche Experience – Driving & Track Experiences',
          link: 'https://www.porsche.com/uk/motorsportandevents/experience/',
          children: []
        },
        {
          id: 'mainmenu.experience.carreracup',
          text: 'Carrera Cup GB',
          link: 'https://www.porsche.com/uk/motorsportandevents/porschecarreracupgreatbritain/',
          children: []
        },
        {
          id: 'mainmenu.experience.sprintchallenge',
          text: 'Cayman Islands Sprint Challenge GB',
          link: 'https://www.porsche.com/uk/motorsportandevents/porschesprintchallengegb/',
          children: []
        },
        {
          id: 'mainmenu.experience.motorsport',
          text: 'Motorsport',
          link: 'https://www.porsche.com/uk/motorsportandevents/motorsport/',
          children: []
        },
        {
          id: 'mainmenu.experience.porscheclubs',
          text: 'Porsche Clubs',
          link: 'https://www.porsche.com/uk/motorsportandevents/porscheclubs/',
          children: []
        },
        {
          id: 'mainmenu.experience.golfsport',
          text: 'Golf Sports',
          link: 'https://www.porsche.com/international/motorsportandevents/golf/',
          children: []
        },
        {
          id: 'mainmenu.experience.christophorus',
          text: 'Christophorus – The Porsche Magazine',
          link: 'https://www.porsche.com/uk/aboutporsche/christophorusmagazine/',
          children: []
        },
        {
          id: 'mainmenu.experience.visitmuseum',
          text: 'Visit the Porsche Museum',
          link: 'https://www.porsche.com/uk/aboutporsche/porschemuseum/',
          children: []
        }
      ]
    },
    {
      id: 'mainmenu.company',
      text: 'Company',
      link: '',
      children: []
    }
  ]
};
export const mockLocaleContent = {
  menu: {
    text: 'Menu',
    description: '',
    id: 'mainmenu.menu'
  },
  accessibilityStatement: null,
  crest: {
    id: 'mainmenu.crest',
    text: 'Crest',
    description: '',
    link: 'https://www.porsche.com/uk',
    icon: ''
  },
  back: {
    text: 'Back',
    description: '',
    id: 'mainmenu.back'
  },
  close: {
    text: 'Close',
    description: '',
    id: 'mainmenu.close'
  },
  countryRecommender: mockCountryRecommenderContent,
  contextualButtons: {},
  contextualLinks: {},
  models: mockModelsContent,
  dealer: mockDealerContent,
  myPorsche: mockMyPorscheContent,
  vehiclePurchase: {
    id: 'mainmenu.vehiclepurchase',
    text: 'Vehicle Purchase',
    link: '',
    children: [
      {
        id: 'mainmenu.vehiclepurchase.newusedinventory',
        text: 'New & Used Inventory',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.configure',
        text: 'Configure',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.compare',
        text: 'Compare',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.approved',
        text: 'Approved - Pre-owned Cars & Warranty',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.onlinepayment',
        text: 'Online Bill Payment',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.exclusive',
        text: 'Exclusive Manufactur - Individual Vehicles',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      }
    ]
  },
  shop: {
    id: 'mainmenu.shops',
    text: 'Shop',
    link: 'https://shop.porsche.com',
    cart: 'https://shop.porsche.com/gb/en-GB/cart',
    wishlist: {
      id: 'mainmenu.shops.wishlist',
      text: 'Wishlist',
      description: '',
      link: 'https://shop.porsche.com/gb/en-GB/wishlist',
      icon: ''
    },
    children: [
      {
        id: 'mainmenu.services.accessories',
        text: 'Accessories',
        link: 'https://www.porsche.com/silverstone/en/',
        children: []
      }
    ],
    additionalContent: [],
    availableLocales: ['de-DE', 'de-CH', 'it-CH', 'en-GB', 'en-RO'],
    availableLocalesOnlyForShop: ['en-GB', 'en-RO'],
    logoLinksToShopLocales: ['en-GB', 'en-RO']
  },
  externalShop: null,
  experience: {
    id: 'mainmenu.experience',
    text: 'Experience',
    link: '',
    children: [
      {
        id: 'mainmenu.experience.silverstone',
        text: 'Experience Centre Silverstone',
        link: 'https://www.porsche.com/silverstone/en/',
        children: []
      },
      {
        id: 'mainmenu.experience.events',
        text: 'Events & Partnerships',
        link: 'https://www.porsche.com/uk/motorsportandevents/events-and-partnerships/',
        children: []
      },
      {
        id: 'mainmenu.experience.experience',
        text: 'Porsche Experience – Driving & Track Experiences',
        link: 'https://www.porsche.com/uk/motorsportandevents/experience/',
        children: []
      },
      {
        id: 'mainmenu.experience.carreracup',
        text: 'Carrera Cup GB',
        link: 'https://www.porsche.com/uk/motorsportandevents/porschecarreracupgreatbritain/',
        children: []
      },
      {
        id: 'mainmenu.experience.sprintchallenge',
        text: 'Cayman Islands Sprint Challenge GB',
        link: 'https://www.porsche.com/uk/motorsportandevents/porschesprintchallengegb/',
        children: []
      },
      {
        id: 'mainmenu.experience.motorsport',
        text: 'Motorsport',
        link: 'https://www.porsche.com/uk/motorsportandevents/motorsport/',
        children: []
      },
      {
        id: 'mainmenu.experience.porscheclubs',
        text: 'Porsche Clubs',
        link: 'https://www.porsche.com/uk/motorsportandevents/porscheclubs/',
        children: []
      },
      {
        id: 'mainmenu.experience.golfsport',
        text: 'Golf Sports',
        link: 'https://www.porsche.com/international/motorsportandevents/golf/',
        children: []
      },
      {
        id: 'mainmenu.experience.christophorus',
        text: 'Christophorus – The Porsche Magazine',
        link: 'https://www.porsche.com/uk/aboutporsche/christophorusmagazine/',
        children: []
      },
      {
        id: 'mainmenu.experience.visitmuseum',
        text: 'Visit the Porsche Museum',
        link: 'https://www.porsche.com/uk/aboutporsche/porschemuseum/',
        children: []
      }
    ]
  },
  services: {
    id: 'mainmenu.services',
    text: 'Services',
    link: '',
    children: [
      {
        id: 'mainmenu.vehiclepurchase.eperformance',
        text: 'E-Mobility & E-Performance',
        link: 'https://www.porsche.com/uk/aboutporsche/e-performance/',
        children: []
      },
      {
        id: 'mainmenu.services.tequipment',
        text: 'Tequipment – Genuine Accessories',
        link: 'https://www.porsche.com/uk/accessoriesandservice/tequipment/',
        children: []
      },
      {
        id: 'mainmenu.services.exclusivemanufaktur',
        text: 'Exclusive Manufaktur – Individual Vehicles',
        link: 'https://www.porsche.com/uk/accessoriesandservice/exclusive-manufaktur/',
        children: []
      },
      {
        id: 'mainmenu.services.financialservices',
        text: 'Financial Services – Leasing & Financing',
        link: 'https://www.porsche.com/uk/accessoriesandservice/financialservices/',
        children: []
      },
      {
        id: 'mainmenu.services.approved',
        text: 'Approved – Pre-Owned Cars & Warranty',
        link: 'https://www.porsche.com/uk/approvedused/porscheapproved/',
        children: []
      },
      {
        id: 'mainmenu.services.garageservice',
        text: 'Service & Maintenance',
        link: 'https://www.porsche.com/uk/accessoriesandservice/porscheservice/',
        children: []
      },
      {
        id: 'mainmenu.services.classic',
        text: 'Classic – Service & Parts',
        link: 'https://www.porsche.com/uk/accessoriesandservice/classic/',
        children: []
      },
      {
        id: 'mainmenu.services.buyingowning',
        text: 'Buying & Owning',
        link: 'https://www.porsche.com/uk/accessoriesandservice/owningyourporsche/',
        children: []
      },
      {
        id: 'mainmenu.services.autoinsurance',
        text: 'Vehicle Insurance',
        link: 'https://www.insurewithporsche.co.uk',
        children: []
      },
      {
        id: 'mainmenu.services.serviceforbusinesses',
        text: 'Contract hire for business',
        link: 'https://www.porsche.com/uk/accessoriesandservice/financialservices-contacthire/',
        children: []
      },
      {
        id: 'mainmenu.services.sellyourporsche',
        text: 'Sell your Porsche',
        link: 'https://www.porsche.com/uk/approvedused/sellyourporsche/',
        children: []
      },
      {
        id: 'mainmenu.services.recallsearch',
        text: 'Recall Search',
        link: 'https://recall.porsche.com/prod/pag/vinrecalllookup_V3.nsf/VIN?ReadForm&c=greatbritain',
        children: []
      }
    ]
  },
  legacyMore: mockMoreContent,
  legacyShop: mockShopContent,
  error: mockErrorContent,
  loadingDescription: 'Loading...'
};
export const mockLocaleContentWithoutShop = {
  menu: {
    text: 'Menu',
    description: '',
    id: 'mainmenu.menu'
  },
  accessibilityStatement: null,
  crest: {
    id: 'mainmenu.crest',
    text: 'Crest',
    description: '',
    link: 'https://www.porsche.com/uk',
    icon: ''
  },
  back: {
    text: 'Back',
    description: '',
    id: 'mainmenu.back'
  },
  close: {
    text: 'Close',
    description: '',
    id: 'mainmenu.close'
  },
  countryRecommender: mockCountryRecommenderContent,
  contextualButtons: {},
  contextualLinks: {},
  models: mockModelsContent,
  dealer: mockDealerContent,
  myPorsche: mockMyPorscheContent,
  vehiclePurchase: {
    id: 'mainmenu.vehiclepurchase',
    text: 'Vehicle Purchase',
    link: '',
    children: [
      {
        id: 'mainmenu.vehiclepurchase.newusedinventory',
        text: 'New & Used Inventory',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.configure',
        text: 'Configure',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.compare',
        text: 'Compare',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.approved',
        text: 'Approved - Pre-owned Cars & Warranty',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.onlinepayment',
        text: 'Online Bill Payment',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.exclusive',
        text: 'Exclusive Manufactur - Individual Vehicles',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      }
    ]
  },
  shop: null,
  externalShop: null,
  experience: {
    id: 'mainmenu.experience',
    text: 'Experience',
    link: '',
    children: [
      {
        id: 'mainmenu.experience.silverstone',
        text: 'Experience Centre Silverstone',
        link: 'https://www.porsche.com/silverstone/en/',
        children: []
      },
      {
        id: 'mainmenu.experience.events',
        text: 'Events & Partnerships',
        link: 'https://www.porsche.com/uk/motorsportandevents/events-and-partnerships/',
        children: []
      },
      {
        id: 'mainmenu.experience.experience',
        text: 'Porsche Experience – Driving & Track Experiences',
        link: 'https://www.porsche.com/uk/motorsportandevents/experience/',
        children: []
      },
      {
        id: 'mainmenu.experience.carreracup',
        text: 'Carrera Cup GB',
        link: 'https://www.porsche.com/uk/motorsportandevents/porschecarreracupgreatbritain/',
        children: []
      },
      {
        id: 'mainmenu.experience.sprintchallenge',
        text: 'Cayman Islands Sprint Challenge GB',
        link: 'https://www.porsche.com/uk/motorsportandevents/porschesprintchallengegb/',
        children: []
      },
      {
        id: 'mainmenu.experience.motorsport',
        text: 'Motorsport',
        link: 'https://www.porsche.com/uk/motorsportandevents/motorsport/',
        children: []
      },
      {
        id: 'mainmenu.experience.porscheclubs',
        text: 'Porsche Clubs',
        link: 'https://www.porsche.com/uk/motorsportandevents/porscheclubs/',
        children: []
      },
      {
        id: 'mainmenu.experience.golfsport',
        text: 'Golf Sports',
        link: 'https://www.porsche.com/international/motorsportandevents/golf/',
        children: []
      },
      {
        id: 'mainmenu.experience.christophorus',
        text: 'Christophorus – The Porsche Magazine',
        link: 'https://www.porsche.com/uk/aboutporsche/christophorusmagazine/',
        children: []
      },
      {
        id: 'mainmenu.experience.visitmuseum',
        text: 'Visit the Porsche Museum',
        link: 'https://www.porsche.com/uk/aboutporsche/porschemuseum/',
        children: []
      }
    ]
  },
  services: {
    id: 'mainmenu.services',
    text: 'Services',
    link: '',
    children: [
      {
        id: 'mainmenu.vehiclepurchase.eperformance',
        text: 'E-Mobility & E-Performance',
        link: 'https://www.porsche.com/uk/aboutporsche/e-performance/',
        children: []
      },
      {
        id: 'mainmenu.services.tequipment',
        text: 'Tequipment – Genuine Accessories',
        link: 'https://www.porsche.com/uk/accessoriesandservice/tequipment/',
        children: []
      },
      {
        id: 'mainmenu.services.exclusivemanufaktur',
        text: 'Exclusive Manufaktur – Individual Vehicles',
        link: 'https://www.porsche.com/uk/accessoriesandservice/exclusive-manufaktur/',
        children: []
      },
      {
        id: 'mainmenu.services.financialservices',
        text: 'Financial Services – Leasing & Financing',
        link: 'https://www.porsche.com/uk/accessoriesandservice/financialservices/',
        children: []
      },
      {
        id: 'mainmenu.services.approved',
        text: 'Approved – Pre-Owned Cars & Warranty',
        link: 'https://www.porsche.com/uk/approvedused/porscheapproved/',
        children: []
      },
      {
        id: 'mainmenu.services.garageservice',
        text: 'Service & Maintenance',
        link: 'https://www.porsche.com/uk/accessoriesandservice/porscheservice/',
        children: []
      },
      {
        id: 'mainmenu.services.classic',
        text: 'Classic – Service & Parts',
        link: 'https://www.porsche.com/uk/accessoriesandservice/classic/',
        children: []
      },
      {
        id: 'mainmenu.services.buyingowning',
        text: 'Buying & Owning',
        link: 'https://www.porsche.com/uk/accessoriesandservice/owningyourporsche/',
        children: []
      },
      {
        id: 'mainmenu.services.autoinsurance',
        text: 'Vehicle Insurance',
        link: 'https://www.insurewithporsche.co.uk',
        children: []
      },
      {
        id: 'mainmenu.services.serviceforbusinesses',
        text: 'Contract hire for business',
        link: 'https://www.porsche.com/uk/accessoriesandservice/financialservices-contacthire/',
        children: []
      },
      {
        id: 'mainmenu.services.sellyourporsche',
        text: 'Sell your Porsche',
        link: 'https://www.porsche.com/uk/approvedused/sellyourporsche/',
        children: []
      },
      {
        id: 'mainmenu.services.recallsearch',
        text: 'Recall Search',
        link: 'https://recall.porsche.com/prod/pag/vinrecalllookup_V3.nsf/VIN?ReadForm&c=greatbritain',
        children: []
      }
    ]
  },
  legacyMore: mockMoreContent,
  legacyShop: mockShopContent,
  error: mockErrorContent,
  loadingDescription: 'Loading...'
};
export const mockContentWithoutShopMyporsche = {
  menu: {
    text: 'Menu',
    description: '',
    id: 'mainmenu.menu'
  },
  accessibilityStatement: null,
  crest: {
    id: 'mainmenu.crest',
    text: 'Crest',
    description: '',
    link: 'https://www.porsche.com/uk',
    icon: ''
  },
  back: {
    text: 'Back',
    description: '',
    id: 'mainmenu.back'
  },
  close: {
    text: 'Close',
    description: '',
    id: 'mainmenu.close'
  },
  countryRecommender: mockCountryRecommenderContent,
  contextualButtons: {},
  contextualLinks: {},
  models: mockModelsContent,
  dealer: mockDealerContent,
  myPorsche: null,
  vehiclePurchase: {
    id: 'mainmenu.vehiclepurchase',
    text: 'Vehicle Purchase',
    link: '',
    children: [
      {
        id: 'mainmenu.vehiclepurchase.newusedinventory',
        text: 'New & Used Inventory',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.configure',
        text: 'Configure',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.compare',
        text: 'Compare',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.approved',
        text: 'Approved - Pre-owned Cars & Warranty',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.onlinepayment',
        text: 'Online Bill Payment',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      },
      {
        id: 'mainmenu.vehiclepurchase.exclusive',
        text: 'Exclusive Manufactur - Individual Vehicles',
        link: 'http://www.porsche.com/gb/en-GB/test-shop',
        children: []
      }
    ]
  },
  shop: null,
  externalShop: null,
  experience: {
    id: 'mainmenu.experience',
    text: 'Experience',
    link: '',
    children: [
      {
        id: 'mainmenu.experience.silverstone',
        text: 'Experience Centre Silverstone',
        link: 'https://www.porsche.com/silverstone/en/',
        children: []
      },
      {
        id: 'mainmenu.experience.events',
        text: 'Events & Partnerships',
        link: 'https://www.porsche.com/uk/motorsportandevents/events-and-partnerships/',
        children: []
      },
      {
        id: 'mainmenu.experience.experience',
        text: 'Porsche Experience – Driving & Track Experiences',
        link: 'https://www.porsche.com/uk/motorsportandevents/experience/',
        children: []
      },
      {
        id: 'mainmenu.experience.carreracup',
        text: 'Carrera Cup GB',
        link: 'https://www.porsche.com/uk/motorsportandevents/porschecarreracupgreatbritain/',
        children: []
      },
      {
        id: 'mainmenu.experience.sprintchallenge',
        text: 'Cayman Islands Sprint Challenge GB',
        link: 'https://www.porsche.com/uk/motorsportandevents/porschesprintchallengegb/',
        children: []
      },
      {
        id: 'mainmenu.experience.motorsport',
        text: 'Motorsport',
        link: 'https://www.porsche.com/uk/motorsportandevents/motorsport/',
        children: []
      },
      {
        id: 'mainmenu.experience.porscheclubs',
        text: 'Porsche Clubs',
        link: 'https://www.porsche.com/uk/motorsportandevents/porscheclubs/',
        children: []
      },
      {
        id: 'mainmenu.experience.golfsport',
        text: 'Golf Sports',
        link: 'https://www.porsche.com/international/motorsportandevents/golf/',
        children: []
      },
      {
        id: 'mainmenu.experience.visitmuseum',
        text: 'Visit the Porsche Museum',
        link: 'https://www.porsche.com/uk/aboutporsche/porschemuseum/',
        children: []
      }
    ]
  },
  services: {
    id: 'mainmenu.services',
    text: 'Services',
    link: '',
    children: [
      {
        id: 'mainmenu.vehiclepurchase.eperformance',
        text: 'E-Mobility & E-Performance',
        link: 'https://www.porsche.com/uk/aboutporsche/e-performance/',
        children: []
      },
      {
        id: 'mainmenu.services.tequipment',
        text: 'Tequipment – Genuine Accessories',
        link: 'https://www.porsche.com/uk/accessoriesandservice/tequipment/',
        children: []
      },
      {
        id: 'mainmenu.services.exclusivemanufaktur',
        text: 'Exclusive Manufaktur – Individual Vehicles',
        link: 'https://www.porsche.com/uk/accessoriesandservice/exclusive-manufaktur/',
        children: []
      },
      {
        id: 'mainmenu.services.financialservices',
        text: 'Financial Services – Leasing & Financing',
        link: 'https://www.porsche.com/uk/accessoriesandservice/financialservices/',
        children: []
      },
      {
        id: 'mainmenu.services.approved',
        text: 'Approved – Pre-Owned Cars & Warranty',
        link: 'https://www.porsche.com/uk/approvedused/porscheapproved/',
        children: []
      },
      {
        id: 'mainmenu.services.garageservice',
        text: 'Service & Maintenance',
        link: 'https://www.porsche.com/uk/accessoriesandservice/porscheservice/',
        children: []
      },
      {
        id: 'mainmenu.services.classic',
        text: 'Classic – Service & Parts',
        link: 'https://www.porsche.com/uk/accessoriesandservice/classic/',
        children: []
      },
      {
        id: 'mainmenu.services.buyingowning',
        text: 'Buying & Owning',
        link: 'https://www.porsche.com/uk/accessoriesandservice/owningyourporsche/',
        children: []
      },
      {
        id: 'mainmenu.services.autoinsurance',
        text: 'Vehicle Insurance',
        link: 'https://www.insurewithporsche.co.uk',
        children: []
      },
      {
        id: 'mainmenu.services.serviceforbusinesses',
        text: 'Contract hire for business',
        link: 'https://www.porsche.com/uk/accessoriesandservice/financialservices-contacthire/',
        children: []
      },
      {
        id: 'mainmenu.services.sellyourporsche',
        text: 'Sell your Porsche',
        link: 'https://www.porsche.com/uk/approvedused/sellyourporsche/',
        children: []
      },
      {
        id: 'mainmenu.services.recallsearch',
        text: 'Recall Search',
        link: 'https://recall.porsche.com/prod/pag/vinrecalllookup_V3.nsf/VIN?ReadForm&c=greatbritain',
        children: []
      }
    ]
  },
  legacyMore: mockMoreContent,
  legacyShop: mockShopContent,
  error: mockErrorContent,
  loadingDescription: 'Loading...'
};
export const mockNewContent = {
  'de-CH': mockLocaleContent,
  'sv-SE': Object.assign(Object.assign({}, mockLocaleContent), { shop: Object.assign(Object.assign({}, mockLocaleContent.shop), { availableLocales: ['sv-SE'], availableLocalesOnlyForShop: ['sv-SE'] }) }),
  'it-CH': mockLocaleContent,
  'en-US': Object.assign(Object.assign({}, mockLocaleContent), { accessibilityStatement: {
      id: 'mainmenu.accessibilitystatement',
      text: 'Accessibility Statement',
      description: '',
      link: 'https://www.porsche.com/usa/accessibilitystatement/',
      icon: ''
    } }),
  'en-GB': Object.assign(Object.assign({}, mockLocaleContent), { myPorsche: null })
};
export const mockLevel1Items = [
  {
    elementId: RoutingKeys.MODELS,
    routingKeys: [RoutingKeys.MODELS],
    analyticsId: 'mainmenu.models',
    text: 'Models'
  },
  {
    elementId: RoutingKeys.VEHICLE_PURCHASE,
    routingKeys: [RoutingKeys.VEHICLE_PURCHASE],
    analyticsId: 'mainmenu.vehiclepurchase',
    text: 'Vehicle Purchase'
  },
  {
    elementId: RoutingKeys.SHOP,
    routingKeys: [RoutingKeys.SHOP],
    analyticsId: 'mainmenu.shops',
    text: 'Shop',
    link: 'https://shop.porsche.com',
    datalayerEventAction: EventActions.NAVIGATION_LINK_CLICK,
    iconRight: 'none'
  },
  {
    elementId: RoutingKeys.SERVICES,
    routingKeys: [RoutingKeys.SERVICES],
    analyticsId: 'mainmenu.services',
    text: 'Services'
  },
  {
    elementId: RoutingKeys.EXPERIENCE,
    routingKeys: [RoutingKeys.EXPERIENCE],
    analyticsId: 'mainmenu.experience',
    text: 'Experience'
  },
  {
    elementId: RoutingKeys.DEALER_SEARCH,
    routingKeys: [RoutingKeys.DEALER_SEARCH],
    analyticsId: 'mainmenu.finddealer',
    text: 'Find a Dealer',
    iconRight: 'none',
    link: 'http://www.porsche.com/outbound-link'
  },
  {
    elementId: RoutingKeys.MY_PORSCHE,
    routingKeys: [RoutingKeys.MY_PORSCHE],
    analyticsId: 'mainmenu.myporsche',
    text: 'My Porsche',
    showIconModifier: false,
    subtext: 'Mirinda Doe.'
  }
];
export const mockLevel1ItemsWithMissingItems = [
  {
    elementId: RoutingKeys.MODELS,
    routingKeys: [RoutingKeys.MODELS],
    analyticsId: 'mainmenu.models',
    text: 'Models'
  },
  {
    elementId: RoutingKeys.VEHICLE_PURCHASE,
    routingKeys: [RoutingKeys.VEHICLE_PURCHASE],
    analyticsId: 'mainmenu.vehiclepurchase',
    text: 'Vehicle Purchase'
  },
  {
    elementId: RoutingKeys.SERVICES,
    routingKeys: [RoutingKeys.SERVICES],
    analyticsId: 'mainmenu.services',
    text: 'Services'
  },
  {
    elementId: RoutingKeys.EXPERIENCE,
    routingKeys: [RoutingKeys.EXPERIENCE],
    analyticsId: 'mainmenu.experience',
    text: 'Experience'
  },
  {
    elementId: RoutingKeys.DEALER_SEARCH,
    routingKeys: [RoutingKeys.DEALER_SEARCH],
    analyticsId: 'mainmenu.finddealer',
    link: 'http://www.porsche.com/all-dealers-map',
    text: 'Find a Dealer',
    iconRight: 'arrow-head-right',
    onClick: undefined
  }
];
export const mockShopItems = {
  id: 'shop item',
  text: 'first child',
  link: '',
  children: [
    {
      id: 'item-1',
      text: 'item 1',
      link: 'link1',
      children: []
    },
    {
      id: 'item-2',
      text: 'item 2',
      link: 'link2',
      children: []
    },
    {
      id: 'item-3',
      text: 'item 3',
      link: 'link3',
      children: []
    }
  ]
};
export const mockExperienceItem = {
  id: 'mainmenu.experience',
  text: 'Experience',
  link: '',
  children: [
    {
      id: 'mainmenu.experience',
      text: 'Erleben',
      link: '',
      children: [
        {
          id: 'mainmenu.experience.golfsport',
          text: 'Golf',
          link: '',
          children: [
            {
              id: 'mainmenu.services.financialservices',
              text: 'Golf',
              children: [],
              link: 'https://preview.ppn.porsche.com/swiss/de/accessoriesandservices/porschefinancialservices/'
            }
          ]
        },
        {
          id: 'mainmenu.experience.motorsport',
          text: 'Motorsport',
          children: [],
          link: 'https://preview.ppn.porsche.com/swiss/de/motorsportandevents/motorsport/'
        }
      ]
    }
  ]
};
export const mockVehiclePurchaseItem = {
  id: 'vehicle purchase item',
  text: 'first child',
  link: '',
  children: [
    {
      id: 'item-1',
      text: 'item 1',
      link: 'link1',
      children: []
    },
    {
      id: 'item-2',
      text: 'item 2',
      link: 'link2',
      children: []
    }
  ]
};
export const mockDealerSearchRegionResponse = {
  regions: [
    {
      regionName: 'Anhui',
      regionNameLocalized: '安徽',
      cities: [
        {
          cityName: 'Hefei',
          cityNameLocalized: '合肥',
          dealers: {
            '0f7cf750-4eb5-11e0-afb2-ca213412ec80': {
              ppnDealer: {
                id: '0f7cf750-4eb5-11e0-afb2-ca213412ec80',
                name: 'Porsche Centre Hefei Baohe',
                nameLocalized: '合肥包河保时捷中心',
                porschePartnerNo: '7000350',
                ppnOrgId: '5678',
                address: {
                  street: 'No.49 Changchun Street,Baohe District Hefei,Anhui Province',
                  postalCode: '230051',
                  city: 'Hefei',
                  state: 'Anhui',
                  countryCode: 'CN'
                },
                addressLocalized: {
                  street: '安徽合肥市包河区长春街49号',
                  state: '安徽',
                  city: '合肥'
                },
                contactDetails: {
                  phoneNumber: '+86 551 6366 3911',
                  emailAddress: 'bain.wang@porsche-hefei.com',
                  homepage: 'http://www.porsche-hefei.com',
                  contactOpeningHours: [
                    {
                      day: 'MONDAY',
                      open: '09:00',
                      close: '19:00'
                    }
                  ]
                },
                market: 'Greater China',
                coordinates: {
                  latitude: 31.79735,
                  longitude: 117.35171
                }
              }
            }
          }
        },
        {
          cityName: 'Wuhu',
          cityNameLocalized: null,
          dealers: {
            '736f1d00-aa31-11e6-9720-ba9b3b423ca0': {
              ppnDealer: {
                id: '736f1d00-aa31-11e6-9720-ba9b3b423ca0',
                name: 'Porsche Centre Wuhu',
                nameLocalized: '芜湖保时捷中心',
                porschePartnerNo: '7000840',
                ppnOrgId: '17525',
                address: {
                  street: 'Auto zone expansion area, North Yijiang Road, Jiujiang District, Wuhu City, Anhui Province',
                  postalCode: '241007',
                  city: 'Wuhu',
                  state: 'Anhui',
                  countryCode: 'CN'
                },
                addressLocalized: {
                  street: '安徽省芜湖市鸠江区弋江北路汽车园扩大区',
                  state: '安徽',
                  city: '芜湖'
                },
                contactDetails: {
                  phoneNumber: '+86 152 1221 0092',
                  emailAddress: 'lili.chen@porsche-wuhu.com',
                  homepage: 'http://www.porsche-wuhu.com/',
                  contactOpeningHours: [
                    {
                      day: 'TUESDAY',
                      open: '09:00',
                      close: '19:00'
                    }
                  ]
                },
                market: 'Greater China',
                coordinates: {
                  latitude: 31.36842,
                  longitude: 118.421211
                }
              }
            }
          }
        }
      ]
    },
    {
      regionName: 'Beijing',
      regionNameLocalized: null,
      cities: [
        {
          cityName: 'Beijing',
          cityNameLocalized: '北京',
          dealers: {
            'd10832c0-4583-11dc-ab2a-b4dd756d08a1': {
              ppnDealer: {
                id: 'd10832c0-4583-11dc-ab2a-b4dd756d08a1',
                name: 'Porsche Centre Beijing Yizhuang',
                nameLocalized: '北京亦庄保时捷中心',
                porschePartnerNo: '7000050',
                ppnOrgId: '12609',
                address: {
                  street: 'No. A1, East Ring North Road, BDA',
                  postalCode: '100176',
                  city: 'Beijing',
                  state: 'Beijing',
                  countryCode: 'CN'
                },
                addressLocalized: {
                  street: '北京经济技术开发区东环北路甲1号',
                  state: '北京',
                  city: '北京'
                },
                contactDetails: {
                  phoneNumber: '+86 10 5965 9911',
                  emailAddress: 'chao.luo@porsche-beijing-yizhuang.com',
                  homepage: 'www.porsche-beijing-yizhuang.com',
                  contactOpeningHours: [
                    {
                      day: 'WEDNESDAY',
                      open: '09:00',
                      close: '19:00'
                    }
                  ]
                },
                market: 'Greater China',
                coordinates: {
                  latitude: 39.80778,
                  longitude: 116.51679
                }
              }
            },
            'cfac5b00-4040-11e2-8516-dee31fa7d00e': {
              ppnDealer: {
                id: 'cfac5b00-4040-11e2-8516-dee31fa7d00e',
                name: 'Porsche Centre Beijing Shijingshan',
                nameLocalized: '北京石景山保时捷中心',
                porschePartnerNo: '7000380',
                ppnOrgId: '3302',
                address: {
                  street: 'No.36 Shidai Huayuan South Road, Shijingshan District',
                  postalCode: '100043',
                  city: 'Beijing',
                  state: 'Beijing',
                  countryCode: 'CN'
                },
                addressLocalized: {
                  street: '北京市石景山区时代花园南路36号',
                  state: '北京',
                  city: '北京'
                },
                contactDetails: {
                  phoneNumber: '+86 10 8898 1911',
                  emailAddress: 'francis.xiao@porsche-beijing-shijingshan.com',
                  homepage: 'http://www.porsche-beijing-shijingshan.com',
                  contactOpeningHours: [
                    {
                      day: 'FRIDAY',
                      open: '09:00',
                      close: '19:00'
                    }
                  ]
                },
                market: 'Greater China',
                coordinates: {
                  latitude: 39.91835,
                  longitude: 116.19941
                }
              }
            }
          }
        }
      ]
    }
  ]
};
export const mockUserInformation = {
  savedSearches: 3,
  savedVehicles: 1,
  unreadMessages: 2,
  userData: {
    firstName: 'Mirinda',
    lastName: 'Doe',
    porscheId: 'm.doe@porsche.ag',
    userId: 'f0hasos1or8p7nbi'
  }
};
export const mockExternalShop = {
  id: 'mainmenu.shops',
  text: 'Shop',
  description: '',
  link: 'https://www.porsche-external-shop.com/uk',
  icon: ''
};
export const mockDealerDetailsOpeningHours = {
  monday: {
    day: 'MONDAY',
    open: '09:00',
    close: '19:00'
  },
  tuesday: {
    day: 'TUESDAY',
    open: '09:00',
    close: '19:00'
  },
  wednesday: {
    day: 'WEDNESDAY',
    open: '09:00',
    close: '19:00'
  },
  thursday: {
    day: 'THURSDAY',
    open: '09:00',
    close: '19:00'
  },
  friday: {
    day: 'FRIDAY',
    open: '09:00',
    close: '19:00'
  },
  daily_lunch: {
    day: 'DAILY_LUNCH',
    open: '09:00',
    close: '19:00'
  }
};
