// @porschehn/navigation/dist/collection/components/heading/heading.js
import { h, Host } from "@stencil/core";
import { isString } from "../../utility/helper";
export class Heading {
  constructor() {
    this.text = undefined;
    this.subtext = undefined;
  }
  render() {
    return (h(Host, null, h("phn-p-heading", { size: "inherit" }, this.text), isString(this.subtext) ? (h("div", { class: "text-limit" }, h("phn-p-text", { color: "contrast-high", size: "xx-small", tag: "div", ellipsis: "true" }, this.subtext))) : null));
  }
  static get is() { return "phn-heading"; }
  static get encapsulation() { return "scoped"; }
  static get originalStyleUrls() {
    return {
      "$": ["heading.scss"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["heading.css"]
    };
  }
  static get properties() {
    return {
      "text": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": true,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "text",
        "reflect": false
      },
      "subtext": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string | undefined",
          "references": {}
        },
        "required": false,
        "optional": true,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "subtext",
        "reflect": false
      }
    };
  }
}
