// @porsche-design-system/components-js/ag-grid/cjs/styles.cjs
'use strict';

/* Auto Generated Start */
const borderRadiusMedium = "8px";
const borderRadiusSmall = "4px";
const fontFamily = "'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif";
const fontSizeTextSmall = "1rem";
const fontSizeTextXSmall = "clamp(0.81rem, 0.23vw + 0.77rem, 0.88rem)";
const fontWeightSemiBold = 600;
const themeDarkBackgroundBase = "#0E0E12";
const themeDarkBackgroundFrosted = "hsl(240 3% 26%/35%)";
const themeDarkBackgroundSurface = "#212225";
const themeDarkContrastLow = "#404044";
const themeDarkContrastMedium = "#88898C";
const themeDarkNotificationError = "#FC4040";
const themeDarkNotificationInfo = "#178BFF";
const themeDarkPrimary = "#FBFCFF";
const themeDarkStateDisabled = "#7E7F82";
const themeDarkStateFocus = "#1A44EA";
const themeDarkStateHover = "rgba(148, 149, 152, .18)";
const themeLightBackgroundBase = "#FFF";
const themeLightBackgroundFrosted = "hsl(240 4% 85%/35%)";
const themeLightBackgroundSurface = "#EEEFF2";
const themeLightContrastLow = "#D8D8DB";
const themeLightContrastMedium = "#6B6D70";
const themeLightNotificationError = "#CC1922";
const themeLightNotificationInfo = "#2762EC";
const themeLightPrimary = "#010205";
const themeLightStateDisabled = "#949598";
const themeLightStateFocus = "#1A44EA";
const themeLightStateHover = "rgba(148, 149, 152, .18)";
const themeDarkContrastHigh = "#AFB0B3";
const themeLightContrastHigh = "#535457";
const borderWidthBase = "2px";
const spacingStaticXSmall = "4px";
const themeDarkNotificationSuccess = "#09D087";
const themeLightNotificationSuccess = "#197E10";
const themeLightBackgroundSurfaceDarken = "#CBCED7";
const themeDarkBackgroundSurfaceDarken = "#040405";
const themeLightSuccessColorDarken = "#0E4809";
const themeDarkSuccessColorDarken = "#069561";
/* Auto Generated End */

exports.borderRadiusMedium = borderRadiusMedium;
exports.borderRadiusSmall = borderRadiusSmall;
exports.borderWidthBase = borderWidthBase;
exports.fontFamily = fontFamily;
exports.fontSizeTextSmall = fontSizeTextSmall;
exports.fontSizeTextXSmall = fontSizeTextXSmall;
exports.fontWeightSemiBold = fontWeightSemiBold;
exports.spacingStaticXSmall = spacingStaticXSmall;
exports.themeDarkBackgroundBase = themeDarkBackgroundBase;
exports.themeDarkBackgroundFrosted = themeDarkBackgroundFrosted;
exports.themeDarkBackgroundSurface = themeDarkBackgroundSurface;
exports.themeDarkBackgroundSurfaceDarken = themeDarkBackgroundSurfaceDarken;
exports.themeDarkContrastHigh = themeDarkContrastHigh;
exports.themeDarkContrastLow = themeDarkContrastLow;
exports.themeDarkContrastMedium = themeDarkContrastMedium;
exports.themeDarkNotificationError = themeDarkNotificationError;
exports.themeDarkNotificationInfo = themeDarkNotificationInfo;
exports.themeDarkNotificationSuccess = themeDarkNotificationSuccess;
exports.themeDarkPrimary = themeDarkPrimary;
exports.themeDarkStateDisabled = themeDarkStateDisabled;
exports.themeDarkStateFocus = themeDarkStateFocus;
exports.themeDarkStateHover = themeDarkStateHover;
exports.themeDarkSuccessColorDarken = themeDarkSuccessColorDarken;
exports.themeLightBackgroundBase = themeLightBackgroundBase;
exports.themeLightBackgroundFrosted = themeLightBackgroundFrosted;
exports.themeLightBackgroundSurface = themeLightBackgroundSurface;
exports.themeLightBackgroundSurfaceDarken = themeLightBackgroundSurfaceDarken;
exports.themeLightContrastHigh = themeLightContrastHigh;
exports.themeLightContrastLow = themeLightContrastLow;
exports.themeLightContrastMedium = themeLightContrastMedium;
exports.themeLightNotificationError = themeLightNotificationError;
exports.themeLightNotificationInfo = themeLightNotificationInfo;
exports.themeLightNotificationSuccess = themeLightNotificationSuccess;
exports.themeLightPrimary = themeLightPrimary;
exports.themeLightStateDisabled = themeLightStateDisabled;
exports.themeLightStateFocus = themeLightStateFocus;
exports.themeLightStateHover = themeLightStateHover;
exports.themeLightSuccessColorDarken = themeLightSuccessColorDarken;
