// @porschehn/navigation/dist/collection/components/dealer-search-results/dealer-search-map-link.js
import { navContentStore } from "../../state/nav-content-store";
import { h } from "@stencil/core";
import { dealerSearchResultsStore } from "../../state/dealer-search-results-store";
import globalDataLayerService from "../../services/analytics/global-data-layer-service";
import { ClickElementTypes, DEALER_SEARCH_DISPLAY_ELEMENT_ID, ElementTypes, EventActions, GEOLOCATION_BUTTON_SEARCH_TERM } from "../../utility/constants";
import analyticsManager from "../../services/analytics/analytics-manager";
import { RoutingKeys } from "../../entities/routing-key";
import { isString } from "../../utility/helper";
export function DealerSearchMapLink() {
  const { dealer: dealerContent } = navContentStore.state;
  function emitDealersMapEvent() {
    const map = dealerContent.search.map;
    const { results, searchTerm } = dealerSearchResultsStore.state;
    globalDataLayerService.pushDataLayerEvent(EventActions.DEALER_MAP_CLICK, {
      componentClick: {
        clickElementId: map.id,
        clickElementName: map.text,
        clickElementType: ClickElementTypes.Link,
        targetUrl: map.link
      },
      componentDisplay: {
        displayElementId: DEALER_SEARCH_DISPLAY_ELEMENT_ID,
        displayElementType: analyticsManager.calculateDisplayElementType(RoutingKeys.DEALER_SEARCH, ElementTypes.dealerSearchForm),
        displayElementName: dealerContent.text
      },
      componentSearch: {
        searchTerm: isString(searchTerm) ? searchTerm : GEOLOCATION_BUTTON_SEARCH_TERM,
        numberOfResults: results === null || results === void 0 ? void 0 : results.length
      }
    });
  }
  return (h("a", { class: 'animated', href: dealerContent.search.map.link, "aria-label": dealerContent.search.map.description }, h("phn-p-button", { onClick: emitDealersMapEvent, variant: "tertiary" }, dealerContent.search.map.text, ' ')));
}
