// @porschehn/navigation/dist/collection/components/dealer-search-results/dealer-search-results.js
import { Fragment, h, Host } from "@stencil/core";
import { dealerSearchResultsStore } from "../../state/dealer-search-results-store";
import { navContentStore } from "../../state/nav-content-store";
import { DealerSearchFormItemCard } from "./dealer-search-form-item-card";
import ErrorMessage from "../dealer-search-form/error-message";
import { DealerSearchMapLink } from "./dealer-search-map-link";
export class DealerSearchResults {
  constructor() {
    this.showOverlay = false;
    this.locale = undefined;
    this.breakpoint = undefined;
  }
  render() {
    const { results, ppnDealers } = dealerSearchResultsStore.state;
    const { error: errorContent } = navContentStore.state;
    if (results === null)
      return null;
    if ((results === null || results === void 0 ? void 0 : results.length) === 0)
      return (h(Fragment, null, h(ErrorMessage, { errorMessage: errorContent.noResults, onDismiss: () => null })));
    const dealerId = results[0];
    if (results.length === 1 && ppnDealers[dealerId]) {
      dealerSearchResultsStore.state.savedDealerCookie = {
        id: dealerId,
        latitude: ppnDealers[dealerId].ppnDealer.coordinates.latitude,
        longitude: ppnDealers[dealerId].ppnDealer.coordinates.longitude
      };
      return h("phn-dealer-details-drawer", { breakpoint: this.breakpoint, dealerId: dealerId, locale: this.locale });
    }
    return (h(Host, null, h(Fragment, null, results.map((dealerId) => {
      const dealer = ppnDealers[dealerId];
      if (dealer === undefined || dealer.ppnDealer === undefined) {
        return null;
      }
      return h(DealerSearchFormItemCard, { dealer: dealer, showOverlay: this.showOverlay, locale: this.locale });
    }), h(DealerSearchMapLink, null))));
  }
  static get is() { return "phn-dealer-search-results"; }
  static get encapsulation() { return "scoped"; }
  static get originalStyleUrls() {
    return {
      "$": ["dealer-search-results.scss"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["dealer-search-results.css"]
    };
  }
  static get properties() {
    return {
      "showOverlay": {
        "type": "boolean",
        "mutable": false,
        "complexType": {
          "original": "boolean",
          "resolved": "boolean",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "show-overlay",
        "reflect": false,
        "defaultValue": "false"
      },
      "locale": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": true,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "locale",
        "reflect": false
      },
      "breakpoint": {
        "type": "number",
        "mutable": false,
        "complexType": {
          "original": "number",
          "resolved": "number",
          "references": {}
        },
        "required": true,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "breakpoint",
        "reflect": false
      }
    };
  }
}
