// @porschehn/navigation/dist/components/data-layer-event-sender.js
import { E as EventActions, g as globalDataLayerService, C as ClickElementTypes, a as ElementTypes } from './global-data-layer-service.js';
import { a as analyticsManager } from './analytics-manager.js';
import { r as removeLastRoutingSegment } from './routing-key.js';

function sendDataLayerEvent({ datalayerEventAction = EventActions.NAVIGATION_CLICK, analyticsId, text, link, elementId, clickElementType, customDatalayerProperties = {} }) {
  globalDataLayerService.pushDataLayerEvent(datalayerEventAction, Object.assign(Object.assign({ componentClick: Object.assign({ clickElementId: analyticsId, clickElementName: text, targetElement: analyticsManager.getModelTargetElement(elementId, text), clickElementType }, (clickElementType === ClickElementTypes.Link ? { targetUrl: link } : {})), componentDisplay: {
      displayElementId: analyticsManager.calculateDisplayElementId(elementId),
      displayElementType: analyticsManager.calculateDisplayElementType(removeLastRoutingSegment(elementId), ElementTypes.menuItem),
      displayElementName: text
    } }, customDatalayerProperties), (datalayerEventAction === EventActions.MODEL_RANGE_CLICK
    ? {
      car: {
        modelRangeName: text.toLowerCase()
      }
    }
    : {})));
}

export { sendDataLayerEvent as s };
