// @porschehn/navigation/dist/components/dealer-search-form.js
import { proxyCustomElement, HTMLElement, h, Host } from '@stencil/core/internal/client';
import { a as showDealerSearchDropdown } from './dealer-search-helper.js';
import { s as splitLocale } from './global-data-layer-service.js';
import { n as navContentStore } from './nav-content-store.js';
import { d as defineCustomElement$7 } from './autocomplete-search-field.js';
import { d as defineCustomElement$6 } from './dealer-details-drawer.js';
import { d as defineCustomElement$5 } from './dealer-search-by-number-of-dealers.js';
import { d as defineCustomElement$4 } from './dealer-search-form-dropdown.js';
import { d as defineCustomElement$3 } from './dealer-search-form-geolocation.js';
import { d as defineCustomElement$2 } from './dealer-search-results.js';
import { d as defineCustomElement$1 } from './menu-item-card.js';

const dealerSearchFormCss = "@-webkit-keyframes transition-forward-in-animation{from{left:150%}to{left:0%}}@keyframes transition-forward-in-animation{from{left:150%}to{left:0%}}@-webkit-keyframes transition-backward-in-animation{from{left:-150%}to{left:0%}}@keyframes transition-backward-in-animation{from{left:-150%}to{left:0%}}@-webkit-keyframes single-drawer-transition-forward-in-animation{from{left:100%}to{left:0%}}@keyframes single-drawer-transition-forward-in-animation{from{left:100%}to{left:0%}}@-webkit-keyframes fade-in-animation{from{opacity:0;background-color:rgba(0, 0, 0, 0)}to{opacity:1;background-color:rgba(0, 0, 0, 0.6)}}@keyframes fade-in-animation{from{opacity:0;background-color:rgba(0, 0, 0, 0)}to{opacity:1;background-color:rgba(0, 0, 0, 0.6)}}@-webkit-keyframes slide-in-from-left-absolute{from{left:-1000px}to{left:0px}}@keyframes slide-in-from-left-absolute{from{left:-1000px}to{left:0px}}@-webkit-keyframes slideDown{from{max-height:0}to{max-height:1000px}}@keyframes slideDown{from{max-height:0}to{max-height:1000px}}@-webkit-keyframes slideUp{from{max-height:1000px}to{max-height:0}}@keyframes slideUp{from{max-height:1000px}to{max-height:0}}@-webkit-keyframes slideLeft{from{max-width:0}to{max-width:1000px}}@keyframes slideLeft{from{max-width:0}to{max-width:1000px}}@-webkit-keyframes slideRight{from{max-width:1000px}to{max-width:0}}@keyframes slideRight{from{max-width:1000px}to{max-width:0}}@-webkit-keyframes darken{from{background-color:rgba(0, 0, 0, 0)}to{background-color:rgba(0, 0, 0, 0.8)}}@keyframes darken{from{background-color:rgba(0, 0, 0, 0)}to{background-color:rgba(0, 0, 0, 0.8)}}@-webkit-keyframes lighten{from{background-color:rgba(0, 0, 0, 0.8)}to{background-color:rgba(0, 0, 0, 0)}}@keyframes lighten{from{background-color:rgba(0, 0, 0, 0.8)}to{background-color:rgba(0, 0, 0, 0)}}@-webkit-keyframes fadeInFromBottom{0%{opacity:0;top:36px}100%{opacity:1;top:0px}}@keyframes fadeInFromBottom{0%{opacity:0;top:36px}100%{opacity:1;top:0px}}@-webkit-keyframes fadeOutToTop{0%{opacity:1;bottom:0px}100%{opacity:0;bottom:16px}}@keyframes fadeOutToTop{0%{opacity:1;bottom:0px}100%{opacity:0;bottom:16px}}@-webkit-keyframes slideLeftDoubleDrawer{from{right:-1000px}to{right:0px}}@keyframes slideLeftDoubleDrawer{from{right:-1000px}to{right:0px}}@-webkit-keyframes slideRightDoubleDrawer{from{right:0px}to{right:-1000px}}@keyframes slideRightDoubleDrawer{from{right:0px}to{right:-1000px}}@-webkit-keyframes fadeIn{0%{opacity:0}100%{opacity:1}}@keyframes fadeIn{0%{opacity:0}100%{opacity:1}}@-webkit-keyframes fadeOut{0%{opacity:1}100%{opacity:0}}@keyframes fadeOut{0%{opacity:1}100%{opacity:0}}.transition-forward-animation-out.sc-phn-dealer-search-form{-webkit-transition:left 250ms;transition:left 250ms;position:relative !important;left:-150% !important}.transition-forward-animation-in.sc-phn-dealer-search-form{position:relative;-webkit-animation:transition-forward-in-animation 250ms;animation:transition-forward-in-animation 250ms}.transition-backward-animation-out.sc-phn-dealer-search-form{-webkit-transition:left 250ms;transition:left 250ms;position:relative !important;left:150% !important}.transition-backward-animation-in.sc-phn-dealer-search-form{position:relative !important;-webkit-animation:transition-backward-in-animation 250ms;animation:transition-backward-in-animation 250ms}.slide-out-to-right-fixed.sc-phn-dealer-search-form{-webkit-transition:left 400ms;transition:left 400ms;position:fixed;left:100% !important}.slide-in-from-right-fixed.sc-phn-dealer-search-form{position:fixed;-webkit-animation:single-drawer-transition-forward-in-animation 400ms;animation:single-drawer-transition-forward-in-animation 400ms}.phn-flex-col.sc-phn-dealer-search-form{display:-ms-flexbox;display:flex;-ms-flex-direction:column;flex-direction:column}.phn-flex-row.sc-phn-dealer-search-form{display:-ms-flexbox;display:flex;-ms-flex-direction:row;flex-direction:row}.phn-align-center.sc-phn-dealer-search-form{-ms-flex-align:center;align-items:center}.phn-justify-around.sc-phn-dealer-search-form{-ms-flex-pack:distribute;justify-content:space-around}.phn-justify-between.sc-phn-dealer-search-form{-ms-flex-pack:justify;justify-content:space-between}.phn-flex-wrap.sc-phn-dealer-search-form{-ms-flex-wrap:wrap;flex-wrap:wrap}.sc-phn-dealer-search-form-h{display:block}.skeleton.sc-phn-dealer-search-form-h{pointer-events:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.skeleton.sc-phn-dealer-search-form-h .skeleton-headline.sc-phn-dealer-search-form{width:auto;height:3.125rem;background:#d8d8db;margin-bottom:1.75rem;border-radius:4px}.sc-phn-dealer-search-form-h phn-headline.sc-phn-dealer-search-form{margin-bottom:2.25rem}";

const DealerSearchForm = /*@__PURE__*/ proxyCustomElement(class DealerSearchForm extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.dealer = undefined;
    this.locale = undefined;
    this.showHeadline = true;
    this.breakpoint = undefined;
    this.showOverlay = false;
    this.formErrorMessage = null;
    this.showSkeleton = true;
    this.animateResults = false;
  }
  toggleOverlay(e) {
    this.showOverlay = e.detail;
  }
  loadedAutoCompleteListener(_event) {
    this.showSkeleton = false;
  }
  getDealerFormStrategy() {
    const { dealer } = navContentStore.state;
    if (dealer.numberOfDealers <= 5) {
      return 'byNumberOfDealers';
    }
    if (showDealerSearchDropdown(splitLocale(this.locale).country))
      return 'dropdown';
    return 'geolocation';
  }
  formComponent() {
    const dealerSearchFormStrategies = {
      geolocation: () => h("phn-dealer-search-form-geolocation", { locale: this.locale }),
      dropdown: () => h("phn-dealer-search-form-dropdown", { locale: this.locale, breakpoint: this.breakpoint }),
      byNumberOfDealers: () => h("phn-dealer-search-by-number-of-dealers", { locale: this.locale })
    };
    const formStrategy = this.getDealerFormStrategy();
    return dealerSearchFormStrategies[formStrategy]();
  }
  render() {
    return (h(Host, { class: this.showSkeleton ? 'skeleton' : '' }, this.formComponent(), h("phn-dealer-search-results", { locale: this.locale, breakpoint: this.breakpoint, showOverlay: this.showOverlay })));
  }
  static get assetsDirs() { return ["./assets"]; }
  get el() { return this; }
  static get style() { return dealerSearchFormCss; }
}, [2, "phn-dealer-search-form", {
    "dealer": [16],
    "locale": [1],
    "showHeadline": [4, "show-headline"],
    "breakpoint": [2],
    "showOverlay": [32],
    "formErrorMessage": [32],
    "showSkeleton": [32],
    "animateResults": [32]
  }, [[0, "predictionsToggle", "toggleOverlay"], [0, "removeSkeletonOnDealerSearchForm", "loadedAutoCompleteListener"]]]);
function defineCustomElement() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["phn-dealer-search-form", "phn-autocomplete-search-field", "phn-dealer-details-drawer", "phn-dealer-search-by-number-of-dealers", "phn-dealer-search-form-dropdown", "phn-dealer-search-form-geolocation", "phn-dealer-search-results", "phn-menu-item-card"];
  components.forEach(tagName => { switch (tagName) {
    case "phn-dealer-search-form":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, DealerSearchForm);
      }
      break;
    case "phn-autocomplete-search-field":
      if (!customElements.get(tagName)) {
        defineCustomElement$7();
      }
      break;
    case "phn-dealer-details-drawer":
      if (!customElements.get(tagName)) {
        defineCustomElement$6();
      }
      break;
    case "phn-dealer-search-by-number-of-dealers":
      if (!customElements.get(tagName)) {
        defineCustomElement$5();
      }
      break;
    case "phn-dealer-search-form-dropdown":
      if (!customElements.get(tagName)) {
        defineCustomElement$4();
      }
      break;
    case "phn-dealer-search-form-geolocation":
      if (!customElements.get(tagName)) {
        defineCustomElement$3();
      }
      break;
    case "phn-dealer-search-results":
      if (!customElements.get(tagName)) {
        defineCustomElement$2();
      }
      break;
    case "phn-menu-item-card":
      if (!customElements.get(tagName)) {
        defineCustomElement$1();
      }
      break;
  } });
}
defineCustomElement();

export { DealerSearchForm as D, defineCustomElement as d };
