// @porschehn/navigation/dist/components/level-1.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import { b as getLevel1Items } from './analytics-manager.js';
import { d as dealerSearchResultsStore } from './dealer-search-results-store.js';
import { n as navContentStore } from './nav-content-store.js';
import { n as navStateStore } from './nav-state-store.js';
import { q as isDesktop, r as MenuItemVariant } from './helper.js';
import { i as isShopAvailable } from './shop-helper.js';
import { d as defineCustomElement$1 } from './menu-item.js';

const level1Css = "@-webkit-keyframes transition-forward-in-animation{from{left:150%}to{left:0%}}@keyframes transition-forward-in-animation{from{left:150%}to{left:0%}}@-webkit-keyframes transition-backward-in-animation{from{left:-150%}to{left:0%}}@keyframes transition-backward-in-animation{from{left:-150%}to{left:0%}}@-webkit-keyframes single-drawer-transition-forward-in-animation{from{left:100%}to{left:0%}}@keyframes single-drawer-transition-forward-in-animation{from{left:100%}to{left:0%}}@-webkit-keyframes fade-in-animation{from{opacity:0;background-color:rgba(0, 0, 0, 0)}to{opacity:1;background-color:rgba(0, 0, 0, 0.6)}}@keyframes fade-in-animation{from{opacity:0;background-color:rgba(0, 0, 0, 0)}to{opacity:1;background-color:rgba(0, 0, 0, 0.6)}}@-webkit-keyframes slide-in-from-left-absolute{from{left:-1000px}to{left:0px}}@keyframes slide-in-from-left-absolute{from{left:-1000px}to{left:0px}}@-webkit-keyframes slideDown{from{max-height:0}to{max-height:1000px}}@keyframes slideDown{from{max-height:0}to{max-height:1000px}}@-webkit-keyframes slideUp{from{max-height:1000px}to{max-height:0}}@keyframes slideUp{from{max-height:1000px}to{max-height:0}}@-webkit-keyframes slideLeft{from{max-width:0}to{max-width:1000px}}@keyframes slideLeft{from{max-width:0}to{max-width:1000px}}@-webkit-keyframes slideRight{from{max-width:1000px}to{max-width:0}}@keyframes slideRight{from{max-width:1000px}to{max-width:0}}@-webkit-keyframes darken{from{background-color:rgba(0, 0, 0, 0)}to{background-color:rgba(0, 0, 0, 0.8)}}@keyframes darken{from{background-color:rgba(0, 0, 0, 0)}to{background-color:rgba(0, 0, 0, 0.8)}}@-webkit-keyframes lighten{from{background-color:rgba(0, 0, 0, 0.8)}to{background-color:rgba(0, 0, 0, 0)}}@keyframes lighten{from{background-color:rgba(0, 0, 0, 0.8)}to{background-color:rgba(0, 0, 0, 0)}}@-webkit-keyframes fadeInFromBottom{0%{opacity:0;top:36px}100%{opacity:1;top:0px}}@keyframes fadeInFromBottom{0%{opacity:0;top:36px}100%{opacity:1;top:0px}}@-webkit-keyframes fadeOutToTop{0%{opacity:1;bottom:0px}100%{opacity:0;bottom:16px}}@keyframes fadeOutToTop{0%{opacity:1;bottom:0px}100%{opacity:0;bottom:16px}}@-webkit-keyframes slideLeftDoubleDrawer{from{right:-1000px}to{right:0px}}@keyframes slideLeftDoubleDrawer{from{right:-1000px}to{right:0px}}@-webkit-keyframes slideRightDoubleDrawer{from{right:0px}to{right:-1000px}}@keyframes slideRightDoubleDrawer{from{right:0px}to{right:-1000px}}@-webkit-keyframes fadeIn{0%{opacity:0}100%{opacity:1}}@keyframes fadeIn{0%{opacity:0}100%{opacity:1}}@-webkit-keyframes fadeOut{0%{opacity:1}100%{opacity:0}}@keyframes fadeOut{0%{opacity:1}100%{opacity:0}}.transition-forward-animation-out.sc-phn-level-1{-webkit-transition:left 250ms;transition:left 250ms;position:relative !important;left:-150% !important}.transition-forward-animation-in.sc-phn-level-1{position:relative;-webkit-animation:transition-forward-in-animation 250ms;animation:transition-forward-in-animation 250ms}.transition-backward-animation-out.sc-phn-level-1{-webkit-transition:left 250ms;transition:left 250ms;position:relative !important;left:150% !important}.transition-backward-animation-in.sc-phn-level-1{position:relative !important;-webkit-animation:transition-backward-in-animation 250ms;animation:transition-backward-in-animation 250ms}.slide-out-to-right-fixed.sc-phn-level-1{-webkit-transition:left 400ms;transition:left 400ms;position:fixed;left:100% !important}.slide-in-from-right-fixed.sc-phn-level-1{position:fixed;-webkit-animation:single-drawer-transition-forward-in-animation 400ms;animation:single-drawer-transition-forward-in-animation 400ms}.phn-flex-col.sc-phn-level-1{display:-ms-flexbox;display:flex;-ms-flex-direction:column;flex-direction:column}.phn-flex-row.sc-phn-level-1{display:-ms-flexbox;display:flex;-ms-flex-direction:row;flex-direction:row}.phn-align-center.sc-phn-level-1{-ms-flex-align:center;align-items:center}.phn-justify-around.sc-phn-level-1{-ms-flex-pack:distribute;justify-content:space-around}.phn-justify-between.sc-phn-level-1{-ms-flex-pack:justify;justify-content:space-between}.phn-flex-wrap.sc-phn-level-1{-ms-flex-wrap:wrap;flex-wrap:wrap}.sc-phn-level-1-h{display:block}.sc-phn-level-1-h phn-menu-item.sc-phn-level-1{margin-top:2.25rem}.sc-phn-level-1-h phn-menu-item.sc-phn-level-1:first-of-type{margin-top:0}";

var __rest = (undefined && undefined.__rest) || function (s, e) {
  var t = {};
  for (var p in s)
    if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
      t[p] = s[p];
  if (s != null && typeof Object.getOwnPropertySymbols === "function")
    for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
      if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
        t[p[i]] = s[p[i]];
    }
  return t;
};
const Level1 = /*@__PURE__*/ proxyCustomElement(class Level1 extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.mobileMenuItemClick = createEvent(this, "mobileMenuItemClick", 7);
    this.locale = undefined;
    this.breakpoint = undefined;
  }
  showRightIconRight(iconRight) {
    const { app } = navStateStore.state;
    const isOneUiMobileShop = isShopAvailable(this.locale, app) && !isDesktop(this.breakpoint);
    if (iconRight !== undefined && isOneUiMobileShop) {
      iconRight = 'arrow-head-right';
    }
    return iconRight;
  }
  render() {
    const { loggedInState, unreadMessagesCount, userConsent } = navStateStore.state;
    const level1Items = getLevel1Items(navContentStore.state, dealerSearchResultsStore.state, this.locale, loggedInState, unreadMessagesCount, userConsent);
    return (h("nav", null, level1Items.map((_a) => {
      var { routingKeys, elementId, iconRight, showIconModifier } = _a, level1Item = __rest(_a, ["routingKeys", "elementId", "iconRight", "showIconModifier"]);
      return (h("phn-menu-item", Object.assign({ key: elementId, elementId: elementId, onClick: () => {
          if (level1Item.link === undefined) {
            this.mobileMenuItemClick.emit(routingKeys);
          }
        }, iconRight: iconRight !== undefined ? this.showRightIconRight(iconRight) : 'arrow-head-right', variant: MenuItemVariant.Level1 }, level1Item), showIconModifier ? (h("span", { slot: "icon-modifier", "aria-hidden": "true", class: "myporsche-unread-messages-indicator", "data-test-id": "unread-messages-indicator" })) : null));
    })));
  }
  static get style() { return level1Css; }
}, [2, "phn-level-1", {
    "locale": [1],
    "breakpoint": [2]
  }]);
function defineCustomElement() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["phn-level-1", "phn-menu-item"];
  components.forEach(tagName => { switch (tagName) {
    case "phn-level-1":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, Level1);
      }
      break;
    case "phn-menu-item":
      if (!customElements.get(tagName)) {
        defineCustomElement$1();
      }
      break;
  } });
}
defineCustomElement();

export { Level1 as L, defineCustomElement as d };
