// @porschehn/navigation/dist/components/logo.js
import { proxyCustomElement, HTMLElement, createEvent, h } from '@stencil/core/internal/client';
import { g as globalDataLayerService, E as EventActions, C as ClickElementTypes } from './global-data-layer-service.js';
import { c as componentsJs } from './index2.js';
import { n as navContentStore } from './nav-content-store.js';
import { N as NavigationTheme, m as NavigationMode, q as isDesktop, a as isString, o as getRightTheme } from './helper.js';
import { b as isShopAvailableAndLogoLinksToShop } from './shop-helper.js';

const logoCss = "";

const Logo = /*@__PURE__*/ proxyCustomElement(class Logo extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.logoLoaded = createEvent(this, "logoLoaded", 7);
    this.size = undefined;
    this.locale = undefined;
    this.app = undefined;
    this.theme = NavigationTheme.light;
    this.mode = NavigationMode.navbar;
    this.breakpoint = undefined;
  }
  sendDataLayerEvent(crest) {
    globalDataLayerService.pushDataLayerEvent(EventActions.CREST_CLICK, {
      componentClick: {
        clickElementId: crest.id,
        clickElementName: crest.text,
        clickElementType: ClickElementTypes.Link,
        targetUrl: crest.link
      }
    });
  }
  componentWillLoad() {
    componentsJs.componentsReady(this.el) // TR: I believe this first call waits for the phn-logo element to be ready
      .then(() => componentsJs.componentsReady(this.el)) // TR: Then this second call actually waits for its children (i.e.: phn-p-marque) to be ready
      .then(() => {
      this.logoLoaded.emit();
    });
  }
  getLogoLink(locale, app, crest, shop) {
    return isShopAvailableAndLogoLinksToShop(locale, app, shop) ? shop.link : crest.link;
  }
  getLogoAriaLabel(locale, app, shop) {
    return isShopAvailableAndLogoLinksToShop(locale, app, shop) ? 'Porsche Shop Homepage' : 'Porsche.com Homepage';
  }
  render() {
    const { crest, shop } = navContentStore.state;
    const logoLink = this.getLogoLink(this.locale, this.app, crest, shop);
    const logoAriaLabel = this.getLogoAriaLabel(this.locale, this.app, shop);
    return isDesktop(this.breakpoint) ? (h("phn-p-wordmark", { onClick: () => this.sendDataLayerEvent(crest), href: isString(logoLink) ? logoLink : undefined, aria: { 'aria-label': logoAriaLabel }, size: "small", theme: getRightTheme(this.theme, this.mode) })) : (h("phn-p-crest", { onClick: () => this.sendDataLayerEvent(crest), href: isString(logoLink) ? logoLink : undefined, aria: { 'aria-label': logoAriaLabel } }));
  }
  get el() { return this; }
  static get style() { return logoCss; }
}, [2, "phn-logo", {
    "size": [1],
    "locale": [1],
    "app": [1],
    "theme": [1],
    "mode": [1],
    "breakpoint": [2]
  }]);
function defineCustomElement() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["phn-logo"];
  components.forEach(tagName => { switch (tagName) {
    case "phn-logo":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, Logo);
      }
      break;
  } });
}
defineCustomElement();

export { Logo as L, defineCustomElement as d };
