// @porschehn/navigation/dist/components/nav-content-store.js
import { c as createStore } from './navi-error.js';
import { a as isString } from './helper.js';

const EMPTY_LINK = { id: '', text: '', link: '', icon: '', description: '' };
const EMPTY_BUTTON = { id: '', text: '', description: '' };
const EMPTY_COUNTRY_RECOMMENDER = {
  statement: '',
  region: '',
  language: '',
  change: EMPTY_BUTTON
};
const EMPTY_MODELS = {
  text: '',
  id: '',
  all: {
    id: '',
    text: '',
    link: '',
    icon: '',
    description: ''
  },
  series: []
};
const EMPTY_MENU = {
  id: '',
  text: '',
  link: '',
  children: []
};
const EMPTY_SHOP_MENU = {
  cart: '',
  additionalContent: [],
  id: '',
  text: '',
  link: '',
  children: [],
  availableLocales: [],
  availableLocalesOnlyForShop: [],
  logoLinksToShopLocales: []
};
const EMPTY_MYPORSCHE = {
  id: '',
  text: '',
  link: '',
  loggedInFallback: '',
  login: EMPTY_LINK,
  loginText: '',
  logout: EMPTY_LINK,
  messages: EMPTY_LINK,
  more: null,
  signup: EMPTY_LINK,
  signupText: '',
  links: [],
  savedSearches: null,
  savedVehicles: null
};
const EMPTY_DEALER = {
  id: '',
  text: '',
  numberOfDealers: 42,
  search: {
    id: '',
    text: '',
    form: {
      inputPlaceholder: '',
      inputDescription: ''
    },
    clear: EMPTY_BUTTON,
    or: '',
    geolocation: EMPTY_BUTTON,
    dropdown: {
      province: '',
      city: '',
      selectAProvince: '',
      selectACity: ''
    },
    load: '',
    map: EMPTY_LINK,
    outboundLink: EMPTY_LINK
  },
  details: {
    id: '',
    finder: EMPTY_LINK,
    bookService: EMPTY_LINK,
    testDrive: EMPTY_LINK,
    backToSearch: ''
  },
  openingHours: null
};
const EMPTY_ERRORS = {
  wrong: '',
  retry: '',
  noResults: '',
  noInput: '',
  invalidInput: '',
  geolocationDenied: ''
};
const EMPTY_CONTENT = {
  menu: EMPTY_BUTTON,
  crest: EMPTY_LINK,
  back: EMPTY_BUTTON,
  close: EMPTY_BUTTON,
  accessibilityStatement: EMPTY_LINK,
  countryRecommender: EMPTY_COUNTRY_RECOMMENDER,
  contextualButtons: {},
  contextualLinks: {},
  models: EMPTY_MODELS,
  dealer: EMPTY_DEALER,
  myPorsche: EMPTY_MYPORSCHE,
  vehiclePurchase: EMPTY_MENU,
  shop: EMPTY_SHOP_MENU,
  externalShop: EMPTY_LINK,
  experience: EMPTY_MENU,
  services: EMPTY_MENU,
  error: EMPTY_ERRORS,
  loadingDescription: '',
  legacyMore: EMPTY_MENU,
  legacyShop: EMPTY_MENU
};
function findMenuItemById(menu, id) {
  if (!isString(id)) {
    return null;
  }
  if (menu !== null) {
    if (menu.id === id) {
      return menu;
    }
    if (!Array.isArray(menu.children)) {
      return null;
    }
    for (const child of menu.children) {
      const foundChild = findMenuItemById(child, id);
      if (foundChild !== null) {
        return foundChild;
      }
    }
  }
  return null;
}

const navContentStore = createStore(EMPTY_CONTENT);

export { findMenuItemById as f, navContentStore as n };
