// @porschehn/navigation/dist/components/wrapper.js
import { proxyCustomElement, HTMLElement, h, Host } from '@stencil/core/internal/client';
import { E as Environments } from './navi-error.js';
import { w as windowService } from './window-service.js';
import { n as navContentStore } from './nav-content-store.js';
import { n as navStateStore } from './nav-state-store.js';
import { N as NavigationTheme, m as NavigationMode, A as Application, V as VERSION, g as getHTMLElement } from './helper.js';
import { a as animationManager } from './animation-manager.js';
import { i as isShopAvailable } from './shop-helper.js';
import { d as defineCustomElement$w } from './contextual-drawer2.js';
import { d as defineCustomElement$v } from './autocomplete-search-field.js';
import { d as defineCustomElement$u } from './back-button.js';
import { d as defineCustomElement$t } from './backdrop.js';
import { d as defineCustomElement$s } from './burger-button.js';
import { d as defineCustomElement$r } from './car-body-types-drawer.js';
import { d as defineCustomElement$q } from './country-recommender.js';
import { d as defineCustomElement$p } from './dealer-details-drawer.js';
import { d as defineCustomElement$o } from './dealer-search-by-number-of-dealers.js';
import { d as defineCustomElement$n } from './dealer-search-form.js';
import { d as defineCustomElement$m } from './dealer-search-form-dropdown.js';
import { d as defineCustomElement$l } from './dealer-search-form-geolocation.js';
import { d as defineCustomElement$k } from './dealer-search-results.js';
import { d as defineCustomElement$j } from './double-drawer.js';
import { d as defineCustomElement$i } from './drawer-header.js';
import { d as defineCustomElement$h } from './heading.js';
import { d as defineCustomElement$g } from './level-0.js';
import { d as defineCustomElement$f } from './level-0-icons.js';
import { d as defineCustomElement$e } from './level-1.js';
import { d as defineCustomElement$d } from './logo.js';
import { d as defineCustomElement$c } from './menu.js';
import { d as defineCustomElement$b } from './menu-item.js';
import { d as defineCustomElement$a } from './menu-item-card.js';
import { d as defineCustomElement$9 } from './models.js';
import { d as defineCustomElement$8 } from './my-porsche-icon-button.js';
import { d as defineCustomElement$7 } from './myporsche-drawer.js';
import { d as defineCustomElement$6 } from './myporsche-login.js';
import { d as defineCustomElement$5 } from './router.js';
import { d as defineCustomElement$4 } from './side-drawer.js';
import { d as defineCustomElement$3 } from './side-drawer-item.js';
import { d as defineCustomElement$2 } from './signature-heading.js';
import { d as defineCustomElement$1 } from './single-drawer.js';

const wrapperCss = "@-webkit-keyframes transition-forward-in-animation{from{left:150%}to{left:0%}}@keyframes transition-forward-in-animation{from{left:150%}to{left:0%}}@-webkit-keyframes transition-backward-in-animation{from{left:-150%}to{left:0%}}@keyframes transition-backward-in-animation{from{left:-150%}to{left:0%}}@-webkit-keyframes single-drawer-transition-forward-in-animation{from{left:100%}to{left:0%}}@keyframes single-drawer-transition-forward-in-animation{from{left:100%}to{left:0%}}@-webkit-keyframes fade-in-animation{from{opacity:0;background-color:rgba(0, 0, 0, 0)}to{opacity:1;background-color:rgba(0, 0, 0, 0.6)}}@keyframes fade-in-animation{from{opacity:0;background-color:rgba(0, 0, 0, 0)}to{opacity:1;background-color:rgba(0, 0, 0, 0.6)}}@-webkit-keyframes slide-in-from-left-absolute{from{left:-1000px}to{left:0px}}@keyframes slide-in-from-left-absolute{from{left:-1000px}to{left:0px}}@-webkit-keyframes slideDown{from{max-height:0}to{max-height:1000px}}@keyframes slideDown{from{max-height:0}to{max-height:1000px}}@-webkit-keyframes slideUp{from{max-height:1000px}to{max-height:0}}@keyframes slideUp{from{max-height:1000px}to{max-height:0}}@-webkit-keyframes slideLeft{from{max-width:0}to{max-width:1000px}}@keyframes slideLeft{from{max-width:0}to{max-width:1000px}}@-webkit-keyframes slideRight{from{max-width:1000px}to{max-width:0}}@keyframes slideRight{from{max-width:1000px}to{max-width:0}}@-webkit-keyframes darken{from{background-color:rgba(0, 0, 0, 0)}to{background-color:rgba(0, 0, 0, 0.8)}}@keyframes darken{from{background-color:rgba(0, 0, 0, 0)}to{background-color:rgba(0, 0, 0, 0.8)}}@-webkit-keyframes lighten{from{background-color:rgba(0, 0, 0, 0.8)}to{background-color:rgba(0, 0, 0, 0)}}@keyframes lighten{from{background-color:rgba(0, 0, 0, 0.8)}to{background-color:rgba(0, 0, 0, 0)}}@-webkit-keyframes fadeInFromBottom{0%{opacity:0;top:36px}100%{opacity:1;top:0px}}@keyframes fadeInFromBottom{0%{opacity:0;top:36px}100%{opacity:1;top:0px}}@-webkit-keyframes fadeOutToTop{0%{opacity:1;bottom:0px}100%{opacity:0;bottom:16px}}@keyframes fadeOutToTop{0%{opacity:1;bottom:0px}100%{opacity:0;bottom:16px}}@-webkit-keyframes slideLeftDoubleDrawer{from{right:-1000px}to{right:0px}}@keyframes slideLeftDoubleDrawer{from{right:-1000px}to{right:0px}}@-webkit-keyframes slideRightDoubleDrawer{from{right:0px}to{right:-1000px}}@keyframes slideRightDoubleDrawer{from{right:0px}to{right:-1000px}}@-webkit-keyframes fadeIn{0%{opacity:0}100%{opacity:1}}@keyframes fadeIn{0%{opacity:0}100%{opacity:1}}@-webkit-keyframes fadeOut{0%{opacity:1}100%{opacity:0}}@keyframes fadeOut{0%{opacity:1}100%{opacity:0}}.transition-forward-animation-out.sc-phn-wrapper{-webkit-transition:left 250ms;transition:left 250ms;position:relative !important;left:-150% !important}.transition-forward-animation-in.sc-phn-wrapper{position:relative;-webkit-animation:transition-forward-in-animation 250ms;animation:transition-forward-in-animation 250ms}.transition-backward-animation-out.sc-phn-wrapper{-webkit-transition:left 250ms;transition:left 250ms;position:relative !important;left:150% !important}.transition-backward-animation-in.sc-phn-wrapper{position:relative !important;-webkit-animation:transition-backward-in-animation 250ms;animation:transition-backward-in-animation 250ms}.slide-out-to-right-fixed.sc-phn-wrapper{-webkit-transition:left 400ms;transition:left 400ms;position:fixed;left:100% !important}.slide-in-from-right-fixed.sc-phn-wrapper{position:fixed;-webkit-animation:single-drawer-transition-forward-in-animation 400ms;animation:single-drawer-transition-forward-in-animation 400ms}.phn-flex-col.sc-phn-wrapper{display:-ms-flexbox;display:flex;-ms-flex-direction:column;flex-direction:column}.phn-flex-row.sc-phn-wrapper{display:-ms-flexbox;display:flex;-ms-flex-direction:row;flex-direction:row}.phn-align-center.sc-phn-wrapper{-ms-flex-align:center;align-items:center}.phn-justify-around.sc-phn-wrapper{-ms-flex-pack:distribute;justify-content:space-around}.phn-justify-between.sc-phn-wrapper{-ms-flex-pack:justify;justify-content:space-between}.phn-flex-wrap.sc-phn-wrapper{-ms-flex-wrap:wrap;flex-wrap:wrap}.sc-phn-wrapper-h{display:block}.sc-phn-wrapper-h .limit-stretch.sc-phn-wrapper{-webkit-box-sizing:content-box;box-sizing:content-box;margin:0 0.5vw}.sc-phn-wrapper-h .preview-badge.sc-phn-wrapper{display:block;position:absolute;top:0;right:0;background-color:#cc1922;z-index:9999;padding:0 0.125rem}.sc-phn-wrapper-h .main-content.sc-phn-wrapper{width:20px}.sc-phn-wrapper-h .main-content.sc-phn-wrapper:focus{outline-style:none}.sc-phn-wrapper-h .screenreader-only.sc-phn-wrapper{position:absolute;width:1px;height:1px;white-space:nowrap;overflow:hidden;clip:rect(1px, 1px, 1px, 1px);-webkit-clip-path:inset(50%);clip-path:inset(50%)}";

const Wrapper = /*@__PURE__*/ proxyCustomElement(class Wrapper extends HTMLElement {
  constructor() {
    super();
    this.__registerHost();
    this.locale = undefined;
    this.breakpoint = undefined;
    this.loadingContent = false;
    this.receivedContent = false;
    this.theme = NavigationTheme.light;
    this.mode = NavigationMode.navbar;
    this.app = Application.default;
    this.displayLogoOnly = false;
    this.limitedWidth = false;
    this.showCountryRecommender = windowService.redirected;
    this.countryRecommenderClosed = false;
  }
  componentWillLoad() {
    this.el.style.setProperty('--scrollbar-width', windowService.getScrollbarWidth());
    animationManager.initialize(this.el);
  }
  render() {
    const countryRecommenderContent = navContentStore.state.countryRecommender;
    const showCountryRecommender = this.showCountryRecommender && !this.countryRecommenderClosed && !this.loadingContent && this.receivedContent;
    const accessibilityStatement = navContentStore.state.accessibilityStatement;
    return (h(Host, null, navStateStore.state.env === Environments.PREVIEW ? (h("phn-p-text", { theme: "dark", size: "x-small", class: "preview-badge" }, "Header v", VERSION, " - PP content")) : null, showCountryRecommender ? (h("phn-country-recommender", { locale: this.locale, content: countryRecommenderContent, role: "region", "aria-label": "Change recommended Region or Language", onBannerClosed: () => {
        this.countryRecommenderClosed = true;
      } })) : null, this.locale === 'en-US' && accessibilityStatement !== null ? (h("a", { class: "screenreader-only", href: accessibilityStatement.link }, accessibilityStatement.text)) : null, h("phn-level-0", { theme: this.theme, mode: this.mode, locale: this.locale, loadingContent: this.loadingContent, receivedContent: this.receivedContent, breakpoint: this.breakpoint, app: this.app, displayLogoOnly: this.displayLogoOnly, limitedWidth: this.limitedWidth }, isShopAvailable(this.locale, this.app) ? (h("div", { slot: "contextual_drawer_header" }, h("slot", { name: "contextual_drawer_header" }))) : null), h("div", { id: "main", class: "main-content", onBlur: (e) => {
        const el = getHTMLElement(e.currentTarget);
        if (el !== null) {
          el.removeAttribute('tabIndex');
        }
      } })));
  }
  get el() { return this; }
  static get style() { return wrapperCss; }
}, [6, "phn-wrapper", {
    "locale": [1],
    "breakpoint": [2],
    "loadingContent": [4, "loading-content"],
    "receivedContent": [4, "received-content"],
    "theme": [1],
    "mode": [1],
    "app": [1],
    "displayLogoOnly": [4, "display-logo-only"],
    "limitedWidth": [4, "limited-width"],
    "showCountryRecommender": [32],
    "countryRecommenderClosed": [32]
  }]);
function defineCustomElement() {
  if (typeof customElements === "undefined") {
    return;
  }
  const components = ["phn-wrapper", "contextual-drawer", "phn-autocomplete-search-field", "phn-back-button", "phn-backdrop", "phn-burger-button", "phn-car-body-types-drawer", "phn-country-recommender", "phn-dealer-details-drawer", "phn-dealer-search-by-number-of-dealers", "phn-dealer-search-form", "phn-dealer-search-form-dropdown", "phn-dealer-search-form-geolocation", "phn-dealer-search-results", "phn-double-drawer", "phn-drawer-header", "phn-heading", "phn-level-0", "phn-level-0-icons", "phn-level-1", "phn-logo", "phn-menu", "phn-menu-item", "phn-menu-item-card", "phn-models", "phn-my-porsche-icon-button", "phn-myporsche-drawer", "phn-myporsche-login", "phn-router", "phn-side-drawer", "phn-side-drawer-item", "phn-signature-heading", "phn-single-drawer"];
  components.forEach(tagName => { switch (tagName) {
    case "phn-wrapper":
      if (!customElements.get(tagName)) {
        customElements.define(tagName, Wrapper);
      }
      break;
    case "contextual-drawer":
      if (!customElements.get(tagName)) {
        defineCustomElement$w();
      }
      break;
    case "phn-autocomplete-search-field":
      if (!customElements.get(tagName)) {
        defineCustomElement$v();
      }
      break;
    case "phn-back-button":
      if (!customElements.get(tagName)) {
        defineCustomElement$u();
      }
      break;
    case "phn-backdrop":
      if (!customElements.get(tagName)) {
        defineCustomElement$t();
      }
      break;
    case "phn-burger-button":
      if (!customElements.get(tagName)) {
        defineCustomElement$s();
      }
      break;
    case "phn-car-body-types-drawer":
      if (!customElements.get(tagName)) {
        defineCustomElement$r();
      }
      break;
    case "phn-country-recommender":
      if (!customElements.get(tagName)) {
        defineCustomElement$q();
      }
      break;
    case "phn-dealer-details-drawer":
      if (!customElements.get(tagName)) {
        defineCustomElement$p();
      }
      break;
    case "phn-dealer-search-by-number-of-dealers":
      if (!customElements.get(tagName)) {
        defineCustomElement$o();
      }
      break;
    case "phn-dealer-search-form":
      if (!customElements.get(tagName)) {
        defineCustomElement$n();
      }
      break;
    case "phn-dealer-search-form-dropdown":
      if (!customElements.get(tagName)) {
        defineCustomElement$m();
      }
      break;
    case "phn-dealer-search-form-geolocation":
      if (!customElements.get(tagName)) {
        defineCustomElement$l();
      }
      break;
    case "phn-dealer-search-results":
      if (!customElements.get(tagName)) {
        defineCustomElement$k();
      }
      break;
    case "phn-double-drawer":
      if (!customElements.get(tagName)) {
        defineCustomElement$j();
      }
      break;
    case "phn-drawer-header":
      if (!customElements.get(tagName)) {
        defineCustomElement$i();
      }
      break;
    case "phn-heading":
      if (!customElements.get(tagName)) {
        defineCustomElement$h();
      }
      break;
    case "phn-level-0":
      if (!customElements.get(tagName)) {
        defineCustomElement$g();
      }
      break;
    case "phn-level-0-icons":
      if (!customElements.get(tagName)) {
        defineCustomElement$f();
      }
      break;
    case "phn-level-1":
      if (!customElements.get(tagName)) {
        defineCustomElement$e();
      }
      break;
    case "phn-logo":
      if (!customElements.get(tagName)) {
        defineCustomElement$d();
      }
      break;
    case "phn-menu":
      if (!customElements.get(tagName)) {
        defineCustomElement$c();
      }
      break;
    case "phn-menu-item":
      if (!customElements.get(tagName)) {
        defineCustomElement$b();
      }
      break;
    case "phn-menu-item-card":
      if (!customElements.get(tagName)) {
        defineCustomElement$a();
      }
      break;
    case "phn-models":
      if (!customElements.get(tagName)) {
        defineCustomElement$9();
      }
      break;
    case "phn-my-porsche-icon-button":
      if (!customElements.get(tagName)) {
        defineCustomElement$8();
      }
      break;
    case "phn-myporsche-drawer":
      if (!customElements.get(tagName)) {
        defineCustomElement$7();
      }
      break;
    case "phn-myporsche-login":
      if (!customElements.get(tagName)) {
        defineCustomElement$6();
      }
      break;
    case "phn-router":
      if (!customElements.get(tagName)) {
        defineCustomElement$5();
      }
      break;
    case "phn-side-drawer":
      if (!customElements.get(tagName)) {
        defineCustomElement$4();
      }
      break;
    case "phn-side-drawer-item":
      if (!customElements.get(tagName)) {
        defineCustomElement$3();
      }
      break;
    case "phn-signature-heading":
      if (!customElements.get(tagName)) {
        defineCustomElement$2();
      }
      break;
    case "phn-single-drawer":
      if (!customElements.get(tagName)) {
        defineCustomElement$1();
      }
      break;
  } });
}
defineCustomElement();

export { Wrapper as W, defineCustomElement as d };
