// @porschehn/navigation/dist/collection/components/contextual-drawer/contextual-drawer-animation-sequence.js
import animationManager, { makeSequence } from "../../services/animations/animation-manager";
import { AnimationDurations } from "../../utility/constants";
export const registerContextualDrawerOpening = ({ onStartOpening, onFinishOpening }) => {
  animationManager.register('openContextualDrawer', makeSequence([
    [
      {
        onStart: onStartOpening,
        elementSelector: 'contextual-drawer',
        animation: {
          name: 'slideInFromLeft',
          to: 0,
          duration: AnimationDurations.drawerOpen,
          from: -1000
        }
      },
      {
        onEnd: onFinishOpening,
        elementSelector: '.contextual-drawer-backdrop',
        animation: {
          name: 'fadeInBackdrop',
          duration: AnimationDurations.backdrop,
          to: 0.6
        }
      }
    ]
  ], { timingFunction: 'cubicBezierStart' }));
};
export const registerContextualDrawerClosing = ({ onStartClosing, onEndClosing }) => {
  animationManager.register('closeContextualDrawer', makeSequence([
    [
      {
        onStart: onStartClosing,
        elementSelector: 'contextual-drawer',
        animation: {
          name: 'slideOutToLeft',
          to: -1000,
          duration: AnimationDurations.drawer
        }
      },
      {
        elementSelector: '.contextual-drawer-backdrop',
        animation: {
          name: 'fadeOutBackdrop',
          to: 0.6,
          duration: AnimationDurations.backdrop
        },
        onEnd: onEndClosing
      }
    ]
  ], { timingFunction: 'cubicBezierEnd' }));
};
