// @porschehn/navigation/dist/collection/components/contextual-drawer/contextual-drawer-animation-sequence.js
import animationManager, { makeSequence } from "../../services/animations/animation-manager";
export const registerContextualDrawerOpening = ({ onStartOpening, onFinishOpening }) => {
  animationManager.register('openContextualDrawer', makeSequence([
    [
      {
        onStart: onStartOpening,
        elementSelector: 'contextual-drawer',
        animation: {
          className: 'slide-in-from-left',
          cleanUpAfterAnimation: (element) => {
            element.style.left = '0px';
          }
        }
      },
      {
        onEnd: onFinishOpening,
        elementSelector: '.contextual-drawer-backdrop',
        animation: {
          className: 'fade-in-animation',
          cleanUpAfterAnimation: (element) => {
            element.style.backgroundColor = 'rgba(0, 0, 0, 0.6)';
            element.style.opacity = '1';
          }
        }
      }
    ]
  ]));
};
export const registerContextualDrawerClosing = ({ onStartClosing }) => {
  animationManager.register('closeContextualDrawer', makeSequence([
    [
      {
        onStart: onStartClosing,
        elementSelector: 'contextual-drawer',
        animation: {
          className: 'slide-out-to-left',
          cleanUpBeforeAnimation: (element) => {
            element.style.left = '';
          }
          // cleanUpBeforeAnimation: (element) => {
          //   element.style.left = '0px';
          // }
        }
      },
      {
        elementSelector: '.contextual-drawer-backdrop',
        animation: {
          className: 'fade-out-animation',
          cleanUpBeforeAnimation: (element) => {
            element.style.backgroundColor = '';
            element.style.opacity = '';
          }
        }
      }
    ]
  ]));
};
