// @porschehn/navigation/dist/cjs/analytics-manager-4e31846c.js
'use strict';

const routingKey = require('./routing-key-2cb3d5c5.js');
const routerStore = require('./router-store-1a4e63e5.js');
const navContentStore = require('./nav-content-store-ed0e79aa.js');
const helper = require('./helper-67121a1c.js');
const menuItemsService = require('./menu-items-service-3605196c.js');

const state = {};
function setDrawerType(drawerType) {
  state.drawerType = drawerType;
}
function calculateDisplayElementId(routingKeyForNavigatingForward) {
  const { drawerType } = state;
  const mainDrawerRoutingKey = (helper.isString(routingKeyForNavigatingForward)
    ? routingKey.getPreviousRoutingKey(routingKeyForNavigatingForward)
    : routerStore.routerStore.state.routingKeyHistory[routerStore.routerStore.state.routingKeyPointer]) || '';
  const lastMainDrawerRoutingKey = routingKey.getLastRoutingSegment(mainDrawerRoutingKey);
  const mainDrawerLevel = routingKey.determineMainDrawerLevel(mainDrawerRoutingKey);
  const sideDrawerLevel = mainDrawerLevel - 1;
  if (!helper.isString(mainDrawerRoutingKey)) {
    return 'level0';
  }
  const mainDrawerDisplayElementId = helper.isString(mainDrawerRoutingKey)
    ? `level${mainDrawerLevel}-${lastMainDrawerRoutingKey}`
    : `level${mainDrawerLevel - 1}-unknown`;
  if (drawerType === 'doubleDrawer') {
    if (mainDrawerRoutingKey === '') {
      return `level${sideDrawerLevel}-unknown_level${mainDrawerLevel}-unknown`;
    }
    const sideDrawerRoutingKey = menuItemsService.getSideDrawerRoutingKey(mainDrawerRoutingKey);
    return `level${sideDrawerLevel}-${sideDrawerRoutingKey}_${mainDrawerDisplayElementId}`;
  }
  return mainDrawerDisplayElementId;
}
function calculateDisplayElementType(activeRoutingKey, elementType, isSideDrawer) {
  const { drawerType } = state;
  const mainDrawerLevel = routingKey.determineMainDrawerLevel(activeRoutingKey);
  if (drawerType === 'doubleDrawer') {
    let sideDrawerLevel = mainDrawerLevel;
    if (isSideDrawer) {
      sideDrawerLevel = sideDrawerLevel - 1;
    }
    return `level${sideDrawerLevel}_${elementType}`;
  }
  else {
    return helper.isString(activeRoutingKey)
      ? `level${mainDrawerLevel}_${elementType}`
      : `level${mainDrawerLevel - 1}_${elementType}`;
  }
}
function getModelName(bodyType) {
  const urlSplitIntoPieces = bodyType.link.split('/');
  const lastElementInUrlSplit = urlSplitIntoPieces[urlSplitIntoPieces.length - 1];
  let modelName;
  if (lastElementInUrlSplit === '') {
    if (urlSplitIntoPieces[urlSplitIntoPieces.length - 2] === undefined) {
      modelName = '';
    }
    else {
      modelName = urlSplitIntoPieces[urlSplitIntoPieces.length - 2].split('-').join(' ');
    }
  }
  else {
    modelName = lastElementInUrlSplit.split('-').join(' ');
  }
  return modelName;
}
function getModelTargetElement(elementId, text) {
  if (elementId.includes('models')) {
    const mainDrawerLevel = routingKey.determineMainDrawerLevel(elementId);
    return `level${mainDrawerLevel}_${text.toLowerCase()}`;
  }
  return;
}
function getDrawerType() {
  return state.drawerType;
}
function getDealerResultsLoadDisplayElementId(elementId) {
  if (getDrawerType() === 'doubleDrawer') {
    return elementId;
  }
  else {
    return calculateDisplayElementId();
  }
}
function getDealerResultsLoadDisplayElementType() {
  if (getDrawerType() === 'doubleDrawer') {
    return navContentStore.DEALER_SEARCH_ELEMENT_TYPE;
  }
  else {
    return calculateDisplayElementType('', navContentStore.ElementTypes.dealerSearchForm);
  }
}
const analyticsManager = {
  setDrawerType,
  getDealerResultsLoadDisplayElementId,
  calculateDisplayElementId,
  getDealerResultsLoadDisplayElementType,
  calculateDisplayElementType,
  getModelName,
  getModelTargetElement
};

exports.analyticsManager = analyticsManager;
