// @porschehn/navigation/dist/cjs/content-service-cd957c0f.js
'use strict';

const navContentStore = require('./nav-content-store-ed0e79aa.js');
const helper = require('./helper-67121a1c.js');
const shopHelper = require('./shop-helper-d333e4b0.js');
const featureToggles = require('./feature-toggles-bb23f232.js');

const contentCache = new Map();
async function fetchContent(locale, app) {
  var _a;
  const config = helper.getConfig(featureToggles.featureToggles.env);
  const isShopEnabled = featureToggles.featureToggles.isShopToggleActive() && app === helper.Application.shop;
  const naviContentPromise = fetchNaviContent(config.CONTENT_URL, locale);
  const shopContentPromise = isShopEnabled ? fetchShopContent(config.SHOP_CONTENT_URL, locale) : Promise.resolve(null);
  const [naviContentResult, shopContentResult] = await Promise.allSettled([naviContentPromise, shopContentPromise]);
  if (naviContentResult.status === 'rejected') {
    throw naviContentResult.reason;
  }
  const naviContent = shopHelper.modifyShopItemDependingOnMarket(locale, app, naviContentResult.value);
  const shopContent = shopContentResult.status === 'fulfilled' ? shopContentResult.value : null;
  if (shopContent === null || naviContent.shop === null) {
    return naviContent;
  }
  return Object.assign(Object.assign({}, naviContent), { shop: Object.assign(Object.assign({}, naviContent.shop), { children: shopContent.menuItems || ((_a = naviContent.shop) === null || _a === void 0 ? void 0 : _a.children) || [], additionalContent: shopContent.additionalContent, availableLocales: naviContent.shop.availableLocales, availableLocalesOnlyForShop: naviContent.shop.availableLocalesOnlyForShop }) });
}
async function fetchNaviContent(contentUrl, locale) {
  const { country, language } = navContentStore.splitLocale(locale);
  const cacheKey = helper.constructCacheKey(contentUrl, locale);
  const cachedContent = contentCache.get(cacheKey);
  if (cachedContent !== undefined) {
    return cachedContent;
  }
  const data = await fetch(`${contentUrl}/${`${language}-${country}`}.json`, {
    headers: {
      'Cache-Control': 'no-cache',
      'Content-Type': 'application/json'
    }
  });
  if (data.status === 404) {
    throw new navContentStore.NaviError(`No content for locale ${locale} found`, navContentStore.NaviErrorTypes.NO_RESULTS);
  }
  if (!data.ok) {
    throw new navContentStore.NaviError(`Could not get content for locale ${locale}`, navContentStore.NaviErrorTypes.GENERAL);
  }
  const content = await data.json();
  contentCache.set(cacheKey, content);
  return content;
}
async function fetchShopContent(contentUrl, locale) {
  const { country, language } = navContentStore.splitLocale(locale);
  const data = await fetch(`${contentUrl}/${country.toLowerCase()}/${`${language}-${country}`}/navigation`, {
    headers: {
      'Content-Type': 'application/json',
      'x-vercel-protection-bypass': 'hXMkVgu6erfnX2ydOdeSCixWDWbPjBUk'
    }
  });
  if (data.status === 404) {
    throw new navContentStore.NaviError(`No shop content for locale ${locale} found`, navContentStore.NaviErrorTypes.NO_RESULTS);
  }
  if (!data.ok) {
    throw new navContentStore.NaviError(`Could not get shop content for locale ${locale}`, navContentStore.NaviErrorTypes.GENERAL);
  }
  return data.json();
}
function constructFlagURL(footerAssetsUrl, locale) {
  const { country } = navContentStore.splitLocale(locale);
  let flag = country;
  if (flag.includes('-') && !navContentStore.isCountryInOneOfRegions(flag)) {
    flag = country.split('-')[0];
  }
  return `${footerAssetsUrl}/flags/${flag}.svg`;
}
async function getFlagURL(footerAssetsUrl, locale) {
  try {
    const flagURL = constructFlagURL(footerAssetsUrl, locale);
    const res = await fetch(flagURL);
    // Only set flagURL if flag actually exists
    if (res.ok) {
      return flagURL;
    }
  }
  catch (err) {
    // ignore error
  }
  return '';
}

exports.fetchContent = fetchContent;
exports.getFlagURL = getFlagURL;
