// @porschehn/navigation/dist/cjs/data-layer-event-sender-e5275043.js
'use strict';

const navContentStore = require('./nav-content-store-ed0e79aa.js');
const globalDataLayerService = require('./global-data-layer-service-534dfa5c.js');
const analyticsManager = require('./analytics-manager-4e31846c.js');
const routingKey = require('./routing-key-2cb3d5c5.js');

function sendDataLayerEvent({ datalayerEventAction = navContentStore.EventActions.NAVIGATION_CLICK, analyticsId, text, link, elementId, clickElementType, customDatalayerProperties = {} }) {
  globalDataLayerService.globalDataLayerService.pushDataLayerEvent(datalayerEventAction, Object.assign(Object.assign({ componentClick: Object.assign({ clickElementId: analyticsId, clickElementName: text, targetElement: analyticsManager.analyticsManager.getModelTargetElement(elementId, text), clickElementType }, (clickElementType === navContentStore.ClickElementTypes.Link ? { targetUrl: link } : {})), componentDisplay: {
      displayElementId: analyticsManager.analyticsManager.calculateDisplayElementId(elementId),
      displayElementType: analyticsManager.analyticsManager.calculateDisplayElementType(routingKey.removeLastRoutingSegment(elementId), navContentStore.ElementTypes.menuItem),
      displayElementName: text
    } }, customDatalayerProperties), (datalayerEventAction === navContentStore.EventActions.MODEL_RANGE_CLICK
    ? {
      car: {
        modelRangeName: text.toLowerCase()
      }
    }
    : {})));
}

exports.sendDataLayerEvent = sendDataLayerEvent;
