// @porschehn/navigation/dist/cjs/data-layer-event-sender-297098a6.js
'use strict';

const navContentStore = require('./nav-content-store-792bb73e.js');
const globalDataLayerService = require('./global-data-layer-service-a697b177.js');
const analyticsManager = require('./analytics-manager-c01a9024.js');
const routingKey = require('./routing-key-39c28377.js');

function sendDataLayerEvent({ datalayerEventAction = navContentStore.EventActions.NAVIGATION_CLICK, analyticsId, text, link, elementId, clickElementType, customDatalayerProperties = {} }) {
  globalDataLayerService.globalDataLayerService.pushDataLayerEvent(datalayerEventAction, Object.assign(Object.assign({ componentClick: Object.assign({ clickElementId: analyticsId, clickElementName: text, targetElement: analyticsManager.analyticsManager.getModelTargetElement(elementId, text), clickElementType }, (clickElementType === navContentStore.ClickElementTypes.Link ? { targetUrl: link } : {})), componentDisplay: {
      displayElementId: analyticsManager.analyticsManager.calculateDisplayElementId(elementId),
      displayElementType: analyticsManager.analyticsManager.calculateDisplayElementType(routingKey.removeLastRoutingSegment(elementId), navContentStore.ElementTypes.menuItem),
      displayElementName: text
    } }, customDatalayerProperties), (datalayerEventAction === navContentStore.EventActions.MODEL_RANGE_CLICK
    ? {
      car: {
        modelRangeName: text.toLowerCase()
      }
    }
    : {})));
}

exports.sendDataLayerEvent = sendDataLayerEvent;
