// @porschehn/navigation/dist/cjs/double-drawer-animation-sequences-48f2e029.js
'use strict';

const animationManager = require('./animation-manager-697ad19b.js');
const animations = require('./animations-34f77238.js');

const resetToFarRight = (el) => {
  el.style.left = '150%';
  el.style.position = 'relative';
};
const mainDrawerHTMLSelector = '#main-drawer > div:first-child';
const registerOpenDoubleDrawerSequence = ({ onAnimationStarts, onAnimationEnds }) => {
  const sequence = [
    [
      {
        onStart: onAnimationStarts,
        elementSelector: 'phn-double-drawer',
        animation: {
          className: 'opening-animation',
          cleanUpAfterAnimation: (element) => {
            element.style.left = '0px';
          }
        }
      },
      {
        elementSelector: 'phn-backdrop',
        animation: {
          className: 'fade-in-animation',
          cleanUpAfterAnimation: (element) => {
            element.style.opacity = '1';
            element.style.backgroundColor = 'rgba(0, 0, 0, 0.6)';
          }
        },
        onEnd: onAnimationEnds
      }
    ]
  ];
  animationManager.animationManager.register(animations.AnimationSequences.OPEN_DOUBLE_DRAWER, animationManager.makeSequence(sequence));
};
const registerCloseDoubleDrawerSequence = ({ onBackdropFadeOutEnd, onAnimationStarts, onAnimationEnds }) => {
  const sequence = [
    [
      {
        elementSelector: 'phn-double-drawer',
        onStart: onAnimationStarts,
        animation: {
          className: 'closing-animation',
          cleanUpBeforeAnimation: (element) => {
            element.style.left = '';
          }
        }
      },
      {
        elementSelector: 'phn-backdrop',
        animation: {
          className: 'fade-out-animation',
          cleanUpBeforeAnimation: (element) => {
            element.style.opacity = '';
            element.style.backgroundColor = '';
          }
        },
        onEnd: () => {
          onBackdropFadeOutEnd();
          onAnimationEnds();
        }
      }
    ]
  ];
  animationManager.animationManager.register(animations.AnimationSequences.CLOSE_DOUBLE_DRAWER, animationManager.makeSequence(sequence));
};
const registerTransitionSequence = ({ onSlideMainDrawerOutEnd }, direction = 'normal') => {
  animationManager.animationManager.register(`transitionSameLevelDoubleDrawer${direction === 'reverse' ? 'Reverse' : ''}`, animationManager.makeSequence([
    [
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          className: direction === 'reverse' ? 'transition-backward-animation-out' : 'transition-forward-animation-out',
          cleanUpAfterAnimation: resetToFarRight
        },
        onEnd: onSlideMainDrawerOutEnd
      }
    ],
    [
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          className: direction === 'reverse' ? 'transition-forward-animation-in' : 'transition-forward-animation-in',
          cleanUpBeforeAnimation: (element) => {
            element.style.left = '';
            element.style.position = '';
          }
        }
      }
    ]
  ]));
};
const registerForwardToNextMenuSequence = ({ onSlideMainDrawerOutEnd }) => {
  animationManager.animationManager.register('transitionNextLevelDoubleDrawer', animationManager.makeSequence([
    [
      {
        elementSelector: '#side-drawer > div',
        animation: {
          className: 'transition-forward-animation-out'
        }
      },
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          className: 'transition-forward-animation-out'
        },
        onEnd: onSlideMainDrawerOutEnd
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div',
        animation: {
          className: 'transition-forward-animation-in'
        }
      },
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          className: 'transition-forward-animation-in'
        }
      }
    ]
  ]));
};
const registerBackToPreviousMenuSequence = ({ onSlideMainDrawerOutEnd }) => {
  animationManager.animationManager.register('transitionPreviousLevelDoubleDrawer', animationManager.makeSequence([
    [
      {
        elementSelector: '#side-drawer > div',
        animation: {
          className: 'transition-backward-animation-out'
        }
      },
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          className: 'transition-backward-animation-out'
        },
        onEnd: onSlideMainDrawerOutEnd
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div',
        animation: {
          className: 'transition-backward-animation-in'
        }
      },
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          className: 'transition-backward-animation-in'
        }
      }
    ]
  ]));
};
const isPlayingDoubleDrawerAnimation = () => {
  const doubleDrawerAnimationSequences = [
    animations.AnimationSequences.OPEN_DOUBLE_DRAWER,
    animations.AnimationSequences.CLOSE_DOUBLE_DRAWER,
    animations.AnimationSequences.TRANSITION_NEXT_LEVEL_DOUBLE_DRAWER,
    animations.AnimationSequences.TRANSITION_PREVIOUS_LEVEL_DOUBLE_DRAWER,
    animations.AnimationSequences.TRANSITION_SAME_LEVEL_DOUBLE_DRAWER,
    animations.AnimationSequences.TRANSITION_SAME_LEVEL_DOUBLE_DRAWER_REVERSE
  ];
  return doubleDrawerAnimationSequences.some((sequence) => {
    try {
      return animationManager.animationManager.isPlaying(sequence);
    }
    catch (_a) {
      return false;
    }
  });
};

exports.isPlayingDoubleDrawerAnimation = isPlayingDoubleDrawerAnimation;
exports.registerBackToPreviousMenuSequence = registerBackToPreviousMenuSequence;
exports.registerCloseDoubleDrawerSequence = registerCloseDoubleDrawerSequence;
exports.registerForwardToNextMenuSequence = registerForwardToNextMenuSequence;
exports.registerOpenDoubleDrawerSequence = registerOpenDoubleDrawerSequence;
exports.registerTransitionSequence = registerTransitionSequence;
