// @porschehn/navigation/dist/cjs/double-drawer-animation-sequences-e0155702.js
'use strict';

const animationManager = require('./animation-manager-cd2e5425.js');
const animations = require('./animations-dcc5b59a.js');

const resetToFarRight = (el) => {
  el.style.left = '';
  el.style.right = '150%';
};
const mainDrawerHTMLSelector = '#main-drawer > div:first-child';
const registerOpenDoubleDrawerSequence = ({ onAnimationStarts, onAnimationEnds }) => {
  const sequence = [
    [
      {
        onStart: onAnimationStarts,
        elementSelector: 'phn-double-drawer',
        animation: {
          name: 'slideInFromLeft',
          duration: animations.AnimationDurations.drawerOpen,
          from: -1000
        }
      },
      {
        elementSelector: 'phn-backdrop',
        animation: {
          name: 'fadeInBackdrop',
          duration: animations.AnimationDurations.backdrop,
          to: 0.6
        },
        onEnd: onAnimationEnds
      }
    ]
  ];
  animationManager.animationManager.register(animations.AnimationSequences.OPEN_DOUBLE_DRAWER, animationManager.makeSequence(sequence, { timingFunction: 'cubicBezierStart' }));
};
const registerCloseDoubleDrawerSequence = ({ onBackdropFadeOutEnd, onAnimationStarts, onAnimationEnds }) => {
  const sequence = [
    [
      {
        elementSelector: 'phn-double-drawer',
        onStart: onAnimationStarts,
        animation: {
          name: 'slideOutToLeft',
          duration: animations.AnimationDurations.drawer,
          to: -1000
        }
      },
      {
        elementSelector: 'phn-backdrop',
        animation: {
          name: 'fadeOutBackdrop',
          duration: animations.AnimationDurations.backdrop,
          from: 0.6
        },
        onEnd: () => {
          onBackdropFadeOutEnd();
          onAnimationEnds();
        }
      }
    ]
  ];
  animationManager.animationManager.register(animations.AnimationSequences.CLOSE_DOUBLE_DRAWER, animationManager.makeSequence(sequence, { timingFunction: 'cubicBezierEnd' }));
};
const registerTransitionSequence = ({ onGrowHighlightEnd, onSlideMainDrawerOutEnd }, direction = 'normal') => {
  animationManager.animationManager.register(`transitionSameLevelDoubleDrawer${direction === 'reverse' ? 'Reverse' : ''}`, animationManager.makeSequence([
    [
      {
        elementSelector: '#side-drawer > div > phn-side-drawer > nav > .item-highlight',
        animation: { name: 'shrinkElement', duration: animations.AnimationDurations.highlight }
      }
    ],
    [
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          name: direction === 'reverse' ? 'slideOutToRightRelative' : 'slideOutToLeftRelative',
          to: -150,
          cleanUpAfterAnimation: resetToFarRight,
          duration: animations.AnimationDurations.drawerContent
        },
        onEnd: onSlideMainDrawerOutEnd
      }
    ],
    [
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          name: direction === 'reverse' ? 'slideInFromLeftRelative' : 'slideInFromRightRelative',
          from: -150,
          duration: animations.AnimationDurations.drawerContent
        }
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div > phn-side-drawer > nav > .item-highlight',
        animation: { name: 'growElement', duration: animations.AnimationDurations.highlight },
        onEnd: onGrowHighlightEnd
      }
    ]
  ], { timingFunction: 'easeInOutQuad' }));
};
const registerForwardToNextMenuSequence = ({ onSlideMainDrawerOutEnd, onGrowHighlightEnd }) => {
  animationManager.animationManager.register('transitionNextLevelDoubleDrawer', animationManager.makeSequence([
    [
      {
        elementSelector: '#side-drawer > div > phn-side-drawer > nav > .item-highlight',
        animation: { name: 'shrinkElement', duration: animations.AnimationDurations.highlight }
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div',
        animation: {
          name: 'slideOutToLeftRelative',
          to: -150,
          cleanUpAfterAnimation: resetToFarRight,
          duration: animations.AnimationDurations.drawerContent
        }
      },
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          name: 'slideOutToLeftRelative',
          to: -150,
          cleanUpAfterAnimation: resetToFarRight,
          duration: animations.AnimationDurations.drawerContent
        },
        onEnd: onSlideMainDrawerOutEnd
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div',
        animation: {
          name: 'slideInFromRightRelative',
          from: -150,
          cleanUpAfterAnimation: true,
          duration: animations.AnimationDurations.drawerContent
        }
      },
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          name: 'slideInFromRightRelative',
          from: -150,
          cleanUpAfterAnimation: true,
          duration: animations.AnimationDurations.drawerContent
        }
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div > phn-side-drawer > nav > .item-highlight',
        animation: { name: 'growElement', duration: animations.AnimationDurations.highlight },
        onEnd: onGrowHighlightEnd
      }
    ]
  ], { timingFunction: 'easeInOutQuad' }));
};
const registerBackToPreviousMenuSequence = ({ onGrowHighlightEnd, onSlideMainDrawerOutEnd }) => {
  animationManager.animationManager.register('transitionPreviousLevelDoubleDrawer', animationManager.makeSequence([
    [
      {
        elementSelector: '#side-drawer > div > phn-side-drawer > nav > .item-highlight',
        animation: { name: 'shrinkElement', duration: animations.AnimationDurations.highlight }
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div',
        animation: { name: 'slideOutToRightRelative', to: -150, duration: animations.AnimationDurations.drawerContent }
      },
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: { name: 'slideOutToRightRelative', to: -150, duration: animations.AnimationDurations.drawerContent },
        onEnd: onSlideMainDrawerOutEnd
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div',
        animation: {
          name: 'slideInFromLeftRelative',
          from: -150,
          cleanUpAfterAnimation: true,
          duration: animations.AnimationDurations.drawerContent
        }
      },
      {
        elementSelector: mainDrawerHTMLSelector,
        animation: {
          name: 'slideInFromLeftRelative',
          from: -150,
          cleanUpAfterAnimation: true,
          duration: animations.AnimationDurations.drawerContent
        }
      }
    ],
    [
      {
        elementSelector: '#side-drawer > div > phn-side-drawer > nav > .item-highlight',
        animation: { name: 'growElement', duration: animations.AnimationDurations.highlight },
        onEnd: onGrowHighlightEnd
      }
    ]
  ], { timingFunction: 'easeInOutQuad' }));
};
const isPlayingDoubleDrawerAnimation = () => {
  const doubleDrawerAnimationSequences = [
    animations.AnimationSequences.OPEN_DOUBLE_DRAWER,
    animations.AnimationSequences.CLOSE_DOUBLE_DRAWER,
    animations.AnimationSequences.TRANSITION_NEXT_LEVEL_DOUBLE_DRAWER,
    animations.AnimationSequences.TRANSITION_PREVIOUS_LEVEL_DOUBLE_DRAWER,
    animations.AnimationSequences.TRANSITION_SAME_LEVEL_DOUBLE_DRAWER,
    animations.AnimationSequences.TRANSITION_SAME_LEVEL_DOUBLE_DRAWER_REVERSE
  ];
  return doubleDrawerAnimationSequences.some((sequence) => {
    try {
      return animationManager.animationManager.isPlaying(sequence);
    }
    catch (_a) {
      return false;
    }
  });
};

exports.isPlayingDoubleDrawerAnimation = isPlayingDoubleDrawerAnimation;
exports.registerBackToPreviousMenuSequence = registerBackToPreviousMenuSequence;
exports.registerCloseDoubleDrawerSequence = registerCloseDoubleDrawerSequence;
exports.registerForwardToNextMenuSequence = registerForwardToNextMenuSequence;
exports.registerOpenDoubleDrawerSequence = registerOpenDoubleDrawerSequence;
exports.registerTransitionSequence = registerTransitionSequence;
