// @porschehn/navigation/dist/cjs/phn-logo.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-947214e4.js');
const globalDataLayerService = require('./global-data-layer-service-a697b177.js');
const index$1 = require('./index-62f113b0.js');
const navContentStore = require('./nav-content-store-792bb73e.js');
const naviError = require('./navi-error-3887a972.js');
const shopHelper = require('./shop-helper-c61c217b.js');

const logoCss = "";

const Logo = class {
  constructor(hostRef) {
    index.registerInstance(this, hostRef);
    this.logoLoaded = index.createEvent(this, "logoLoaded", 7);
    this.size = undefined;
    this.locale = undefined;
    this.app = undefined;
    this.theme = naviError.NavigationTheme.light;
    this.mode = naviError.NavigationMode.navbar;
    this.breakpoint = undefined;
  }
  sendDataLayerEvent(crest) {
    globalDataLayerService.globalDataLayerService.pushDataLayerEvent(navContentStore.EventActions.CREST_CLICK, {
      componentClick: {
        clickElementId: crest.id,
        clickElementName: crest.text,
        clickElementType: navContentStore.ClickElementTypes.Link,
        targetUrl: crest.link
      }
    });
  }
  componentWillLoad() {
    index$1.componentsJs.componentsReady(this.el) // TR: I believe this first call waits for the phn-logo element to be ready
      .then(() => index$1.componentsJs.componentsReady(this.el)) // TR: Then this second call actually waits for its children (i.e.: phn-p-marque) to be ready
      .then(() => {
      this.logoLoaded.emit();
    });
  }
  getLogoLink(locale, app, crest, shop) {
    return shopHelper.isShopAvailableAndLogoLinksToShop(locale, app, shop) ? shop.link : crest.link;
  }
  getLogoAriaLabel(locale, app, shop) {
    return shopHelper.isShopAvailableAndLogoLinksToShop(locale, app, shop) ? 'Porsche Shop Homepage' : 'Porsche.com Homepage';
  }
  render() {
    const { crest, shop } = navContentStore.navContentStore.state;
    const logoLink = this.getLogoLink(this.locale, this.app, crest, shop);
    const logoAriaLabel = this.getLogoAriaLabel(this.locale, this.app, shop);
    return naviError.isDesktop(this.breakpoint) ? (index.h("phn-p-wordmark", { onClick: () => this.sendDataLayerEvent(crest), href: naviError.isString(logoLink) ? logoLink : undefined, aria: { 'aria-label': logoAriaLabel }, size: "small", theme: naviError.getRightTheme(this.theme, this.mode) })) : (index.h("phn-p-crest", { onClick: () => this.sendDataLayerEvent(crest), href: naviError.isString(logoLink) ? logoLink : undefined, aria: { 'aria-label': logoAriaLabel } }));
  }
  get el() { return index.getElement(this); }
};
Logo.style = logoCss;

exports.phn_logo = Logo;
