// @porschehn/navigation/dist/cjs/shop-helper-c61c217b.js
'use strict';

const naviError = require('./navi-error-3887a972.js');
const navContentStore = require('./nav-content-store-792bb73e.js');

const shopStateStore = navContentStore.createStore({
  wishlistItemsNumber: 0,
  numberOfShoppingItems: 0,
  shopInformation: { numberOfShoppingItems: 0, wishlistItemsNumber: 0, subTotal: '0' }
});

function isShopAvailable(locale, app) {
  var _a;
  const shopList = (_a = navContentStore.navContentStore.state.shop) === null || _a === void 0 ? void 0 : _a.availableLocales;
  return shopList !== undefined && shopList.includes(locale) && app === naviError.Application.shop;
}
function logoLinksToShop(locale) {
  var _a, _b;
  return ((_b = (_a = navContentStore.navContentStore.state.shop) === null || _a === void 0 ? void 0 : _a.logoLinksToShopLocales) === null || _b === void 0 ? void 0 : _b.includes(locale)) || false;
}
function isShopAvailableAndLogoLinksToShop(locale, app, shop) {
  return isShopAvailable(locale, app) && logoLinksToShop(locale) && shop !== null;
}
function modifyShopItemDependingOnMarket(locale, app, content) {
  const { shop, externalShop } = content;
  if (shop === null)
    return content;
  if (externalShop && app !== naviError.Application.shop) {
    return replaceShopWithExternalShop(content, shop, externalShop);
  }
  if (shopNotAvailableOnGlobalPages(shop, locale, app)) {
    return removeShopItemFromContent(content, shop);
  }
  return content;
}
function replaceShopWithExternalShop(content, shop, externalShop) {
  return Object.assign(Object.assign({}, content), { shop: Object.assign(Object.assign({}, shop), externalShop) });
}
function shopNotAvailableOnGlobalPages(shop, locale, app) {
  return (!(shop === null || shop === void 0 ? void 0 : shop.availableLocales.includes(locale)) ||
    ((shop === null || shop === void 0 ? void 0 : shop.availableLocalesOnlyForShop.includes(locale)) && app !== naviError.Application.shop));
}
function removeShopItemFromContent(content, shop) {
  return Object.assign(Object.assign({}, content), { shop: Object.assign(Object.assign({}, shop), { text: '' }) });
}
function filterWishlistItemsByAuthStatus(menuItems, authStatusList) {
  return menuItems === null || menuItems === void 0 ? void 0 : menuItems.filter((item) => {
    if (item.authStatus === null || item.authStatus === undefined) {
      return authStatusList.includes(null);
    }
    return authStatusList.includes(item.authStatus);
  });
}
function getAvailableShopVariables() {
  const { myPorsche } = navContentStore.navContentStore.state;
  const { loggedInState } = navContentStore.navStateStore.state;
  let { wishlistItemsNumber } = shopStateStore.state.shopInformation;
  if (wishlistItemsNumber === 0) {
    wishlistItemsNumber = shopStateStore.state.wishlistItemsNumber;
  }
  const username = naviError.getUsername(loggedInState, (myPorsche === null || myPorsche === void 0 ? void 0 : myPorsche.loggedInFallback) || '');
  return { wishlistItemsNumber, username };
}
function loadsShopPropsOnStore(props) {
  shopStateStore.set('wishlistItemsNumber', props.wishlistItemsNumber);
  shopStateStore.set('numberOfShoppingItems', props.numberOfShoppingItems);
  shopStateStore.set('shopInformation', props.shopInformation);
}
function getWishlistIconName() {
  const { shopInformation } = shopStateStore.state;
  const { wishlistItemsNumber } = shopStateStore.state;
  const wishlistNumber = shopInformation.wishlistItemsNumber > 0 ? shopInformation.wishlistItemsNumber : wishlistItemsNumber;
  return wishlistNumber > 0 ? 'heart-filled' : 'heart';
}
function getIconNameByMenuItemId(id) {
  const iconsGetterFunctions = {
    wishlist: getWishlistIconName
  };
  const getIconName = iconsGetterFunctions[id];
  if (typeof getIconName !== 'function')
    return null;
  return getIconName();
}

exports.filterWishlistItemsByAuthStatus = filterWishlistItemsByAuthStatus;
exports.getAvailableShopVariables = getAvailableShopVariables;
exports.getIconNameByMenuItemId = getIconNameByMenuItemId;
exports.isShopAvailable = isShopAvailable;
exports.isShopAvailableAndLogoLinksToShop = isShopAvailableAndLogoLinksToShop;
exports.loadsShopPropsOnStore = loadsShopPropsOnStore;
exports.modifyShopItemDependingOnMarket = modifyShopItemDependingOnMarket;
exports.shopStateStore = shopStateStore;
