// @porschehn/navigation/dist/cjs/single-drawer-animation-sequences-2abdefe6.js
'use strict';

const animationManager = require('./animation-manager-cd2e5425.js');
const animations = require('./animations-dcc5b59a.js');

const registerOpenSingleDrawerAnimation = ({ onAnimationStarts, onAnimationEnds }) => {
  const sequence = [
    [
      {
        onStart: onAnimationStarts,
        elementSelector: 'phn-single-drawer .drawer.animating',
        animation: {
          name: 'slideInFromLeft',
          from: -1000,
          duration: animations.AnimationDurations.drawerOpen,
          cleanUpAfterAnimation: true
        }
      },
      {
        elementSelector: 'phn-backdrop',
        animation: {
          name: 'fadeInBackdrop',
          duration: animations.AnimationDurations.backdrop,
          to: 0.6
        },
        onEnd: onAnimationEnds
      }
    ]
  ];
  animationManager.animationManager.register(animations.AnimationSequences.OPEN_SINGLE_DRAWER, animationManager.makeSequence(sequence, { timingFunction: 'cubicBezierStart' }));
};
const registerCloseSingleDrawerAnimation = ({ onAnimationStarts, onAnimationEnds }) => {
  const sequence = [
    [
      {
        onStart: onAnimationStarts,
        elementSelector: 'phn-single-drawer .drawer.animating',
        animation: {
          name: 'slideOutToLeft',
          duration: animations.AnimationDurations.drawer,
          to: -1000
        }
      },
      {
        elementSelector: 'phn-backdrop',
        animation: {
          name: 'fadeOutBackdrop',
          duration: animations.AnimationDurations.backdrop,
          from: 0.6
        },
        onEnd: onAnimationEnds
      }
    ]
  ];
  animationManager.animationManager.register(animations.AnimationSequences.CLOSE_SINGLE_DRAWER, animationManager.makeSequence(sequence, { timingFunction: 'cubicBezierEnd' }));
};
const registerTransitionNextLevelSingleDrawerAnimation = ({ onAnimationEnds }) => {
  animationManager.animationManager.register(animations.AnimationSequences.TRANSITION_NEXT_LEVEL_SINGLE_DRAWER, animationManager.makeSequence([
    [
      {
        elementSelector: 'phn-single-drawer .drawer.animating',
        animation: {
          name: 'slideInFromRightFixed',
          duration: animations.AnimationDurations.drawer,
          timingFunction: 'easeInOutQuad'
        },
        onEnd: onAnimationEnds
      }
    ]
  ]));
};
const registerTransitionPreviousLevelSingleDrawerAnimation = ({ onAnimationEnds }) => {
  animationManager.animationManager.register(animations.AnimationSequences.TRANSITION_PREVIOUS_LEVEL_SINGLE_DRAWER, animationManager.makeSequence([
    [
      {
        elementSelector: 'phn-single-drawer .drawer.animating',
        animation: {
          name: 'slideOutToRightFixed',
          duration: animations.AnimationDurations.drawer,
          timingFunction: 'easeInOutQuad'
        },
        onEnd: onAnimationEnds
      }
    ]
  ]));
};
const isPlayingSingleDrawerAnimation = () => {
  try {
    const singleDrawerAnimationSequences = [
      animations.AnimationSequences.OPEN_SINGLE_DRAWER,
      animations.AnimationSequences.CLOSE_SINGLE_DRAWER,
      animations.AnimationSequences.TRANSITION_NEXT_LEVEL_SINGLE_DRAWER,
      animations.AnimationSequences.TRANSITION_PREVIOUS_LEVEL_SINGLE_DRAWER
    ];
    return singleDrawerAnimationSequences.some((sequence) => {
      try {
        return animationManager.animationManager.isPlaying(sequence);
      }
      catch (_a) {
        return false;
      }
    });
  }
  catch (e) {
    return false;
  }
};

exports.isPlayingSingleDrawerAnimation = isPlayingSingleDrawerAnimation;
exports.registerCloseSingleDrawerAnimation = registerCloseSingleDrawerAnimation;
exports.registerOpenSingleDrawerAnimation = registerOpenSingleDrawerAnimation;
exports.registerTransitionNextLevelSingleDrawerAnimation = registerTransitionNextLevelSingleDrawerAnimation;
exports.registerTransitionPreviousLevelSingleDrawerAnimation = registerTransitionPreviousLevelSingleDrawerAnimation;
