// @porschehn/navigation/dist/collection/components/single-drawer/animations/sequences/register-close-single-drawer-animation.js
import animationManager, { makeSequence } from "../../../../services/animations/animation-manager";
import { AnimationSequences } from "../../../../utility/constants";
export default function registerCloseSingleDrawerAnimation({ onAnimationStarts, onAnimationEnds }) {
  const sequence = [
    [
      {
        onStart: onAnimationStarts,
        elementSelector: 'phn-single-drawer',
        animation: {
          className: 'slide-out-to-left',
          cleanUpBeforeAnimation: (element) => {
            element.style.left = '';
          },
          cleanUpAfterAnimation: (element) => {
            element.style.left = '-1000px';
          }
        }
      },
      {
        elementSelector: 'phn-backdrop',
        animation: {
          className: 'fade-out-animation',
          cleanUpBeforeAnimation: (element) => {
            element.style.backgroundColor = '';
            element.style.opacity = '';
          }
        },
        onEnd: onAnimationEnds
      }
    ]
  ];
  animationManager.register(AnimationSequences.CLOSE_SINGLE_DRAWER, makeSequence(sequence));
}
