// @porschehn/navigation/dist/collection/components/single-drawer/animations/sequences/register-open-single-drawer-animation.js
import animationManager, { makeSequence } from "../../../../services/animations/animation-manager";
import { AnimationSequences } from "../../../../utility/constants";
export default function registerOpenSingleDrawerAnimation({ onAnimationStarts, onAnimationEnds }) {
  const sequence = [
    [
      {
        onStart: onAnimationStarts,
        elementSelector: 'phn-single-drawer',
        animation: {
          className: 'slide-in-from-left',
          cleanUpAfterAnimation: (element) => {
            element.style.left = '0px';
          }
        }
      },
      {
        elementSelector: 'phn-backdrop',
        animation: {
          className: 'fade-in-animation',
          cleanUpAfterAnimation: (element) => {
            element.style.backgroundColor = 'rgba(0, 0, 0, 0.6)';
            element.style.opacity = '1';
          }
        },
        onEnd: onAnimationEnds
      }
    ]
  ];
  animationManager.register(AnimationSequences.OPEN_SINGLE_DRAWER, makeSequence(sequence));
}
