// @porschehn/navigation/dist/collection/services/analytics/analytics-manager.js
import { determineMainDrawerLevel, getLastRoutingSegment, getPreviousRoutingKey } from "../../entities/routing-key";
import { routerStore } from "../../state/router-store";
import { DEALER_SEARCH_ELEMENT_TYPE, ElementTypes } from "../../utility/constants";
import { isString } from "../../utility/helper";
import { getSideDrawerRoutingKey } from "../menu-items-service";
const state = {};
function setDrawerType(drawerType) {
  state.drawerType = drawerType;
}
function calculateDisplayElementId(routingKeyForNavigatingForward) {
  const { drawerType } = state;
  const mainDrawerRoutingKey = (isString(routingKeyForNavigatingForward)
    ? getPreviousRoutingKey(routingKeyForNavigatingForward)
    : routerStore.state.routingKeyHistory[routerStore.state.routingKeyPointer]) || '';
  const lastMainDrawerRoutingKey = getLastRoutingSegment(mainDrawerRoutingKey);
  const mainDrawerLevel = determineMainDrawerLevel(mainDrawerRoutingKey);
  const sideDrawerLevel = mainDrawerLevel - 1;
  if (!isString(mainDrawerRoutingKey)) {
    return 'level0';
  }
  const mainDrawerDisplayElementId = isString(mainDrawerRoutingKey)
    ? `level${mainDrawerLevel}-${lastMainDrawerRoutingKey}`
    : `level${mainDrawerLevel - 1}-unknown`;
  if (drawerType === 'doubleDrawer') {
    if (mainDrawerRoutingKey === '') {
      return `level${sideDrawerLevel}-unknown_level${mainDrawerLevel}-unknown`;
    }
    const sideDrawerRoutingKey = getSideDrawerRoutingKey(mainDrawerRoutingKey);
    return `level${sideDrawerLevel}-${sideDrawerRoutingKey}_${mainDrawerDisplayElementId}`;
  }
  return mainDrawerDisplayElementId;
}
function calculateDisplayElementType(activeRoutingKey, elementType, isSideDrawer) {
  const { drawerType } = state;
  const mainDrawerLevel = determineMainDrawerLevel(activeRoutingKey);
  if (drawerType === 'doubleDrawer') {
    let sideDrawerLevel = mainDrawerLevel;
    if (isSideDrawer) {
      sideDrawerLevel = sideDrawerLevel - 1;
    }
    return `level${sideDrawerLevel}_${elementType}`;
  }
  else {
    return isString(activeRoutingKey)
      ? `level${mainDrawerLevel}_${elementType}`
      : `level${mainDrawerLevel - 1}_${elementType}`;
  }
}
function getModelName(bodyType) {
  const urlSplitIntoPieces = bodyType.link.split('/');
  const lastElementInUrlSplit = urlSplitIntoPieces[urlSplitIntoPieces.length - 1];
  let modelName;
  if (lastElementInUrlSplit === '') {
    if (urlSplitIntoPieces[urlSplitIntoPieces.length - 2] === undefined) {
      modelName = '';
    }
    else {
      modelName = urlSplitIntoPieces[urlSplitIntoPieces.length - 2].split('-').join(' ');
    }
  }
  else {
    modelName = lastElementInUrlSplit.split('-').join(' ');
  }
  return modelName;
}
function getModelTargetElement(elementId, text) {
  if (elementId.includes('models')) {
    const mainDrawerLevel = determineMainDrawerLevel(elementId);
    return `level${mainDrawerLevel}_${text.toLowerCase()}`;
  }
  return;
}
function getDrawerType() {
  return state.drawerType;
}
function getDealerResultsLoadDisplayElementId(elementId) {
  if (getDrawerType() === 'doubleDrawer') {
    return elementId;
  }
  else {
    return calculateDisplayElementId();
  }
}
function getDealerResultsLoadDisplayElementType() {
  if (getDrawerType() === 'doubleDrawer') {
    return DEALER_SEARCH_ELEMENT_TYPE;
  }
  else {
    return calculateDisplayElementType('', ElementTypes.dealerSearchForm);
  }
}
export default {
  setDrawerType,
  getDealerResultsLoadDisplayElementId,
  calculateDisplayElementId,
  getDealerResultsLoadDisplayElementType,
  calculateDisplayElementType,
  getModelName,
  getModelTargetElement
};
