// @porschehn/navigation/dist/collection/services/analytics/data-layer-event-sender.js
import { ClickElementTypes, ElementTypes, EventActions } from "../../utility/constants";
import globalDataLayerService from "./global-data-layer-service";
import analyticsManager from "./analytics-manager";
import { removeLastRoutingSegment } from "../../entities/routing-key";
export function sendDataLayerEvent({ datalayerEventAction = EventActions.NAVIGATION_CLICK, analyticsId, text, link, elementId, clickElementType, customDatalayerProperties = {} }) {
  globalDataLayerService.pushDataLayerEvent(datalayerEventAction, Object.assign(Object.assign({ componentClick: Object.assign({ clickElementId: analyticsId, clickElementName: text, targetElement: analyticsManager.getModelTargetElement(elementId, text), clickElementType }, (clickElementType === ClickElementTypes.Link ? { targetUrl: link } : {})), componentDisplay: {
      displayElementId: analyticsManager.calculateDisplayElementId(elementId),
      displayElementType: analyticsManager.calculateDisplayElementType(removeLastRoutingSegment(elementId), ElementTypes.menuItem),
      displayElementName: text
    } }, customDatalayerProperties), (datalayerEventAction === EventActions.MODEL_RANGE_CLICK
    ? {
      car: {
        modelRangeName: text.toLowerCase()
      }
    }
    : {})));
}
