// @porschehn/navigation/dist/collection/services/analytics/global-data-layer-service.js
import { GlobalDataLayer } from "@porsche/globaldatalayer";
import { Environments } from "../../entities/environment";
import { splitLocale } from "../../entities/locale";
import { DATA_LAYER_APPLICATION_NAME, DATA_LAYER_APP_ID } from "../../utility/constants";
import { isString } from "../../utility/helper";
function getEnvironmentAbbreviation(env) {
  switch (env) {
    case Environments.PRODUCTION:
      return 'P';
    case Environments.PREVIEW:
      return 'PP';
    default:
      return 'K';
  }
}
class GlobalDataLayerService {
  constructor() {
    this.homeNavEventId = /^PAGHomeNav_/;
    this.initialise();
  }
  initialise() {
    this.gdl = new GlobalDataLayer();
  }
  pushDataLayerEvent(eventAction, eventData) {
    if (eventData) {
      this.gdl.push(eventAction, eventData);
    }
    else {
      this.gdl.push(eventAction);
    }
  }
  setEnvironment(env) {
    this.gdl.setDefaultProperties(this.homeNavEventId, {
      context: { environment: getEnvironmentAbbreviation(env) }
    });
  }
  setPageName(pageName) {
    this.gdl.setDefaultProperties(this.homeNavEventId, {
      pageExperience: {
        pageName: pageName
      }
    });
  }
  setUser(userId, loginStatus) {
    this.gdl.setDefaultProperties(this.homeNavEventId, {
      ids: { userId },
      visitor: { loginStatus }
    });
  }
  setLocale(locale) {
    try {
      const { country, language } = splitLocale(locale);
      this.gdl.setDefaultProperties(this.homeNavEventId, {
        context: { country: country, language: language }
      });
    }
    catch (err) {
      // throw the error away
    }
  }
  setDefaultProperties(locale, pageName) {
    let defaultGDLProperties = {
      context: { applicationId: DATA_LAYER_APP_ID },
      visitor: { loginStatus: false }
    };
    try {
      const { country, language } = splitLocale(locale);
      defaultGDLProperties = Object.assign(Object.assign({}, defaultGDLProperties), { context: Object.assign(Object.assign({}, defaultGDLProperties.context), { country, language, applicationName: DATA_LAYER_APPLICATION_NAME }) });
    }
    catch (err) {
      // throw the error away
    }
    if (isString(pageName)) {
      defaultGDLProperties = Object.assign(Object.assign({}, defaultGDLProperties), { pageExperience: { pageName: pageName } });
    }
    this.gdl.setDefaultProperties(this.homeNavEventId, defaultGDLProperties);
  }
}
export default new GlobalDataLayerService();
