// @porschehn/navigation/dist/collection/services/menu-items-service.js
import { findMenuItemById } from "../entities/content";
import { isCountryInOneOfRegions, splitLocale } from "../entities/locale";
import { constructRoutingKey, findTopLevelRoutingKey, getDealerDetailsRoutingKey, getLastRoutingSegment, getModelsRoutingKey, removeLastRoutingSegment, RoutingKeys } from "../entities/routing-key";
import { dealerSearchResultsStore } from "../state/dealer-search-results-store";
import { navContentStore } from "../state/nav-content-store";
import { navStateStore } from "../state/nav-state-store";
import { EventActions } from "../utility/constants";
import { Application } from "../utility/constants/app";
import { getUsername, IsLoggedIn, isString } from "../utility/helper";
import featureToggles from "./feature-toggles";
import abTestingService from "./ab-testing/ab-testing-constructor-service";
export function getLevel1Items(content, dealerSearchResults, locale, loggedInState, unreadMessagesCount, userConsentGiven) {
  var _a, _b;
  const { dealer, models, myPorsche, vehiclePurchase, shop, experience, services } = content;
  const { savedDealerCookie, ppnDealers } = dealerSearchResults;
  const { app } = navStateStore.state;
  const savedDealer = savedDealerCookie !== null ? ppnDealers[savedDealerCookie.id] : undefined;
  // outbound link needed for Singapore market
  let outboundLink;
  if (locale === 'en-SG' && dealer.search.outboundLink) {
    outboundLink = (_a = dealer.search.outboundLink) === null || _a === void 0 ? void 0 : _a.link;
  }
  const dealerRoutingKeys = [RoutingKeys.DEALER_SEARCH];
  if (savedDealer !== undefined && ((_b = dealerSearchResultsStore.state.results) === null || _b === void 0 ? void 0 : _b.length) != 1) {
    dealerRoutingKeys.push(getDealerDetailsRoutingKey(savedDealer.ppnDealer.id));
  }
  const menu = [
    {
      elementId: RoutingKeys.MODELS,
      routingKeys: [RoutingKeys.MODELS],
      analyticsId: models.id,
      text: models.text
    },
    {
      elementId: RoutingKeys.VEHICLE_PURCHASE,
      routingKeys: [RoutingKeys.VEHICLE_PURCHASE],
      analyticsId: vehiclePurchase.id,
      text: vehiclePurchase.text
    },
    {
      elementId: RoutingKeys.SERVICES,
      routingKeys: [RoutingKeys.SERVICES],
      analyticsId: services.id,
      text: services.text
    },
    {
      elementId: RoutingKeys.EXPERIENCE,
      routingKeys: [RoutingKeys.EXPERIENCE],
      analyticsId: experience.id,
      text: experience.text
    }
  ];
  const dealerSearch = {
    elementId: RoutingKeys.DEALER_SEARCH,
    routingKeys: dealerRoutingKeys,
    analyticsId: dealer.id,
    text: dealer.text,
    link: outboundLink,
    iconRight: locale === 'en-SG' && dealer.search.outboundLink ? 'external' : 'arrow-head-right'
  };
  if (shop !== null && shop.text !== '') {
    const shopItem = {
      elementId: RoutingKeys.SHOP,
      routingKeys: [RoutingKeys.SHOP],
      analyticsId: shop.id,
      text: shop.text,
      link: app !== Application.shop ? shop.link : undefined,
      datalayerEventAction: EventActions.NAVIGATION_LINK_CLICK,
      iconRight: 'external'
    };
    menu.splice(2, 0, shopItem);
  }
  if (isDealerSearchEnabled(locale, userConsentGiven)) {
    const savedDealerName = splitLocale(locale).language === 'en'
      ? savedDealer === null || savedDealer === void 0 ? void 0 : savedDealer.ppnDealer.name
      : isString(savedDealer === null || savedDealer === void 0 ? void 0 : savedDealer.ppnDealer.nameLocalized)
        ? savedDealer === null || savedDealer === void 0 ? void 0 : savedDealer.ppnDealer.nameLocalized
        : savedDealer === null || savedDealer === void 0 ? void 0 : savedDealer.ppnDealer.name;
    if (isString(savedDealerName)) {
      dealerSearch.subtext = savedDealerName;
    }
  }
  else {
    dealerSearch.link = dealer.search.map.link;
    dealerSearch.onClick = undefined;
  }
  menu.push(dealerSearch);
  if (myPorsche !== null) {
    const isLoggedIn = IsLoggedIn(loggedInState);
    const username = isLoggedIn ? getUsername(loggedInState, myPorsche.loggedInFallback) : undefined;
    menu.push({
      elementId: RoutingKeys.MY_PORSCHE,
      routingKeys: [RoutingKeys.MY_PORSCHE],
      analyticsId: myPorsche.id,
      text: myPorsche.text,
      subtext: username,
      showIconModifier: isLoggedIn && unreadMessagesCount > 0
    });
  }
  return menu;
}
export function getSideDrawerItems(content, dealerSearchResults, locale, loggedInState, unreadMessagesCount, activeRoutingKey, userConsentGiven) {
  const topLevelRoutingKey = findTopLevelRoutingKey(activeRoutingKey);
  switch (topLevelRoutingKey) {
    case activeRoutingKey:
      break;
    case RoutingKeys.MODELS:
      return content.models.series.map((modelsSeries) => ({
        elementId: getModelsRoutingKey(modelsSeries.id),
        routingKeys: [getModelsRoutingKey(modelsSeries.id)],
        analyticsId: modelsSeries.id,
        text: modelsSeries.name,
        subtext: modelsSeries.tags.map((tag) => tag.text).join(' | '),
        renderSignatures: true,
        datalayerEventAction: EventActions.MODEL_RANGE_CLICK
      }));
    case RoutingKeys.SHOP:
    case RoutingKeys.VEHICLE_PURCHASE:
    case RoutingKeys.SERVICES:
    case RoutingKeys.EXPERIENCE: {
      const parentRoutingKey = removeLastRoutingSegment(activeRoutingKey);
      const menuDrawerParent = getMenuContent(content, parentRoutingKey, topLevelRoutingKey);
      if (menuDrawerParent !== null) {
        return menuDrawerParent.children.map((childItem) => ({
          elementId: constructRoutingKey(parentRoutingKey, childItem.id),
          routingKeys: [constructRoutingKey(parentRoutingKey, childItem.id)],
          analyticsId: childItem.id,
          text: childItem.text,
          link: childItem.link
        }));
      }
      break;
    }
  }
  return getLevel1Items(content, dealerSearchResults, locale, loggedInState, unreadMessagesCount, userConsentGiven);
}
export function getSideDrawerRoutingKey(activeRoutingKey) {
  const topLevelRoutingKey = findTopLevelRoutingKey(activeRoutingKey);
  switch (topLevelRoutingKey) {
    case activeRoutingKey: {
      break;
    }
    case RoutingKeys.DEALER_SEARCH: {
      return RoutingKeys.DEALER_SEARCH;
    }
    case RoutingKeys.MODELS: {
      return RoutingKeys.MODELS;
    }
    case RoutingKeys.SHOP:
    case RoutingKeys.VEHICLE_PURCHASE:
    case RoutingKeys.SERVICES:
    case RoutingKeys.EXPERIENCE: {
      const parentRoutingKey = removeLastRoutingSegment(activeRoutingKey);
      return getLastRoutingSegment(parentRoutingKey);
    }
  }
  return RoutingKeys.MAIN_MENU;
}
export function getMenuContent(content, routingKey, topLevelRoutingKey) {
  const { vehiclePurchase, experience, shop, legacyMore, services } = content;
  const menuContent = {
    [RoutingKeys.VEHICLE_PURCHASE]: vehiclePurchase,
    [RoutingKeys.EXPERIENCE]: experience,
    [RoutingKeys.SHOP]: shop,
    [RoutingKeys.MORE]: legacyMore,
    [RoutingKeys.SERVICES]: services
  }[topLevelRoutingKey];
  if (menuContent === undefined || menuContent === null) {
    return null;
  }
  if (isString(menuContent.experienceName)) {
    menuContent.children = abTestingService().executeContentChange(menuContent);
  }
  const idOfLastRoutingSegment = getLastRoutingSegment(routingKey);
  return idOfLastRoutingSegment === topLevelRoutingKey
    ? menuContent
    : findMenuItemById(menuContent, idOfLastRoutingSegment);
}
export function isDealerSearchEnabled(locale, userConsentGiven) {
  try {
    const { country } = splitLocale(locale);
    const returnDealerSearch = !isCountryInOneOfRegions(country) && userConsentGiven;
    return (featureToggles.isDealerSearchChinaEnabled() || country !== 'CN') && returnDealerSearch;
  }
  catch (_a) {
    // ignore error
  }
  return true;
}
export function getBackButtonText(content, previousRoutingKey) {
  if (!isString(previousRoutingKey)) {
    return '';
  }
  const topLevelRoutingKey = findTopLevelRoutingKey(previousRoutingKey);
  switch (topLevelRoutingKey) {
    case RoutingKeys.MODELS: {
      return content.models.text;
    }
    case RoutingKeys.DEALER_SEARCH:
      if (content.dealer.numberOfDealers === 1) {
        return '';
      }
      return content.dealer.details.backToSearch;
    case RoutingKeys.SHOP:
    case RoutingKeys.VEHICLE_PURCHASE:
    case RoutingKeys.MORE:
    case RoutingKeys.EXPERIENCE: {
      const menuContent = getMenuContent(navContentStore.state, previousRoutingKey, topLevelRoutingKey);
      if (menuContent !== null) {
        return menuContent.text;
      }
      break;
    }
  }
  return '';
}
export function getHeadlineText(content, routingKey, topLevelRoutingKey) {
  var _a;
  const { dealer, myPorsche, models, vehiclePurchase, experience, shop, legacyMore, services } = content;
  let menuContent;
  switch (topLevelRoutingKey) {
    case RoutingKeys.MORE:
    case RoutingKeys.SHOP:
    case RoutingKeys.EXPERIENCE:
    case RoutingKeys.SERVICES:
    case RoutingKeys.VEHICLE_PURCHASE: {
      menuContent = {
        [RoutingKeys.VEHICLE_PURCHASE]: vehiclePurchase,
        [RoutingKeys.EXPERIENCE]: experience,
        [RoutingKeys.SHOP]: shop,
        [RoutingKeys.MORE]: legacyMore,
        [RoutingKeys.SERVICES]: services
      }[topLevelRoutingKey];
      if (menuContent === undefined) {
        return '';
      }
      const idOfLastRoutingSegment = getLastRoutingSegment(routingKey);
      return idOfLastRoutingSegment === topLevelRoutingKey
        ? menuContent === null || menuContent === void 0 ? void 0 : menuContent.text
        : (_a = findMenuItemById(menuContent, idOfLastRoutingSegment)) === null || _a === void 0 ? void 0 : _a.text;
    }
    case RoutingKeys.SERIES:
    case RoutingKeys.MODELS: {
      const routingId = getLastRoutingSegment(routingKey);
      menuContent = models.series.find((modelsSeries) => routingId === modelsSeries.id);
      return menuContent === undefined ? models.text : menuContent === null || menuContent === void 0 ? void 0 : menuContent.id;
    }
    case RoutingKeys.DEALER_SEARCH: {
      return dealer.search.text;
    }
    case RoutingKeys.MY_PORSCHE: {
      return myPorsche === null || myPorsche === void 0 ? void 0 : myPorsche.text;
    }
  }
  return '';
}
export function getSubtext(topLevelRoutingKey, activeRoutingKey, locale) {
  const { myPorsche, models } = navContentStore.state;
  const { loggedInState } = navStateStore.state;
  if (topLevelRoutingKey === RoutingKeys.MY_PORSCHE) {
    const isLoggedIn = IsLoggedIn(loggedInState);
    return isLoggedIn ? getUsername(loggedInState, (myPorsche === null || myPorsche === void 0 ? void 0 : myPorsche.loggedInFallback) || '') : '';
  }
  if (topLevelRoutingKey === RoutingKeys.DEALER_SEARCH) {
    const { savedDealerCookie, ppnDealers } = dealerSearchResultsStore.state;
    const savedDealer = savedDealerCookie !== null ? ppnDealers[savedDealerCookie.id] : undefined;
    const savedDealerName = locale && splitLocale(locale).language === 'en'
      ? savedDealer === null || savedDealer === void 0 ? void 0 : savedDealer.ppnDealer.name
      : isString(savedDealer === null || savedDealer === void 0 ? void 0 : savedDealer.ppnDealer.nameLocalized)
        ? savedDealer === null || savedDealer === void 0 ? void 0 : savedDealer.ppnDealer.nameLocalized
        : savedDealer === null || savedDealer === void 0 ? void 0 : savedDealer.ppnDealer.name;
    return savedDealerCookie ? savedDealerName : '';
  }
  if (topLevelRoutingKey === RoutingKeys.MODELS) {
    if (isString(activeRoutingKey)) {
      const routingId = getLastRoutingSegment(activeRoutingKey);
      const menuContent = models.series.find((modelsSeries) => routingId === modelsSeries.id);
      if (menuContent) {
        return menuContent.tags.map((tag) => tag.text).join(' | ');
      }
    }
  }
  return '';
}
