// @porschehn/navigation/dist/collection/components/back-button/back-button.js
import { h, Host } from "@stencil/core";
import globalDataLayerService from "../../services/analytics/global-data-layer-service";
import { ClickElementTypes, ElementTypes, EventActions } from "../../utility/constants";
import analyticsManager from "../../services/analytics/analytics-manager";
export class BackButton {
  constructor() {
    this.text = '';
    this.analyticsId = '';
    this.activeRoutingKey = undefined;
  }
  sendDataLayerEvent() {
    globalDataLayerService.pushDataLayerEvent(EventActions.MENU_BACK_CLICK, {
      componentClick: {
        clickElementId: this.analyticsId,
        clickElementName: this.text,
        clickElementType: ClickElementTypes.Button
      },
      componentDisplay: {
        displayElementId: analyticsManager.calculateDisplayElementId(),
        displayElementType: analyticsManager.calculateDisplayElementType(this.activeRoutingKey, ElementTypes.backButton, true),
        displayElementName: this.text
      }
    });
  }
  render() {
    return (h(Host, null, h("phn-p-button-pure", { size: "inherit", icon: "arrow-head-left", class: "back-button-text-link", aria: { 'aria-label': 'Back to Menu' }, onClick: () => this.sendDataLayerEvent() }, this.text)));
  }
  static get is() { return "phn-back-button"; }
  static get encapsulation() { return "scoped"; }
  static get originalStyleUrls() {
    return {
      "$": ["back-button.scss"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["back-button.css"]
    };
  }
  static get properties() {
    return {
      "text": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "text",
        "reflect": false,
        "defaultValue": "''"
      },
      "analyticsId": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "analytics-id",
        "reflect": false,
        "defaultValue": "''"
      },
      "activeRoutingKey": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": true,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "active-routing-key",
        "reflect": false
      }
    };
  }
}
