// @porschehn/navigation/dist/collection/components/menu-item-card/menu-item-card.js
import { Fragment, h, Host } from "@stencil/core";
import { ClickElementTypes, EventActions } from "../../utility/constants";
import { getHTMLElement, isString } from "../../utility/helper";
import { sendDataLayerEvent } from "../../services/analytics/data-layer-event-sender";
export class MenuItemCard {
  constructor() {
    this.text = undefined;
    this.analyticsId = undefined;
    this.subtext = '';
    this.elementId = '';
    this.link = '';
    this.datalayerEventAction = EventActions.NAVIGATION_CLICK;
    this.customDatalayerProperties = {};
    this.target = '_self';
    this.iconLeft = '';
  }
  handleKeyDown(e) {
    var _a;
    if (e.key === 'Enter') {
      const anchorElement = (_a = getHTMLElement(e.target)) === null || _a === void 0 ? void 0 : _a.querySelector('a');
      const targetElement = anchorElement ? anchorElement : getHTMLElement(e.target);
      if (targetElement) {
        targetElement.click();
        !anchorElement ? this.sendDataLayerEvent(ClickElementTypes.Button, this.elementId) : null;
      }
    }
  }
  sendDataLayerEvent(clickElementType, elementId) {
    sendDataLayerEvent({
      analyticsId: this.analyticsId,
      text: this.text,
      elementId,
      clickElementType,
      link: this.link,
      customDatalayerProperties: this.customDatalayerProperties,
      datalayerEventAction: this.datalayerEventAction
    });
  }
  renderTitle() {
    return (h("div", { class: "phn-flex-row phn-align-center phn-justify-between title-container" }, this.elementId ? (h("phn-p-text", { className: 'dealer-card-item' }, this.text)) : (h("phn-p-heading", { tag: "h3", size: "medium" }, this.text)), isString(this.iconLeft) ? h("phn-p-icon", { name: this.iconLeft }) : null));
  }
  renderSubtitle() {
    if (!isString(this.subtext)) {
      return null;
    }
    return (h("phn-p-text", { class: 'subtext', color: "contrast-medium", weight: "regular", size: "xx-small", tag: "div", ellipsis: "true", "data-test-id": "subtext" }, this.subtext));
  }
  render() {
    const body = (h(Fragment, null, h("div", { class: "texts" }, this.renderTitle(), this.renderSubtitle()), h("slot", null)));
    if (isString(this.link))
      return (h("a", { tabIndex: -1, onClick: () => {
          this.sendDataLayerEvent(ClickElementTypes.Link, this.elementId);
        }, class: "card-link", href: this.link, target: this.target }, body));
    return (h(Host, null, h("button", { tabIndex: -1, onClick: () => {
        this.sendDataLayerEvent(ClickElementTypes.Button, this.elementId);
      } }, body)));
  }
  static get is() { return "phn-menu-item-card"; }
  static get encapsulation() { return "scoped"; }
  static get originalStyleUrls() {
    return {
      "$": ["menu-item-card.scss"]
    };
  }
  static get styleUrls() {
    return {
      "$": ["menu-item-card.css"]
    };
  }
  static get properties() {
    return {
      "text": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": true,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "text",
        "reflect": false
      },
      "analyticsId": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": true,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "analytics-id",
        "reflect": false
      },
      "subtext": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "subtext",
        "reflect": false,
        "defaultValue": "''"
      },
      "elementId": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "element-id",
        "reflect": false,
        "defaultValue": "''"
      },
      "link": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "link",
        "reflect": false,
        "defaultValue": "''"
      },
      "datalayerEventAction": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "EventAction",
          "resolved": "\"PAGHomeNav_General_Load\" | \"PAGHomeNav_CountryRecommender_Load\" | \"PAGHomeNav_CountryRecommenderChange_Click\" | \"PAGHomeNav_CountryRecommenderClose_Click\" | \"PAGHomeNav_Crest_Click\" | \"PAGHomeNav_Navigation_Load\" | \"PAGHomeNav_ModelRangeOverlay_Load\" | \"PAGHomeNav_MenuBack_Click\" | \"PAGHomeNav_MenuClose_Click\" | \"PAGHomeNav_MetaFunction_Click\" | \"PAGHomeNav_Model_Click\" | \"PAGHomeNav_ModelMetaFunction_Click\" | \"PAGHomeNav_ModelCTA_Click\" | \"PAGHomeNav_ModelRange_Click\" | \"PAGHomeNav_More_Click\" | \"PAGHomeNav_Navigation_Click\" | \"PAGHomeNav_NavigationLink_Click\" | \"PAGHomeNav_FindADealerSearchScreen_Load\" | \"PAGHomeNav_FindADealerResultsScreen_Load\" | \"PAGHomeNav_FindADealerSearch_Click\" | \"PAGHomeNav_FindADealerLocateMe_Click\" | \"PAGHomeNav_FindADealerDealer_Click\" | \"PAGHomeNav_FindADealerDealerScreen_Load\" | \"PAGHomeNav_FindADealerDealerContact_Click\" | \"PAGHomeNav_FindADealerMap_Click\" | \"PAGHomeNav_IconContextLink_Click\" | \"PAGHomeNav_Drawer_Load\" | \"PAGHomeNav_IconGlobalLink_Click\"",
          "references": {
            "EventAction": {
              "location": "import",
              "path": "../../utility/constants",
              "id": "src/utility/constants/index.ts::EventAction"
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "datalayer-event-action",
        "reflect": false,
        "defaultValue": "EventActions.NAVIGATION_CLICK"
      },
      "customDatalayerProperties": {
        "type": "unknown",
        "mutable": false,
        "complexType": {
          "original": "TDataLayer",
          "resolved": "{ basicProduct?: TBasicProduct | undefined; car?: TCar | undefined; componentAudio?: TComponentAudio | undefined; componentClick?: TComponentClick | undefined; componentDisplay?: TComponentDisplay | undefined; componentFile?: TComponentFile | undefined; componentFilter?: TComponentFilter | undefined; componentForm?: TComponentForm | undefined; componentNavigation?: TComponentNavigation | undefined; componentSearch?: TComponentSearch | undefined; componentSort?: TComponentSort | undefined; componentVideo?: TComponentVideo | undefined; consents?: TConsents | undefined; context?: TContext | undefined; financialProduct?: TFinancialProduct | undefined; ids?: TIds | undefined; pageExperience?: TPageExperience | undefined; partner?: TPartner | undefined; personalization?: TPersonalization | undefined; preferences?: TPreferences | undefined; process?: TProcess | undefined; serviceProduct?: TServiceProduct | undefined; socialShareConsents?: TSocialShareConsents | undefined; visitor?: TVisitor | undefined; }",
          "references": {
            "TDataLayer": {
              "location": "import",
              "path": "@porsche/globaldatalayer-events/projects/pcom/all/types",
              "id": ""
            }
          }
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "defaultValue": "{}"
      },
      "target": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "target",
        "reflect": false,
        "defaultValue": "'_self'"
      },
      "iconLeft": {
        "type": "string",
        "mutable": false,
        "complexType": {
          "original": "string",
          "resolved": "string",
          "references": {}
        },
        "required": false,
        "optional": false,
        "docs": {
          "tags": [],
          "text": ""
        },
        "attribute": "icon-left",
        "reflect": false,
        "defaultValue": "''"
      }
    };
  }
  static get elementRef() { return "el"; }
  static get listeners() {
    return [{
        "name": "keydown",
        "method": "handleKeyDown",
        "target": undefined,
        "capture": false,
        "passive": false
      }];
  }
}
