// @porschehn/navigation/dist/esm/data-layer-event-sender-2a4d9808.js
import { E as EventActions, C as ClickElementTypes, f as ElementTypes } from './nav-content-store-ad934a75.js';
import { g as globalDataLayerService } from './global-data-layer-service-2c9dad09.js';
import { a as analyticsManager } from './analytics-manager-87641e35.js';
import { r as removeLastRoutingSegment } from './routing-key-12b2ffae.js';

function sendDataLayerEvent({ datalayerEventAction = EventActions.NAVIGATION_CLICK, analyticsId, text, link, elementId, clickElementType, customDatalayerProperties = {} }) {
  globalDataLayerService.pushDataLayerEvent(datalayerEventAction, Object.assign(Object.assign({ componentClick: Object.assign({ clickElementId: analyticsId, clickElementName: text, targetElement: analyticsManager.getModelTargetElement(elementId, text), clickElementType }, (clickElementType === ClickElementTypes.Link ? { targetUrl: link } : {})), componentDisplay: {
      displayElementId: analyticsManager.calculateDisplayElementId(elementId),
      displayElementType: analyticsManager.calculateDisplayElementType(removeLastRoutingSegment(elementId), ElementTypes.menuItem),
      displayElementName: text
    } }, customDatalayerProperties), (datalayerEventAction === EventActions.MODEL_RANGE_CLICK
    ? {
      car: {
        modelRangeName: text.toLowerCase()
      }
    }
    : {})));
}

export { sendDataLayerEvent as s };
