// @porschehn/navigation/dist/esm/data-layer-event-sender-def87579.js
import { E as EventActions, C as ClickElementTypes, h as ElementTypes } from './nav-content-store-0a34ed51.js';
import { g as globalDataLayerService } from './global-data-layer-service-04808730.js';
import { a as analyticsManager } from './analytics-manager-0d4961b8.js';
import { r as removeLastRoutingSegment } from './routing-key-43812601.js';

function sendDataLayerEvent({ datalayerEventAction = EventActions.NAVIGATION_CLICK, analyticsId, text, link, elementId, clickElementType, customDatalayerProperties = {} }) {
  globalDataLayerService.pushDataLayerEvent(datalayerEventAction, Object.assign(Object.assign({ componentClick: Object.assign({ clickElementId: analyticsId, clickElementName: text, targetElement: analyticsManager.getModelTargetElement(elementId, text), clickElementType }, (clickElementType === ClickElementTypes.Link ? { targetUrl: link } : {})), componentDisplay: {
      displayElementId: analyticsManager.calculateDisplayElementId(elementId),
      displayElementType: analyticsManager.calculateDisplayElementType(removeLastRoutingSegment(elementId), ElementTypes.menuItem),
      displayElementName: text
    } }, customDatalayerProperties), (datalayerEventAction === EventActions.MODEL_RANGE_CLICK
    ? {
      car: {
        modelRangeName: text.toLowerCase()
      }
    }
    : {})));
}

export { sendDataLayerEvent as s };
